<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Non-LTR retrotrasposons and satellites | Conservation and ERVL activation of DUX family</title>
  <meta name="description" content="Bioinformatics and statistics analysis" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Non-LTR retrotrasposons and satellites | Conservation and ERVL activation of DUX family" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Bioinformatics and statistics analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Non-LTR retrotrasposons and satellites | Conservation and ERVL activation of DUX family" />
  
  <meta name="twitter:description" content="Bioinformatics and statistics analysis" />
  

<meta name="author" content="Chao-Jen Wong" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="LTR-activation.html"/>
<link rel="next" href="peaks-line1.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DUXC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#datasets"><i class="fa fa-check"></i><b>0.1</b> Datasets</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#tips"><i class="fa fa-check"></i><b>0.2</b> Tips</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="chipseq.html"><a href="chipseq.html"><i class="fa fa-check"></i><b>1</b> ChIP-seq for the canine model</a><ul>
<li class="chapter" data-level="1.1" data-path="chipseq.html"><a href="chipseq.html#chip-seq-samples"><i class="fa fa-check"></i><b>1.1</b> ChIP-seq samples</a></li>
<li class="chapter" data-level="1.2" data-path="chipseq.html"><a href="chipseq.html#workflow"><i class="fa fa-check"></i><b>1.2</b> workflow</a></li>
<li class="chapter" data-level="1.3" data-path="chipseq.html"><a href="chipseq.html#peaks-and-functional-distribution"><i class="fa fa-check"></i><b>1.3</b> Peaks and functional distribution</a><ul>
<li class="chapter" data-level="1.3.1" data-path="chipseq.html"><a href="chipseq.html#functional-distribution"><i class="fa fa-check"></i><b>1.3.1</b> Functional distribution</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chipseq.html"><a href="chipseq.html#motifs-analysis-need-update"><i class="fa fa-check"></i><b>1.4</b> Motifs analysis NEED UPDATE</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="makeSE.html"><a href="makeSE.html"><i class="fa fa-check"></i><b>2</b> RNA-seq to DESeq2</a><ul>
<li class="chapter" data-level="2.1" data-path="makeSE.html"><a href="makeSE.html#rna-seq-preprocessing"><i class="fa fa-check"></i><b>2.1</b> RNA-seq preprocessing</a></li>
<li class="chapter" data-level="2.2" data-path="makeSE.html"><a href="makeSE.html#reads-counts"><i class="fa fa-check"></i><b>2.2</b> Reads counts</a><ul>
<li class="chapter" data-level="2.2.1" data-path="makeSE.html"><a href="makeSE.html#define-gene-features"><i class="fa fa-check"></i><b>2.2.1</b> Define gene features</a></li>
<li class="chapter" data-level="2.2.2" data-path="makeSE.html"><a href="makeSE.html#read-counts"><i class="fa fa-check"></i><b>2.2.2</b> Read counts</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="makeSE.html"><a href="makeSE.html#deseq2"><i class="fa fa-check"></i><b>2.3</b> DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html"><i class="fa fa-check"></i><b>3</b> DUXC transcriptome</a><ul>
<li class="chapter" data-level="3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#load-datasets-and-settings"><i class="fa fa-check"></i><b>3.1</b> Load datasets and settings</a></li>
<li class="chapter" data-level="3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cinc-duxc-expresison-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2</b> CinC: DUXC expresison in canine muscle cells</a></li>
<li class="chapter" data-level="3.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hinc-dux4-expression-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.3</b> HinC: DUX4 expression in canine muscle cells</a></li>
<li class="chapter" data-level="3.4" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#duxc-alt-expression---caltinc"><i class="fa fa-check"></i><b>3.4</b> DUXC-ALT expression - CALTinC</a><ul>
<li class="chapter" data-level="3.4.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#go-analysis-for-the-down-regulated-gene-set"><i class="fa fa-check"></i><b>3.4.1</b> GO analysis for the down-regulated gene set</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cross-species-models-dux4-expression-in-canine-and-murine"><i class="fa fa-check"></i><b>3.5</b> Cross-species models: DUX4 expression in canine and murine</a><ul>
<li class="chapter" data-level="3.5.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-duxc-expression-in-canine"><i class="fa fa-check"></i><b>3.5.1</b> DUX4 and DUXC expression in canine</a></li>
<li class="chapter" data-level="3.5.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-dux-expression-in-murine"><i class="fa fa-check"></i><b>3.5.2</b> DUX4 and Dux expression in murine</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html"><i class="fa fa-check"></i><b>4</b> DUXC and Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#duxc-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.1</b> DUXC transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs"><i class="fa fa-check"></i><b>4.1.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.1.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome"><i class="fa fa-check"></i><b>4.1.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#dux-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.2</b> Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.2.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs-1"><i class="fa fa-check"></i><b>4.2.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.2.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome-1"><i class="fa fa-check"></i><b>4.2.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html"><i class="fa fa-check"></i><b>5</b> DUXC transcriptome compared to early embryo transcriptome profile</a><ul>
<li class="chapter" data-level="5.1" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#mouse-2c-like-signature"><i class="fa fa-check"></i><b>5.1</b> Mouse 2C-like signature</a></li>
<li class="chapter" data-level="5.2" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#cleavage-stage-signature"><i class="fa fa-check"></i><b>5.2</b> cleavage-stage signature</a></li>
<li class="chapter" data-level="5.3" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#ipscs"><i class="fa fa-check"></i><b>5.3</b> iPSCs</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="repeats.html"><a href="repeats.html"><i class="fa fa-check"></i><b>6</b> Repeat elements</a><ul>
<li class="chapter" data-level="6.1" data-path="repeats.html"><a href="repeats.html#takeaway"><i class="fa fa-check"></i><b>6.1</b> Takeaway</a></li>
<li class="chapter" data-level="6.2" data-path="repeats.html"><a href="repeats.html#repeat-element-expression"><i class="fa fa-check"></i><b>6.2</b> Repeat element expression</a></li>
<li class="chapter" data-level="6.3" data-path="repeats.html"><a href="repeats.html#differential-analysis"><i class="fa fa-check"></i><b>6.3</b> Differential analysis</a><ul>
<li class="chapter" data-level="6.3.1" data-path="repeats.html"><a href="repeats.html#normalization"><i class="fa fa-check"></i><b>6.3.1</b> Normalization</a></li>
<li class="chapter" data-level="6.3.2" data-path="repeats.html"><a href="repeats.html#deseq2-results"><i class="fa fa-check"></i><b>6.3.2</b> DESeq2 results</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="repeats.html"><a href="repeats.html#repeat-family-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> Repeat family enrichment analysis</a><ul>
<li class="chapter" data-level="6.4.1" data-path="repeats.html"><a href="repeats.html#correlation-between-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>6.4.1</b> Correlation between HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="6.4.2" data-path="repeats.html"><a href="repeats.html#repeat-family-activation"><i class="fa fa-check"></i><b>6.4.2</b> Repeat family activation</a></li>
<li class="chapter" data-level="6.4.3" data-path="repeats.html"><a href="repeats.html#repeat-classs-activation"><i class="fa fa-check"></i><b>6.4.3</b> Repeat classs activation</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="repeats.html"><a href="repeats.html#ltr-retroelements-distribution-and-activation"><i class="fa fa-check"></i><b>6.5</b> LTR retroelements distribution and activation</a></li>
<li class="chapter" data-level="6.6" data-path="repeats.html"><a href="repeats.html#peaks-distribution-in-class-of-repeat-elements"><i class="fa fa-check"></i><b>6.6</b> Peaks distribution in class of repeat elements</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="LTR-activation.html"><a href="LTR-activation.html"><i class="fa fa-check"></i><b>7</b> LTR activation</a><ul>
<li class="chapter" data-level="7.1" data-path="LTR-activation.html"><a href="LTR-activation.html#takaway"><i class="fa fa-check"></i><b>7.1</b> Takaway</a></li>
<li class="chapter" data-level="7.2" data-path="LTR-activation.html"><a href="LTR-activation.html#background-in-murine-and-human"><i class="fa fa-check"></i><b>7.2</b> background in murine and human</a></li>
<li class="chapter" data-level="7.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-subfamily-activation-in-canine"><i class="fa fa-check"></i><b>7.3</b> ERVL subfamily activation in canine</a></li>
<li class="chapter" data-level="7.4" data-path="LTR-activation.html"><a href="LTR-activation.html#malr-activation-in-canine"><i class="fa fa-check"></i><b>7.4</b> MaLR activation in canine</a></li>
<li class="chapter" data-level="7.5" data-path="LTR-activation.html"><a href="LTR-activation.html#compare-wih-hinh"><i class="fa fa-check"></i><b>7.5</b> Compare wih HinH</a><ul>
<li class="chapter" data-level="7.5.1" data-path="LTR-activation.html"><a href="LTR-activation.html#copy-number"><i class="fa fa-check"></i><b>7.5.1</b> Copy number</a></li>
<li class="chapter" data-level="7.5.2" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-malr"><i class="fa fa-check"></i><b>7.5.2</b> ERVL-MaLR</a></li>
<li class="chapter" data-level="7.5.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl"><i class="fa fa-check"></i><b>7.5.3</b> ERVL</a></li>
<li class="chapter" data-level="7.5.4" data-path="LTR-activation.html"><a href="LTR-activation.html#erv1"><i class="fa fa-check"></i><b>7.5.4</b> ERV1</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="LTR-activation.html"><a href="LTR-activation.html#duxc-alt-and-ervl-transcriptions"><i class="fa fa-check"></i><b>7.6</b> DUXC-ALT and ERVL transcriptions</a><ul>
<li class="chapter" data-level="7.6.1" data-path="LTR-activation.html"><a href="LTR-activation.html#erv-subfamily-expression"><i class="fa fa-check"></i><b>7.6.1</b> ERV subfamily expression</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="non-LTR.html"><a href="non-LTR.html"><i class="fa fa-check"></i><b>8</b> Non-LTR retrotrasposons and satellites</a><ul>
<li class="chapter" data-level="8.1" data-path="non-LTR.html"><a href="non-LTR.html#lines"><i class="fa fa-check"></i><b>8.1</b> LINEs</a><ul>
<li class="chapter" data-level="8.1.1" data-path="non-LTR.html"><a href="non-LTR.html#l1s-activation"><i class="fa fa-check"></i><b>8.1.1</b> L1s activation</a></li>
<li class="chapter" data-level="8.1.2" data-path="non-LTR.html"><a href="non-LTR.html#line1-and-duxcdux4-binding-sites"><i class="fa fa-check"></i><b>8.1.2</b> LINE1 and DUXC/DUX4 binding sites</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="non-LTR.html"><a href="non-LTR.html#sine"><i class="fa fa-check"></i><b>8.2</b> SINE</a><ul>
<li class="chapter" data-level="8.2.1" data-path="non-LTR.html"><a href="non-LTR.html#distribution-of-sine-family"><i class="fa fa-check"></i><b>8.2.1</b> Distribution of SINE family</a></li>
<li class="chapter" data-level="8.2.2" data-path="non-LTR.html"><a href="non-LTR.html#expression-of-sines-in-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>8.2.2</b> Expression of SINEs in HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="8.2.3" data-path="non-LTR.html"><a href="non-LTR.html#trna-lys"><i class="fa fa-check"></i><b>8.2.3</b> tRNA-Lys</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="non-LTR.html"><a href="non-LTR.html#satellites"><i class="fa fa-check"></i><b>8.3</b> Satellites</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="peaks-line1.html"><a href="peaks-line1.html"><i class="fa fa-check"></i><b>9</b> DUXC peaks and LINE1s</a><ul>
<li class="chapter" data-level="9.1" data-path="peaks-line1.html"><a href="peaks-line1.html#narrow-peaks-and-expressed-l1s"><i class="fa fa-check"></i><b>9.1</b> Narrow Peaks and expressed L1s</a><ul>
<li class="chapter" data-level="9.1.1" data-path="peaks-line1.html"><a href="peaks-line1.html#motif-discovery-using-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.1</b> Motif discovery using L1-overlapped peaks</a></li>
<li class="chapter" data-level="9.1.2" data-path="peaks-line1.html"><a href="peaks-line1.html#match-dux4-and-duxc-motif-to-l1-overapped-peaks"><i class="fa fa-check"></i><b>9.1.2</b> Match DUX4 and DUXC motif to L1-overapped peaks</a></li>
<li class="chapter" data-level="9.1.3" data-path="peaks-line1.html"><a href="peaks-line1.html#location-of-aatca-on-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.3</b> Location of AATCA on L1-overlapped peaks</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="peaks-line1.html"><a href="peaks-line1.html#line1s-and-aatca"><i class="fa fa-check"></i><b>9.2</b> LINE1s and AATCA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i># References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Conservation and ERVL activation of DUX family</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="non-LTR" class="section level1">
<h1><span class="header-section-number">Chapter 8</span> Non-LTR retrotrasposons and satellites</h1>
<p>This chapter investigated the transcriptioal activites of satellites and non-LTR retrotransposons – short interspersed nuclear elements (SINEs) and long interspersed nuclear elements (LINEs).</p>
<p><strong>Summary: non-LTR tretrotrasponses and satellites activiation in canine</strong><br />
Studies (Shaddle 2019; Whidden 2017) have shown that in human myoblast, DUX4 enriched HSATII and families of non-LTR RNA transcripts, such as Alu family of SINE and LINE1 (L1). Therefore, we were interested in examing the conservation of the non-LTR activation by DUX4 in our canine model. Our analysis showed that DUX4 up-regulated 53% of L1s (52/97) and 28.5% of DUX4 binding sites overlap with L1s. Given that L1s constitute 17% of the canine genome, DUX4, whether directly interacted with L1s or not, induced the L1 family’s enrichment in canine myoblasts. On the other hand, the enrichment of L1s by DUXC was rather unimpressive - only 16% of L1s (16/97) exhibited differential expression, and 15.6% DUXC peaks overlap with L1s (supplemental figure).</p>
<p>Unlike human SINEs derived from 7SL-RNA, canine SINEs share no homology with humans (Bentolila 1999). They evolved from tRNA genes (Bentolila 1999) and are categorized by RMSK into tRNA-Lys (72%, SINEC_#), MIR (27%), Deu (0.2%, AmnSINE1, AmnSINE2), and tRNA (0.2%, LFSINE_Vert, MamSINE1) families (supplemental figure). Six out of 11 tRNA-Lys elements (54%) are up-regulated in DUX4 expressed cell line; in fact, all tRNA-Lys have logFC &gt; 1. That is, the tRNA-Lys family is enriched in HinC transcriptome. Though none of the tRNA-Lys was called up-regulated in DUXC cells, their expression is highly correlated with that of in DUX4 expressed cell line (supplemental figure).</p>
<p>Canine (canFam3) has 5 major satellites and satellite-like family: Bs, Carnivore Satellite (CarSat1, CarSat2), Satellite canis Familiaris (SAT1_CF, SAT2_CF, SAT3_CF, SAT4_CF, and SAT6_CF), (CATTC)n, (GAATG)n and SUBTEL_sa. Bs, CarSat1, and SAT1_CF were differentially up-regulated in DUX4 expressed cells, whereas in DUXC expressed cells, Bs element was called up-regulated, and CarSat1 was moderately affected (adjusted p-value = 0.14).</p>
<p><strong>Load and datasets and libraries</strong></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># define directory</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">pkg_dir &lt;-<span class="st"> &quot;~/CompBio/canFam3.DuxFamily&quot;</span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3">fig_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;figures&quot;</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">data_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="co"># load library</span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(DESeq2))</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(tidyverse))</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(tidyr))</a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(plyranges))</a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(ggrepel))</a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(purrr))</a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(canFam3.rmsk))</a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="kw">load</span>(<span class="kw">file.path</span>(<span class="st">&quot;~/CompBio/hg38/hg38.rmsk.rda&quot;</span>))</a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(wesanderson)) <span class="co"># modern palettes</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15"></a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="co"># load datasets and local functions</span></a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="kw">source</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;scripts&quot;</span>, <span class="st">&quot;tools.R&quot;</span>))</a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;HinC_rmsk.rda&quot;</span>)) <span class="co"># made in chapter 6</span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;CinC_rmsk.rda&quot;</span>)) <span class="co"># made in chapter 6</span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;CALTinC_rmsk.rda&quot;</span>)) <span class="co"># made in chapter 6</span></a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;HinH_rmsk.rda&quot;</span>))</a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="kw">load</span>(<span class="kw">file.path</span>(data_dir, <span class="st">&quot;peaks_list.rda&quot;</span>))</a></code></pre></div>
<p><strong>Construct a common set</strong>
Combine the differential analysis results on repeats for HinC and CinC into one <em>data.frame</em> instance.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">comb &lt;-<span class="st"> </span><span class="kw">inner_join</span>(HinC_rmsk<span class="op">$</span>res_df, CinC_rmsk<span class="op">$</span>res_df, </a>
<a class="sourceLine" id="cb2-2" data-line-number="2">                   <span class="dt">by=</span><span class="st">&quot;repName&quot;</span>, <span class="dt">suffix=</span><span class="kw">c</span>(<span class="st">&quot;_HinC&quot;</span>, <span class="st">&quot;_CinC&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">DE_status =</span> <span class="st">&quot;neither&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">DE_status =</span> </a>
<a class="sourceLine" id="cb2-5" data-line-number="5">                  <span class="kw">replace</span>(DE_status, padj_HinC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">|</span><span class="st"> </span>padj_CinC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, <span class="st">&quot;either&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">DE_status =</span> <span class="kw">replace</span>(DE_status, padj_HinC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>padj_CinC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, <span class="st">&quot;both&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">DE_status =</span> <span class="kw">factor</span>(DE_status, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;neither&quot;</span>, <span class="st">&quot;either&quot;</span>, <span class="st">&quot;both&quot;</span>))) </a></code></pre></div>
<p>reference: Analysis of major repetitive DNA sequences in the dog (Canis familiaris) genome, Betolila 1999.</p>
<div id="lines" class="section level2">
<h2><span class="header-section-number">8.1</span> LINEs</h2>
<p><strong>background:</strong> LINEs constitute 20% of canine genome; LINE1 (L1) and LINE2 (L2) families derive 81% and 16% of the LINE repeat class, respectively. Study (Whidden 2017) showed that DUX4 up-regulated, directly or indirectly, impressive numbers of LINE1s in human myoblast. In this project, we expected LINE1 enrichment in DUX4 and DUXC canine cells.</p>
<p><strong>Results:</strong><br />
- DUX4 up-regulated &gt; 50% of L1s and L2s in canine cells<br />
- 35% DUX4 peaks binds to LINEs (monoclonal cell line)<br />
- GSEA suggested LINE1 family is enriched in DUX4 transcriptome in canine cells<br />
- DUX4 up-regulated many carnivore-specific L1s and some Half-L1s (HAL1s, ancient L1-like)<br />
- DUXC up-regulated some LINEs (16% in LINE1 family); GESA does not suggest L1 enrichment in DUXC canine cells<br />
- Both DUX4 and DUXC up-regulate L1M2a (L1) and X6B_LINE (CR1) with impressive fold-change (~ 6 and 4)<br />
- DUXC’s L1-overlapped peaks have a palindromic motif - AATCAATTGATT (243/691); DUX4’s have TAA(G/T/C)TCAATCA (431/999) (<a href="peaks-line1.html#peaks-line1">Chapter 9</a>)
- Peaks overlapping with L1s have enriched <strong>AATCA</strong> sequence (<a href="peaks-line1.html#peaks-line1">Chapter 9</a>)</p>
<pre><code>##   repFamily    length         prop     ypos percent          label
## 1        L1 412795245 81.219909588 40.60995     81%       L1 (81%)
## 2        L2  82659880 16.263821014 89.35182     16%       L2 (16%)
## 3       CR1   9758771  1.920095999 98.44378      2%      CR1 ( 2%)
## 4     RTE-X   2869545  0.564599976 99.68613      1%    RTE-X ( 1%)
## 5   Dong-R4     98128  0.019307265 99.97808      0%  Dong-R4 ( 0%)
## 6  RTE-BovB     56831  0.011181836 99.99332      0% RTE-BovB ( 0%)
## 7       L1?      5511  0.001084322 99.99946      0%      L1? ( 0%)</code></pre>
<p>DUX4 up-regulates &gt; 50% of L1 and L2 retroelements in canine cells, whereas DUXC only up-regulated 16% of L1s. LFC of HinC and CinC are highly correlated.</p>
<div class="figure"><span id="fig:LINE-up-regulation-freq"></span>
<img src="gitbook_files/figure-html/LINE-up-regulation-freq-1.png" alt="Percentage of up-regulation of LINE retroelements in each familiy. Others include unclassified L1, Dong-R4, RTE-BovB and RTR-X." width="672" />
<p class="caption">
Figure 8.1: Percentage of up-regulation of LINE retroelements in each familiy. Others include unclassified L1, Dong-R4, RTE-BovB and RTR-X.
</p>
</div>
<div class="figure"><span id="fig:LINE-transcripts"></span>
<img src="gitbook_files/figure-html/LINE-transcripts-1.png" alt="LINEs expression (LFC) in HinC and CinC." width="672" />
<p class="caption">
Figure 8.2: LINEs expression (LFC) in HinC and CinC.
</p>
</div>
<div id="l1s-activation" class="section level3">
<h3><span class="header-section-number">8.1.1</span> L1s activation</h3>
<p><strong>Question:</strong> Which subfamilies of L1s were up-regulated by DUX4 in canine cells?</p>
<p>DUX4 up-regulated many mammalian-specific, carnivore-specific L1s and some HAL1s.</p>
<pre><code>## Warning: Removed 10 rows containing missing values (position_stack).</code></pre>
<div class="figure"><span id="fig:plot-L1-elements"></span>
<img src="gitbook_files/figure-html/plot-L1-elements-1.png" alt="L1s activation in DUXC and DUX4 in canine cells." width="672" />
<p class="caption">
Figure 8.3: L1s activation in DUXC and DUX4 in canine cells.
</p>
</div>
</div>
<div id="line1-and-duxcdux4-binding-sites" class="section level3">
<h3><span class="header-section-number">8.1.2</span> LINE1 and DUXC/DUX4 binding sites</h3>
<p><em>This section is a summary of <a href="peaks-line1.html#peaks-line1">Chapter 9</a>).</em></p>
<p>Our previous study (Whidden 2017) and canine model suggested that L1 subfamily is enriched in DUX4 expressed human and canine myoblasts. But does DUX4 (or DUXC) directly interact L1s? Are the binding sites on L1s artifacts? To answer the question, we want to first identify the motifs of peaks that overlap with the differentially expressed LINE1 elements (LFC &gt; 1 with adjusted p-value &lt; 0.05) by:<br />
1. use the narrowed peaks (+/- 50 bps from summit) and select the subset that overlap with the differentially expression L1s<br />
2. get sequence of the selected peaks<br />
3. use <em>MEME</em> to find the DUXC/DUX4 motif of these peaks<br />
4. use <em>BioStrings::matchPattern()</em> to find the motif enrichment</p>
<p><strong>Results:</strong>
20.5% (21,888 out of 85,825) of DUX4 narrowed peaks overlap with L1s (25.50%); 14% of DUXC narrowed peaks overlap with L1s. Based on the top 1,000 of these L1-overlapped narrowed peaks of DUXC and DUX4 (on monoclonal cell lines), MEME identified a palindromic motif <strong>AATCAATTGATT</strong> (243/691) in DUXC cells and a motif <strong>TAAGTCAATCAG</strong> (431/999) in DUX4 cells. That is L1s do not have DUX4 nor DUXC motif enrichment; most likely that L1’s up-regulated are directly affected by DUX4 or DUXC in canine cells. Interestingly, however, both motifs of L1-overlapped peaks in DUX4 and DUXC cells obtain a motif <strong>AATCA</strong>, which is the canonical, second part of DUX4, Dux and DUXC motifs. This implies that L1s are enriched with <strong>AATCA</strong> sequences: 42.8% and 46.6% of DUXC and DUX4’s L1-overlapped narrowed peaks have <strong>AATCA</strong> sequences, mostly distributed near the center of the summits (technical details in <a href="peaks-line1.html#peaks-line1">Chapter 9</a>).</p>
<p>Another take away point is that DUXC and DUX4 seem not directly regulated L1s.</p>
</div>
</div>
<div id="sine" class="section level2">
<h2><span class="header-section-number">8.2</span> SINE</h2>
<p><strong>Background:</strong> The short interspersed repetitive elements (SINEs) constitute approximately 10% of canine genome. Canine SINEs evolved from tRNA genes whereas human SINEs evolved from 7SL-RNA, which gave rise to human Alu family (Bentolila 1999); they do not share homology. However, known that human Alu family are enriched by DUX4 transcriptome in human myoblast (Shaddle 2019), we are interested to know whether canine SINEs are also enriched by DUX4 and DUXC in canine myoblast and why?</p>
<p><strong>Results:</strong> RMSK categorized canine SINEs into tRNA-Lys (72%, <code>SINEC_*</code>), MIR (27%), Deu (0.2%, <code>AmnSINE1</code>, <code>AmnSINE2</code>), and tRNA (0.2%, <code>LFSINE_Vert</code>, <code>MamSINE1</code>) families (Figure <a href="non-LTR.html#fig:sine-distribution">8.4</a>) with total 20 name/group and 1,610,013 instances of sequence. Six out of 11 tRNA-Lys (54%, named <code>SINEC_*</code>) are up-regulated in DUX4 expressed cell line, and all tRNA-Lys have logFC &gt; 1. This means the tRNA-Lys family is enriched in HinC transcriptome. Though none of tRNA-Lys elements were called up-regulated by DUXC, their expression is highly correlated with that of in DUX4 expressed cell line.</p>
<p><strong>Note:</strong> (Bentolila 1999) SINE that frequently interrupts and possibly serves as a novel 3’ termination of a subclass of canine LINEs. Murine SINEs are derived from both 7SL-RNA (B1 family) and tRNA.</p>
<div id="distribution-of-sine-family" class="section level3">
<h3><span class="header-section-number">8.2.1</span> Distribution of SINE family</h3>
<div class="figure"><span id="fig:sine-distribution"></span>
<img src="gitbook_files/figure-html/sine-distribution-1.png" alt="Distribution of SINE family." width="672" />
<p class="caption">
Figure 8.4: Distribution of SINE family.
</p>
</div>
</div>
<div id="expression-of-sines-in-hinc-and-cinc-transcriptome" class="section level3">
<h3><span class="header-section-number">8.2.2</span> Expression of SINEs in HinC and CinC transcriptome</h3>
<div class="figure"><span id="fig:SINE-transcripts"></span>
<img src="gitbook_files/figure-html/SINE-transcripts-1.png" alt="SINEs expression in HinC and CinC." width="672" />
<p class="caption">
Figure 8.5: SINEs expression in HinC and CinC.
</p>
</div>
</div>
<div id="trna-lys" class="section level3">
<h3><span class="header-section-number">8.2.3</span> tRNA-Lys</h3>
<p>Table below listed number of up-regulated repeats (name/group) in each SINE family:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">SINE <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(<span class="kw">factor</span>(repFamily_HinC)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n =</span> dplyr<span class="op">::</span><span class="kw">n</span>(), </a>
<a class="sourceLine" id="cb5-3" data-line-number="3">                   <span class="dt">HinC_sig =</span> <span class="kw">sum</span>(padj_HinC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>), <span class="dt">CinC_sig=</span><span class="kw">sum</span>(padj_CinC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>))</a></code></pre></div>
<pre><code>## # A tibble: 4 x 4
##   `factor(repFamily_HinC)`     n HinC_sig CinC_sig
##   &lt;fct&gt;                    &lt;int&gt;    &lt;int&gt;    &lt;int&gt;
## 1 Deu                          2        0        0
## 2 MIR                          4        0        0
## 3 tRNA                         2        0        0
## 4 tRNA-Lys                    11        6        0</code></pre>
</div>
</div>
<div id="satellites" class="section level2">
<h2><span class="header-section-number">8.3</span> Satellites</h2>
<p><strong>Background:</strong> Satellites are consisted with short tandem repeats, and Canine satellite DNA may be more complicated and not restricted to the tandem repeats (Bentolila 1999). Based on RMAK, canine genome (canFam3) has 5 major satellites and satellite-like DNA - Bs, Carnivore Satellite (CarSat1, CarSat2), Satellite canis Familiaris (SAT1_CF, SAT2_CF, SAT3_CF, SAT4_CF and SAT6_CF), <code>(CATTC)n</code>, <code>(GAATG)n</code> and SUBTEL_sa. Bs, CarSat1, Carsat2, and SAT1_CF are the four satellites (repName) show some expression in DUX4 and DUXC expressed canine cell lines and therefore were analyzed. (CarSat1 and SAT1_CF are cetromeric satellite families.)</p>
<p><strong>Results:</strong> Bs, CarSat1, and SAT1_CF were up-regulated in HinC. Bs was up-regulated in CinC; CarSat1 was moderately affected (padj = 0.14).</p>
<p>How does SAT1_CF or CarSat compare to HSETII and GSETII?</p>
<pre><code>## [1] &quot;(CATTC)n&quot;  &quot;(GAATG)n&quot;  &quot;Bs&quot;        &quot;CarSat1&quot;   &quot;Carsat2&quot;   &quot;HSATII&quot;   
## [7] &quot;SAT1_CF&quot;   &quot;SUBTEL_sa&quot;</code></pre>
<pre><code>##   repName log2FoldChange_HinC log2FoldChange_CinC     padj_HinC    padj_CinC
## 1      Bs            6.906711           2.3038255 6.018346e-174 0.0002201754
## 2 CarSat1            7.476028           2.2907455  5.138595e-55 0.1413331726
## 3 Carsat2            1.315169           0.7970658  2.892419e-01 1.0000000000
## 4 SAT1_CF            5.001051           1.3001238 1.679505e-251 0.4698271396</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="LTR-activation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="peaks-line1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["gitbook.pdf", "gitbook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
