<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 DUXC peaks and LINE1s | Conservation and ERVL activation of Canine DUXC</title>
  <meta name="description" content="Bioinformatics and statistics analysis" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 DUXC peaks and LINE1s | Conservation and ERVL activation of Canine DUXC" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Bioinformatics and statistics analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 DUXC peaks and LINE1s | Conservation and ERVL activation of Canine DUXC" />
  
  <meta name="twitter:description" content="Bioinformatics and statistics analysis" />
  

<meta name="author" content="Chao-Jen Wong" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="non-LTR.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DUXC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#datasets"><i class="fa fa-check"></i><b>0.1</b> Datasets</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#tips"><i class="fa fa-check"></i><b>0.2</b> Tips</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="chipseq.html"><a href="chipseq.html"><i class="fa fa-check"></i><b>1</b> ChIP-seq for the canine model</a><ul>
<li class="chapter" data-level="1.1" data-path="chipseq.html"><a href="chipseq.html#chip-seq-samples"><i class="fa fa-check"></i><b>1.1</b> ChIP-seq samples</a></li>
<li class="chapter" data-level="1.2" data-path="chipseq.html"><a href="chipseq.html#workflow"><i class="fa fa-check"></i><b>1.2</b> workflow</a></li>
<li class="chapter" data-level="1.3" data-path="chipseq.html"><a href="chipseq.html#peaks-and-functional-distribution"><i class="fa fa-check"></i><b>1.3</b> Peaks and functional distribution</a><ul>
<li class="chapter" data-level="1.3.1" data-path="chipseq.html"><a href="chipseq.html#functional-distribution"><i class="fa fa-check"></i><b>1.3.1</b> Functional distribution</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chipseq.html"><a href="chipseq.html#motifs-analysis-need-update"><i class="fa fa-check"></i><b>1.4</b> Motifs analysis NEED UPDATE</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="makeSE.html"><a href="makeSE.html"><i class="fa fa-check"></i><b>2</b> RNA-seq to DESeq2</a><ul>
<li class="chapter" data-level="2.1" data-path="makeSE.html"><a href="makeSE.html#rna-seq-preprocessing"><i class="fa fa-check"></i><b>2.1</b> RNA-seq preprocessing</a></li>
<li class="chapter" data-level="2.2" data-path="makeSE.html"><a href="makeSE.html#reads-counts"><i class="fa fa-check"></i><b>2.2</b> Reads counts</a><ul>
<li class="chapter" data-level="2.2.1" data-path="makeSE.html"><a href="makeSE.html#define-gene-features"><i class="fa fa-check"></i><b>2.2.1</b> Define gene features</a></li>
<li class="chapter" data-level="2.2.2" data-path="makeSE.html"><a href="makeSE.html#read-counts"><i class="fa fa-check"></i><b>2.2.2</b> Read counts</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="makeSE.html"><a href="makeSE.html#deseq2"><i class="fa fa-check"></i><b>2.3</b> DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html"><i class="fa fa-check"></i><b>3</b> DUXC transcriptome</a><ul>
<li class="chapter" data-level="3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#load-datasets-and-settings"><i class="fa fa-check"></i><b>3.1</b> Load datasets and settings</a></li>
<li class="chapter" data-level="3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cinc-duxc-expresison-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2</b> CinC: DUXC expresison in canine muscle cells</a></li>
<li class="chapter" data-level="3.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hinc-dux4-expression-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.3</b> HinC: DUX4 expression in canine muscle cells</a></li>
<li class="chapter" data-level="3.4" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#duxc-alt-expression---caltinc"><i class="fa fa-check"></i><b>3.4</b> DUXC-ALT expression - CALTinC</a><ul>
<li class="chapter" data-level="3.4.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#go-analysis-for-the-down-regulated-gene-set"><i class="fa fa-check"></i><b>3.4.1</b> GO analysis for the down-regulated gene set</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cross-species-models-dux4-expression-in-canine-and-murine"><i class="fa fa-check"></i><b>3.5</b> Cross-species models: DUX4 expression in canine and murine</a><ul>
<li class="chapter" data-level="3.5.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-duxc-expression-in-canine"><i class="fa fa-check"></i><b>3.5.1</b> DUX4 and DUXC expression in canine</a></li>
<li class="chapter" data-level="3.5.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-dux-expression-in-murine"><i class="fa fa-check"></i><b>3.5.2</b> DUX4 and Dux expression in murine</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html"><i class="fa fa-check"></i><b>4</b> DUXC and Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#duxc-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.1</b> DUXC transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs"><i class="fa fa-check"></i><b>4.1.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.1.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome"><i class="fa fa-check"></i><b>4.1.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#dux-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.2</b> Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.2.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs-1"><i class="fa fa-check"></i><b>4.2.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.2.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome-1"><i class="fa fa-check"></i><b>4.2.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html"><i class="fa fa-check"></i><b>5</b> DUXC transcriptome compared to early embryo transcriptome profile</a><ul>
<li class="chapter" data-level="5.1" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#mouse-2c-like-signature"><i class="fa fa-check"></i><b>5.1</b> Mouse 2C-like signature</a></li>
<li class="chapter" data-level="5.2" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#cleavage-stage-signature"><i class="fa fa-check"></i><b>5.2</b> cleavage-stage signature</a></li>
<li class="chapter" data-level="5.3" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#ipscs"><i class="fa fa-check"></i><b>5.3</b> iPSCs</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="repeats.html"><a href="repeats.html"><i class="fa fa-check"></i><b>6</b> Repeat elements</a><ul>
<li class="chapter" data-level="6.1" data-path="repeats.html"><a href="repeats.html#takeaway"><i class="fa fa-check"></i><b>6.1</b> Takeaway</a></li>
<li class="chapter" data-level="6.2" data-path="repeats.html"><a href="repeats.html#repeat-element-expression"><i class="fa fa-check"></i><b>6.2</b> Repeat element expression</a></li>
<li class="chapter" data-level="6.3" data-path="repeats.html"><a href="repeats.html#differential-analysis"><i class="fa fa-check"></i><b>6.3</b> Differential analysis</a><ul>
<li class="chapter" data-level="6.3.1" data-path="repeats.html"><a href="repeats.html#normalization"><i class="fa fa-check"></i><b>6.3.1</b> Normalization</a></li>
<li class="chapter" data-level="6.3.2" data-path="repeats.html"><a href="repeats.html#deseq2-results"><i class="fa fa-check"></i><b>6.3.2</b> DESeq2 results</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="repeats.html"><a href="repeats.html#repeat-family-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> Repeat family enrichment analysis</a><ul>
<li class="chapter" data-level="6.4.1" data-path="repeats.html"><a href="repeats.html#correlation-between-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>6.4.1</b> Correlation between HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="6.4.2" data-path="repeats.html"><a href="repeats.html#repeat-family-activation"><i class="fa fa-check"></i><b>6.4.2</b> Repeat family activation</a></li>
<li class="chapter" data-level="6.4.3" data-path="repeats.html"><a href="repeats.html#repeat-classs-activation"><i class="fa fa-check"></i><b>6.4.3</b> Repeat classs activation</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="repeats.html"><a href="repeats.html#ltr-retroelements-distribution-and-activation"><i class="fa fa-check"></i><b>6.5</b> LTR retroelements distribution and activation</a></li>
<li class="chapter" data-level="6.6" data-path="repeats.html"><a href="repeats.html#peaks-distribution-in-class-of-repeat-elements"><i class="fa fa-check"></i><b>6.6</b> Peaks distribution in class of repeat elements</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="LTR-activation.html"><a href="LTR-activation.html"><i class="fa fa-check"></i><b>7</b> LTR activation</a><ul>
<li class="chapter" data-level="7.1" data-path="LTR-activation.html"><a href="LTR-activation.html#takaway"><i class="fa fa-check"></i><b>7.1</b> Takaway</a></li>
<li class="chapter" data-level="7.2" data-path="LTR-activation.html"><a href="LTR-activation.html#background-in-murine-and-human"><i class="fa fa-check"></i><b>7.2</b> background in murine and human</a></li>
<li class="chapter" data-level="7.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-subfamily-activation-in-canine"><i class="fa fa-check"></i><b>7.3</b> ERVL subfamily activation in canine</a></li>
<li class="chapter" data-level="7.4" data-path="LTR-activation.html"><a href="LTR-activation.html#malr-activation-in-canine"><i class="fa fa-check"></i><b>7.4</b> MaLR activation in canine</a></li>
<li class="chapter" data-level="7.5" data-path="LTR-activation.html"><a href="LTR-activation.html#compare-wih-hinh"><i class="fa fa-check"></i><b>7.5</b> Compare wih HinH</a><ul>
<li class="chapter" data-level="7.5.1" data-path="LTR-activation.html"><a href="LTR-activation.html#copy-number"><i class="fa fa-check"></i><b>7.5.1</b> Copy number</a></li>
<li class="chapter" data-level="7.5.2" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-malr"><i class="fa fa-check"></i><b>7.5.2</b> ERVL-MaLR</a></li>
<li class="chapter" data-level="7.5.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl"><i class="fa fa-check"></i><b>7.5.3</b> ERVL</a></li>
<li class="chapter" data-level="7.5.4" data-path="LTR-activation.html"><a href="LTR-activation.html#erv1"><i class="fa fa-check"></i><b>7.5.4</b> ERV1</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="LTR-activation.html"><a href="LTR-activation.html#duxc-alt-and-ervl-transcriptions"><i class="fa fa-check"></i><b>7.6</b> DUXC-ALT and ERVL transcriptions</a><ul>
<li class="chapter" data-level="7.6.1" data-path="LTR-activation.html"><a href="LTR-activation.html#erv-subfamily-expression"><i class="fa fa-check"></i><b>7.6.1</b> ERV subfamily expression</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="LTR-activation.html"><a href="LTR-activation.html#LTR-activation"><i class="fa fa-check"></i><b>8</b> LTR activation</a><ul>
<li class="chapter" data-level="8.1" data-path="LTR-activation.html"><a href="LTR-activation.html#copy-number-1"><i class="fa fa-check"></i><b>8.1</b> Copy number</a></li>
<li class="chapter" data-level="8.2" data-path="LTR-activation.html"><a href="LTR-activation.html#sattillites-activiation"><i class="fa fa-check"></i><b>8.2</b> Sattillites activiation</a></li>
<li class="chapter" data-level="8.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ltrs-activation"><i class="fa fa-check"></i><b>8.3</b> LTRs activation</a><ul>
<li class="chapter" data-level="8.3.1" data-path="LTR-activation.html"><a href="LTR-activation.html#erv1-1"><i class="fa fa-check"></i><b>8.3.1</b> ERV1</a></li>
<li class="chapter" data-level="8.3.2" data-path="LTR-activation.html"><a href="LTR-activation.html#direct-binding-of-dux4-and-duxc"><i class="fa fa-check"></i><b>8.3.2</b> Direct binding of DUX4 and DUXC</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="LTR-activation.html"><a href="LTR-activation.html#line1-activation"><i class="fa fa-check"></i><b>8.4</b> LINE1 activation</a></li>
<li class="chapter" data-level="8.5" data-path="LTR-activation.html"><a href="LTR-activation.html#satellites"><i class="fa fa-check"></i><b>8.5</b> Satellites</a></li>
<li class="chapter" data-level="8.6" data-path="LTR-activation.html"><a href="LTR-activation.html#sines"><i class="fa fa-check"></i><b>8.6</b> SINEs</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="non-LTR.html"><a href="non-LTR.html"><i class="fa fa-check"></i><b>9</b> Non-LTR retrotrasposons and satellites</a><ul>
<li class="chapter" data-level="9.1" data-path="non-LTR.html"><a href="non-LTR.html#lines"><i class="fa fa-check"></i><b>9.1</b> LINEs</a><ul>
<li class="chapter" data-level="9.1.1" data-path="non-LTR.html"><a href="non-LTR.html#line1s-activation"><i class="fa fa-check"></i><b>9.1.1</b> LINE1s activation</a></li>
<li class="chapter" data-level="9.1.2" data-path="non-LTR.html"><a href="non-LTR.html#line1-and-duxcdux4-binding-sites"><i class="fa fa-check"></i><b>9.1.2</b> LINE1 and DUXC/DUX4 binding sites</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="non-LTR.html"><a href="non-LTR.html#sine"><i class="fa fa-check"></i><b>9.2</b> SINE</a><ul>
<li class="chapter" data-level="9.2.1" data-path="non-LTR.html"><a href="non-LTR.html#distribution-of-sine-family"><i class="fa fa-check"></i><b>9.2.1</b> Distribution of SINE family</a></li>
<li class="chapter" data-level="9.2.2" data-path="non-LTR.html"><a href="non-LTR.html#expression-of-sines-in-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>9.2.2</b> Expression of SINEs in HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="9.2.3" data-path="non-LTR.html"><a href="non-LTR.html#trna-lys"><i class="fa fa-check"></i><b>9.2.3</b> tRNA-Lys</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="non-LTR.html"><a href="non-LTR.html#satellites-1"><i class="fa fa-check"></i><b>9.3</b> Satellites</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="peaks-line1.html"><a href="peaks-line1.html"><i class="fa fa-check"></i><b>10</b> DUXC peaks and LINE1s</a><ul>
<li class="chapter" data-level="10.1" data-path="peaks-line1.html"><a href="peaks-line1.html#narrow-peaks-and-expressed-l1s"><i class="fa fa-check"></i><b>10.1</b> Narrow Peaks and expressed L1s</a><ul>
<li class="chapter" data-level="10.1.1" data-path="peaks-line1.html"><a href="peaks-line1.html#motif-discovery-using-l1-overlapped-peaks"><i class="fa fa-check"></i><b>10.1.1</b> Motif discovery using L1-overlapped peaks</a></li>
<li class="chapter" data-level="10.1.2" data-path="peaks-line1.html"><a href="peaks-line1.html#match-dux4-and-duxc-motif-to-l1-overapped-peaks"><i class="fa fa-check"></i><b>10.1.2</b> Match DUX4 and DUXC motif to L1-overapped peaks</a></li>
<li class="chapter" data-level="10.1.3" data-path="peaks-line1.html"><a href="peaks-line1.html#location-of-aatca-on-l1-overlapped-peaks"><i class="fa fa-check"></i><b>10.1.3</b> Location of AATCA on L1-overlapped peaks</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="peaks-line1.html"><a href="peaks-line1.html#line1s-and-aatca"><i class="fa fa-check"></i><b>10.2</b> LINE1s and AATCA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i># References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Conservation and ERVL activation of Canine DUXC</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="peaks-line1" class="section level1">
<h1><span class="header-section-number">Chapter 10</span> DUXC peaks and LINE1s</h1>
<p>To answer whether DUXC and DUX4 directly regulate to L1s, we want to check if DUXC an DUX4’s motifs are enriched in the peaks that overlap with L1s. To do so, we did the following steps:<br />
1. use the narrowed peaks (+/- 50 bps from summit) and select the subset that overlap with the differentially expressed L1s (LFC &gt; 1; adjusted p-value &lt; 0.05; compared with control lucifereas samples)<br />
2. get sequence of the selected peaks<br />
3. use <em>MEME</em> to find the DUXC/DUX4 motif of these peaks<br />
4. use <em>BioStrings::matchPattern()</em> to find the motif enrichment</p>
<p><strong>Results:</strong> DUXC’s L1-overlapped peaks has a palindromic motif - AATCAATTGATT (243/691); DUX4’s has motif TAA(G/T/C)TCAATCA (431/999). DUXC and DUX4 motifs are not enriched in peaks overlapping with up-regulated L1s. However, 41% and 47% of DUXC and DUX4 L1-overlaped peaks contain the conanical mofit AATCA; for those containing AATCA, ~ 70% have AATCA located within 25 bps from the summit.</p>
<p><strong>Load packages and datasets</strong></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(plyranges))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(ShortRead))</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(Biostrings))</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(ggplot2))</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(BSgenome.Cfamiliaris.UCSC.canFam3))</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="co"># load datasets</span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co">#pkg_dir &lt;- &quot;~/tapscott/RNA-Seq/canFam3.DuxFamily&quot;</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8">pkg_dir &lt;-<span class="st"> &quot;~/CompBio/canFam3.DuxFamily&quot;</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9">fig_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;figures&quot;</span>)</a>
<a class="sourceLine" id="cb1-10" data-line-number="10">data_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>)</a>
<a class="sourceLine" id="cb1-11" data-line-number="11">meme_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;meme_narrow_peaks_ol_L1s&quot;</span>)</a>
<a class="sourceLine" id="cb1-12" data-line-number="12"></a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="kw">load</span>(<span class="kw">file.path</span>(data_dir, <span class="st">&quot;peaks_list.rda&quot;</span>))</a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="kw">load</span>(<span class="kw">file.path</span>(data_dir, <span class="st">&quot;CinC_rmsk.rda&quot;</span>))</a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="kw">load</span>(<span class="kw">file.path</span>(data_dir, <span class="st">&quot;HinC_rmsk.rda&quot;</span>))</a></code></pre></div>
<div id="narrow-peaks-and-expressed-l1s" class="section level2">
<h2><span class="header-section-number">10.1</span> Narrow Peaks and expressed L1s</h2>
<ol style="list-style-type: decimal">
<li>Narrow the peaks to 101 bps (+/-50 bps from the summits) of CCH and hDUX4.</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># re-define peaks (+/- 50 summit) = 101 bps</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">.narrow_peaks_100bps &lt;-<span class="st"> </span><span class="cf">function</span>(gr_peaks) {</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">  gr &lt;-<span class="st"> </span><span class="kw">granges</span>(gr_peaks)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">  <span class="kw">mcols</span>(gr) &lt;-<span class="st"> </span><span class="kw">mcols</span>(gr_peaks)[, <span class="kw">c</span>(<span class="st">&quot;abs_summit&quot;</span>, <span class="st">&quot;fold_enrichment&quot;</span>, <span class="st">&quot;X.log10.qvalue.&quot;</span>, </a>
<a class="sourceLine" id="cb2-5" data-line-number="5">                                   <span class="st">&quot;name&quot;</span>, <span class="st">&quot;annotation&quot;</span>, <span class="st">&quot;geneId&quot;</span>)]</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">  <span class="kw">start</span>(gr) &lt;-<span class="st"> </span>gr<span class="op">$</span>abs_summit <span class="op">-</span><span class="st"> </span><span class="dv">50</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">  <span class="kw">end</span>(gr) &lt;-<span class="st"> </span>gr<span class="op">$</span>abs_summit <span class="op">+</span><span class="st"> </span><span class="dv">50</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8">  gr</a>
<a class="sourceLine" id="cb2-9" data-line-number="9">}</a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co"># narrowing peaks to +/- 50 bps from the summit</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11">peaks_DUXC_mono &lt;-<span class="st"> </span><span class="kw">.narrow_peaks_100bps</span>(peaks_list<span class="op">$</span>Sample7_ImmCnMb_CCH_mono) <span class="co"># 6060</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12">peaks_hDUX4_mono &lt;-<span class="st"> </span><span class="kw">.narrow_peaks_100bps</span>(peaks_list<span class="op">$</span>Sample6_ImmCnMb_hDUX4_mono) <span class="co">#85825</span></a></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Load the read counts, <code>DESeqDataseq</code> instance of canFam3 RMSK:</li>
</ol>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># load read counts for the canFam3 Repeat Masker annotation</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(canFam3.rmsk))</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">L1_rmsk &lt;-<span class="st"> </span>canFam3.rmsk <span class="op">%&gt;%</span><span class="st"> </span>plyranges<span class="op">::</span><span class="kw">filter</span>(repFamily <span class="op">==</span><span class="st"> &quot;L1&quot;</span>)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">L1_rmsk<span class="op">$</span>dummy_repNamename &lt;-<span class="st"> </span><span class="kw">paste0</span>(L1_rmsk<span class="op">$</span>repName, <span class="st">&quot;_&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(L1_rmsk))</a></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Find peaks overlapping with up-regulated L1s</li>
</ol>
<ul>
<li>DUXC: 691 out of 6031 narrowed peaks are overlapping with up-regulated L1s (11.46%)<br />
</li>
<li>DUX4: 17908 out of 85825 narrowed peaks are overlapping with up-regulated L1s (20.87%)</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># select narrowed peaks (intersect) overlapping with L1s </span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2">.get_overlap_exp_L1 &lt;-<span class="st"> </span><span class="cf">function</span>(peaks_gr, L1_rmsk, de_LINE1) {</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">  <span class="co"># </span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4">  ol &lt;-<span class="st"> </span><span class="kw">join_overlap_inner</span>(peaks_gr, L1_rmsk, <span class="dt">minoverlap=</span>50L) </a>
<a class="sourceLine" id="cb4-5" data-line-number="5">  <span class="kw">message</span>(<span class="kw">sprintf</span>(<span class="st">&quot;%s out of %s narrowed peaks are overlapping with L1s (%2.2f%%).&quot;</span>, </a>
<a class="sourceLine" id="cb4-6" data-line-number="6">                  <span class="kw">length</span>(<span class="kw">unique</span>(ol<span class="op">$</span>name)), <span class="kw">length</span>(peaks_gr), </a>
<a class="sourceLine" id="cb4-7" data-line-number="7">                  <span class="kw">length</span>(<span class="kw">unique</span>(ol<span class="op">$</span>name))<span class="op">/</span><span class="kw">length</span>(peaks_gr) <span class="op">*</span><span class="st"> </span><span class="dv">100</span>))</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">  <span class="co"># overlap with up-regulated L1?</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9">  ol_de_LINE1 &lt;-<span class="st"> </span>ol <span class="op">%&gt;%</span><span class="st"> </span>plyranges<span class="op">::</span><span class="kw">filter</span>(repName <span class="op">%in%</span><span class="st"> </span><span class="kw">as.character</span>(de_LINE1<span class="op">$</span>repName))</a>
<a class="sourceLine" id="cb4-10" data-line-number="10">  <span class="kw">message</span>(<span class="kw">sprintf</span>(<span class="st">&quot;%s out of %s narrowed peaks are overlapping with up-regulated L1s (%2.2f%%).&quot;</span>, </a>
<a class="sourceLine" id="cb4-11" data-line-number="11">                  <span class="kw">length</span>(<span class="kw">unique</span>(ol_de_LINE1<span class="op">$</span>name)), <span class="kw">length</span>(peaks_gr), </a>
<a class="sourceLine" id="cb4-12" data-line-number="12">                  <span class="kw">length</span>(<span class="kw">unique</span>(ol_de_LINE1<span class="op">$</span>name))<span class="op">/</span><span class="kw">length</span>(peaks_gr) <span class="op">*</span><span class="st"> </span><span class="dv">100</span>))</a>
<a class="sourceLine" id="cb4-13" data-line-number="13">  <span class="kw">return</span>(<span class="kw">list</span>(<span class="dt">ol=</span>ol, <span class="dt">ol_de_LINE1=</span>ol_de_LINE1))</a>
<a class="sourceLine" id="cb4-14" data-line-number="14">}</a>
<a class="sourceLine" id="cb4-15" data-line-number="15"></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co"># DUXC</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17">de_LINE1 &lt;-<span class="st"> </span>CinC_rmsk<span class="op">$</span>res_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, repFamily <span class="op">==</span><span class="st"> &quot;L1&quot;</span>) <span class="co">#16</span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="kw">message</span>(<span class="st">&quot;DUXC monoclonal sample:&quot;</span>)</a></code></pre></div>
<pre><code>## DUXC monoclonal sample:</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">DUXC_mono &lt;-<span class="st"> </span><span class="kw">.get_overlap_exp_L1</span>(peaks_DUXC_mono, L1_rmsk, de_LINE1)</a></code></pre></div>
<pre><code>## 875 out of 6031 narrowed peaks are overlapping with L1s (14.51%).</code></pre>
<pre><code>## 691 out of 6031 narrowed peaks are overlapping with up-regulated L1s (11.46%).</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co"># DUX4</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">de_LINE1 &lt;-<span class="st"> </span>HinC_rmsk<span class="op">$</span>res_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, repFamily <span class="op">==</span><span class="st"> &quot;L1&quot;</span>) </a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="kw">message</span>(<span class="st">&quot;DUX4 monocloal sample:&quot;</span>)</a></code></pre></div>
<pre><code>## DUX4 monocloal sample:</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">DUX4_mono &lt;-<span class="st"> </span><span class="kw">.get_overlap_exp_L1</span>(peaks_hDUX4_mono, L1_rmsk, de_LINE1)</a></code></pre></div>
<pre><code>## 21888 out of 85825 narrowed peaks are overlapping with L1s (25.50%).</code></pre>
<pre><code>## 17908 out of 85825 narrowed peaks are overlapping with up-regulated L1s (20.87%).</code></pre>
<div id="motif-discovery-using-l1-overlapped-peaks" class="section level3">
<h3><span class="header-section-number">10.1.1</span> Motif discovery using L1-overlapped peaks</h3>
<p>The code below makes FASTA files for the narrowed peaks that overlap with up-regulated L1s and run MEME to exam the enrichment of DUXC/DUX4 motifs. DUXC’s L1-overlapped peaks has a palindromic motif - <strong>AATCAATTGATT</strong> (243/691); DUX4’s L1-overlaped peaks has motif <strong>TAA(G/T/C)TCAATCA</strong> (431/999). Both motifs contain the canonical 2nd part of DUX4/DUXC motif - <strong>AATCA</strong>. Does that mean both DUXC and DUX4 were attracted to bind to this half motif and create artifacts of peaks?</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="co"># must move to rhino so that I can use MEME (4.11.1-foo-2015b) module</span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2">.peaks_seq_to_MEME &lt;-<span class="st"> </span><span class="cf">function</span>(peaks, BS, <span class="dt">write_FASTA=</span>TURE, meme_dir, prefix, <span class="dt">max.peaks=</span>999L) {</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">    <span class="co">#require(BSgeome)</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4">    <span class="co"># max.peaks must be integer</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5">    <span class="cf">if</span> (<span class="kw">length</span>(peaks) <span class="op">&gt;</span><span class="st"> </span>max.peaks) {</a>
<a class="sourceLine" id="cb14-6" data-line-number="6">      <span class="cf">if</span> (<span class="op">!</span><span class="kw">is.integer</span>(max.peaks)) max.peaks &lt;-<span class="st"> </span><span class="kw">as.integer</span>(max.peaks)</a>
<a class="sourceLine" id="cb14-7" data-line-number="7">      <span class="co">#idx &lt;- sample(1:length(peaks), size=max.peaks)</span></a>
<a class="sourceLine" id="cb14-8" data-line-number="8">      <span class="co">#peaks &lt;- peaks %&gt;% plyranges::arrange(plyranges::desc(X.log10.qvalue.))</span></a>
<a class="sourceLine" id="cb14-9" data-line-number="9">      or &lt;-<span class="st"> </span><span class="kw">order</span>(peaks<span class="op">$</span>X.log10.qvalue., <span class="dt">decreasing=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb14-10" data-line-number="10">      peaks &lt;-<span class="st"> </span>peaks[or[<span class="dv">1</span><span class="op">:</span>max.peaks]]</a>
<a class="sourceLine" id="cb14-11" data-line-number="11">    }      </a>
<a class="sourceLine" id="cb14-12" data-line-number="12">    </a>
<a class="sourceLine" id="cb14-13" data-line-number="13">    peaks_seq &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(BS, peaks)</a>
<a class="sourceLine" id="cb14-14" data-line-number="14">    <span class="kw">names</span>(peaks_seq) &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(peaks_seq)</a>
<a class="sourceLine" id="cb14-15" data-line-number="15">    tmp_file &lt;-<span class="st"> </span><span class="kw">file.path</span>(meme_dir, <span class="kw">paste0</span>(prefix, <span class="st">&quot;_narrow_peakseq.fa&quot;</span>))</a>
<a class="sourceLine" id="cb14-16" data-line-number="16">    </a>
<a class="sourceLine" id="cb14-17" data-line-number="17">    ShortRead<span class="op">::</span><span class="kw">writeFasta</span>(peaks_seq, <span class="dt">file=</span>tmp_file)</a>
<a class="sourceLine" id="cb14-18" data-line-number="18">    cmt &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;meme %s -oc %s -nmotifs 3 -maxw 12 -V -mod zoops -maxsize 110000 -p 4 -V&quot;</span>,</a>
<a class="sourceLine" id="cb14-19" data-line-number="19">                   tmp_file, <span class="kw">file.path</span>(meme_dir, prefix))</a>
<a class="sourceLine" id="cb14-20" data-line-number="20">    <span class="kw">system</span>(cmt)</a>
<a class="sourceLine" id="cb14-21" data-line-number="21">    <span class="co"># search the meme result and return the enriched motif</span></a>
<a class="sourceLine" id="cb14-22" data-line-number="22">    <span class="kw">return</span>(peaks_seq)</a>
<a class="sourceLine" id="cb14-23" data-line-number="23">}</a>
<a class="sourceLine" id="cb14-24" data-line-number="24"></a>
<a class="sourceLine" id="cb14-25" data-line-number="25"><span class="co"># system(&quot;ml MEME/5.1.1-foss-2019b-Perl-5.30.0-Python-3.7.4&quot;)</span></a>
<a class="sourceLine" id="cb14-26" data-line-number="26"><span class="co">#&#39; make fasta files for the narrowed peaks that overlaps with LINE1 elements and run MEME</span></a>
<a class="sourceLine" id="cb14-27" data-line-number="27"><span class="kw">library</span>(BSgenome.Cfamiliaris.UCSC.canFam3)</a>
<a class="sourceLine" id="cb14-28" data-line-number="28">meme_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;meme_narrow_peaks_ol_L1s&quot;</span>)</a>
<a class="sourceLine" id="cb14-29" data-line-number="29">DUXC_mono_peakseq &lt;-<span class="st"> </span><span class="kw">.peaks_seq_to_MEME</span>(DUXC_mono<span class="op">$</span>ol_de_LINE1,</a>
<a class="sourceLine" id="cb14-30" data-line-number="30">                                        <span class="dt">BS=</span>BSgenome.Cfamiliaris.UCSC.canFam3, </a>
<a class="sourceLine" id="cb14-31" data-line-number="31">                                        <span class="dt">meme_dir =</span> meme_dir, <span class="dt">prefix=</span><span class="st">&quot;DUXC_mono&quot;</span>, <span class="dt">max.peaks=</span>999L)</a>
<a class="sourceLine" id="cb14-32" data-line-number="32">DUX4_mono_peakseq &lt;-<span class="st"> </span><span class="kw">.peaks_seq_to_MEME</span>(DUX4_mono<span class="op">$</span>ol_de_LINE1,</a>
<a class="sourceLine" id="cb14-33" data-line-number="33">                                        <span class="dt">BS=</span>BSgenome.Cfamiliaris.UCSC.canFam3, </a>
<a class="sourceLine" id="cb14-34" data-line-number="34">                                        <span class="dt">meme_dir =</span> meme_dir, <span class="dt">prefix=</span><span class="st">&quot;DUX4_mono&quot;</span>, <span class="dt">max.peaks=</span>999L)</a></code></pre></div>
<p>Top DUXC L1-overlapped peaks motif is AATCA AT TGATT (243/691); Top DUX4 L1-overlapped peaks is TAA G/T/C T C AATCA (431/999). Both shared the conanical DUX4/DUXC/mDux motif, <strong>AATCA</strong>, but none of them matched the whole DUXC nor DUX4 motifs.</p>
<p>DUXC L1-overlapped peaks motif:
<img src="../meme_narrow_peaks_ol_L1s/DUXC_mono/logo1.png" title="DUXC L1-overlapped peaks motif" alt="alt text" />
<img src="../meme_narrow_peaks_ol_L1s/DUX4_mono/logo1.png" title="DUXC L1-overlapped peaks motif" alt="alt text" /></p>
</div>
<div id="match-dux4-and-duxc-motif-to-l1-overapped-peaks" class="section level3">
<h3><span class="header-section-number">10.1.2</span> Match DUX4 and DUXC motif to L1-overapped peaks</h3>
<p>Here I break DUX4 and DUXC motifs into two seperate motifs. The first part of the motif is the diverged one and second the canonical: AATCA. Then I use <code>BioStrings::matchPattern()</code> to check if the L1-overlapped peaks contains those specific short form and continues forms of motif. About 42% and 47% of these peaks contains the canonical motif - <code>AATCA</code>.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="co"># use BioStrings:matchPattern to find the motif</span></a>
<a class="sourceLine" id="cb15-2" data-line-number="2">hDUX4_motif &lt;-<span class="st"> &quot;TAACCTAATCA&quot;</span></a>
<a class="sourceLine" id="cb15-3" data-line-number="3">mDux_motif  &lt;-<span class="st"> &quot;TGATTCAATCA&quot;</span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4">DUXC_motif  &lt;-<span class="st"> &quot;TAATCCAATCA&quot;</span></a>
<a class="sourceLine" id="cb15-5" data-line-number="5"></a>
<a class="sourceLine" id="cb15-6" data-line-number="6">motif2 &lt;-<span class="st"> &quot;AATCA&quot;</span> <span class="co"># second part of cononical motif (DUX4, Dux and DUXC) / same as motif5</span></a>
<a class="sourceLine" id="cb15-7" data-line-number="7"><span class="co"># DUXC mono</span></a>
<a class="sourceLine" id="cb15-8" data-line-number="8">.match_motif_pattern &lt;-<span class="st"> </span><span class="cf">function</span>(peaks_gr, prefix) {</a>
<a class="sourceLine" id="cb15-9" data-line-number="9">  peaks_seq &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(<span class="dt">x=</span>BSgenome.Cfamiliaris.UCSC.canFam3, <span class="dt">names=</span>peaks_gr)</a>
<a class="sourceLine" id="cb15-10" data-line-number="10">  count_motif &lt;-<span class="st"> </span><span class="kw">vcountPattern</span>(<span class="dt">pattern=</span>motif2, <span class="dt">subject=</span>peaks_seq) </a>
<a class="sourceLine" id="cb15-11" data-line-number="11">  <span class="kw">message</span>(<span class="kw">sprintf</span>(<span class="st">&quot;%s: %2.2f %% have motif %s (canonical 2nd motif)&quot;</span>, prefix,</a>
<a class="sourceLine" id="cb15-12" data-line-number="12">                  <span class="dv">100</span><span class="op">*</span>(<span class="kw">sum</span>(count_motif <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">/</span><span class="st"> </span><span class="kw">length</span>(peaks_seq)), motif2))</a>
<a class="sourceLine" id="cb15-13" data-line-number="13"></a>
<a class="sourceLine" id="cb15-14" data-line-number="14">  <span class="co">#count_motif &lt;- vcountPattern(pattern=DUXC_motif, subject=peaks_seq, max.mismatch=1) </span></a>
<a class="sourceLine" id="cb15-15" data-line-number="15">  <span class="co">#message(sprintf(&quot;%s: %2.2f %% have motif %s (DUXC, 1 mismatch)&quot;, </span></a>
<a class="sourceLine" id="cb15-16" data-line-number="16">  <span class="co">#                prefix, 100*(sum(count_motif &gt; 0) / length(peaks_seq)), DUXC_motif))</span></a>
<a class="sourceLine" id="cb15-17" data-line-number="17">  </a>
<a class="sourceLine" id="cb15-18" data-line-number="18">  <span class="co">#count_motif &lt;- vcountPattern(pattern=hDUX4_motif, subject=peaks_seq, max.mismatch=1) </span></a>
<a class="sourceLine" id="cb15-19" data-line-number="19">  <span class="co">#message(sprintf(&quot;%s: %2.2f %% have motif %s (hDUX4, 1 mismatch)&quot;, </span></a>
<a class="sourceLine" id="cb15-20" data-line-number="20">  <span class="co">#                prefix, 100*(sum(count_motif &gt; 0) / length(peaks_seq)), hDUX4_motif))</span></a>
<a class="sourceLine" id="cb15-21" data-line-number="21">  </a>
<a class="sourceLine" id="cb15-22" data-line-number="22">  <span class="co">#count_motif &lt;- vcountPattern(pattern=mDux_motif, subject=peaks_seq, max.mismatch=1) </span></a>
<a class="sourceLine" id="cb15-23" data-line-number="23">  <span class="co">#message(sprintf(&quot;%s: %2.2f %% have motif %s (mDux, 1 mismatch)&quot;, prefix, </span></a>
<a class="sourceLine" id="cb15-24" data-line-number="24">  <span class="co">#                100*(sum(count_motif &gt; 0) / length(peaks_seq)), mDux_motif))</span></a>
<a class="sourceLine" id="cb15-25" data-line-number="25">  </a>
<a class="sourceLine" id="cb15-26" data-line-number="26">  <span class="kw">return</span>(<span class="kw">invisible</span>())</a>
<a class="sourceLine" id="cb15-27" data-line-number="27">}</a>
<a class="sourceLine" id="cb15-28" data-line-number="28"></a>
<a class="sourceLine" id="cb15-29" data-line-number="29"><span class="kw">.match_motif_pattern</span>(DUXC_mono<span class="op">$</span>ol_de_LINE1, <span class="dt">prefix=</span><span class="st">&quot;DUXC mono&quot;</span>)</a></code></pre></div>
<pre><code>## DUXC mono: 41.18 % have motif AATCA (canonical 2nd motif)</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw">.match_motif_pattern</span>(DUX4_mono<span class="op">$</span>ol_de_LINE1, <span class="dt">prefix=</span><span class="st">&quot;DUX4 mono&quot;</span>)</a></code></pre></div>
<pre><code>## DUX4 mono: 47.00 % have motif AATCA (canonical 2nd motif)</code></pre>
</div>
<div id="location-of-aatca-on-l1-overlapped-peaks" class="section level3">
<h3><span class="header-section-number">10.1.3</span> Location of AATCA on L1-overlapped peaks</h3>
<p>About 70% of AATCA are located within 25 bps from the summit.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="co"># for each peaks that match to AATCA, what&#39;s the relative location of AATCA to the summit?</span></a>
<a class="sourceLine" id="cb19-2" data-line-number="2">.motif_location &lt;-<span class="st"> </span><span class="cf">function</span>(peaks, <span class="dt">tf_name=</span><span class="st">&quot;DUXC mono&quot;</span>) {</a>
<a class="sourceLine" id="cb19-3" data-line-number="3">  <span class="co"># this function finds the location of AATCA (TGATT) w.r.t peaks</span></a>
<a class="sourceLine" id="cb19-4" data-line-number="4">  motif &lt;-<span class="st"> &quot;AATCA&quot;</span></a>
<a class="sourceLine" id="cb19-5" data-line-number="5">  rev_motif &lt;-<span class="st"> &quot;TGATT&quot;</span></a>
<a class="sourceLine" id="cb19-6" data-line-number="6">  peaks_seq &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(<span class="dt">x=</span>BSgenome.Cfamiliaris.UCSC.canFam3, <span class="dt">names=</span>peaks)</a>
<a class="sourceLine" id="cb19-7" data-line-number="7">  m_index &lt;-<span class="st"> </span><span class="kw">vmatchPattern</span>(motif, peaks_seq)</a>
<a class="sourceLine" id="cb19-8" data-line-number="8">  rev_m_index &lt;-<span class="st"> </span><span class="kw">vmatchPattern</span>(rev_motif, peaks_seq)</a>
<a class="sourceLine" id="cb19-9" data-line-number="9">  hist_data &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">motif =</span> <span class="kw">c</span>(<span class="kw">rep</span>(motif, <span class="kw">length</span>(<span class="kw">unlist</span>(m_index))),</a>
<a class="sourceLine" id="cb19-10" data-line-number="10">                                  <span class="kw">rep</span>(rev_motif, <span class="kw">length</span>(<span class="kw">unlist</span>(rev_m_index)))),</a>
<a class="sourceLine" id="cb19-11" data-line-number="11">                        <span class="dt">start =</span> <span class="kw">c</span>(<span class="kw">start</span>(<span class="kw">unlist</span>(m_index)), <span class="kw">start</span>(<span class="kw">unlist</span>(rev_m_index))))</a>
<a class="sourceLine" id="cb19-12" data-line-number="12">  tmp &lt;-<span class="st"> </span><span class="kw">vcountPattern</span>(motif, peaks_seq)</a>
<a class="sourceLine" id="cb19-13" data-line-number="13">  tmp2 &lt;-<span class="st"> </span><span class="kw">vcountPattern</span>(rev_motif, peaks_seq)</a>
<a class="sourceLine" id="cb19-14" data-line-number="14">  title &lt;-<span class="st"> </span><span class="kw">paste0</span>(tf_name, <span class="st">&quot;: &quot;</span>, <span class="kw">sum</span>(tmp2 <span class="op">+</span><span class="st"> </span>tmp <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>), <span class="st">&quot;/&quot;</span>, <span class="kw">length</span>(peaks_seq) )</a>
<a class="sourceLine" id="cb19-15" data-line-number="15">  <span class="co"># message: % between 25 - 75</span></a>
<a class="sourceLine" id="cb19-16" data-line-number="16">  msg &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%2.0f%% between [25, 75]  (summit is at 50)&quot;</span>, </a>
<a class="sourceLine" id="cb19-17" data-line-number="17">                 <span class="dv">100</span><span class="op">*</span><span class="st"> </span><span class="kw">sum</span>(hist_data<span class="op">$</span>start <span class="op">&gt;</span><span class="st"> </span><span class="dv">24</span> <span class="op">&amp;</span><span class="st"> </span>hist_data<span class="op">$</span>start <span class="op">&lt;</span><span class="st"> </span><span class="dv">76</span>)<span class="op">/</span><span class="kw">nrow</span>(hist_data))</a>
<a class="sourceLine" id="cb19-18" data-line-number="18">  <span class="kw">message</span>(msg)</a>
<a class="sourceLine" id="cb19-19" data-line-number="19">  <span class="kw">ggplot</span>(hist_data, <span class="kw">aes</span>(<span class="dt">x=</span>start)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb19-20" data-line-number="20"><span class="st">    </span><span class="kw">geom_histogram</span>(<span class="dt">bins=</span><span class="dv">10</span>, <span class="dt">color=</span><span class="st">&quot;gray25&quot;</span>, <span class="dt">fill=</span><span class="st">&quot;gray90&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb19-21" data-line-number="21"><span class="st">    </span><span class="co">#geom_density(fill=&quot;gray90&quot;) +</span></a>
<a class="sourceLine" id="cb19-22" data-line-number="22"><span class="st">    </span><span class="kw">facet_wrap</span>( <span class="op">~</span><span class="st"> </span>motif) <span class="op">+</span></a>
<a class="sourceLine" id="cb19-23" data-line-number="23"><span class="st">    </span><span class="kw">labs</span>(<span class="dt">title=</span>title) <span class="op">+</span></a>
<a class="sourceLine" id="cb19-24" data-line-number="24"><span class="st">    </span><span class="kw">theme_bw</span>()</a>
<a class="sourceLine" id="cb19-25" data-line-number="25">}</a></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">tidy_duxc_mono_ol &lt;-<span class="st"> </span>DUXC_mono<span class="op">$</span>ol <span class="op">%&gt;%</span><span class="st"> </span>plyranges<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">duplicated</span>(name)) <span class="co"># use unique peaks</span></a>
<a class="sourceLine" id="cb20-2" data-line-number="2">gg &lt;-<span class="st"> </span><span class="kw">.motif_location</span>(tidy_duxc_mono_ol, <span class="dt">tf_name =</span> <span class="st">&quot;DUXC mono&quot;</span>)</a></code></pre></div>
<pre><code>## 67% between [25, 75]  (summit is at 50)</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">gg</a></code></pre></div>
<p><img src="gitbook_files/figure-html/duxc-mono-AATCA-matched-histogram-1.png" width="672" /></p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">tidy_dux4_mono_ol &lt;-<span class="st"> </span>DUX4_mono<span class="op">$</span>ol <span class="op">%&gt;%</span><span class="st"> </span>plyranges<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">duplicated</span>(name)) <span class="co"># use unique peaks</span></a>
<a class="sourceLine" id="cb23-2" data-line-number="2">gg &lt;-<span class="st"> </span><span class="kw">.motif_location</span>(tidy_dux4_mono_ol, <span class="dt">tf_name=</span><span class="st">&quot;DUX4 mono&quot;</span>)</a></code></pre></div>
<pre><code>## 69% between [25, 75]  (summit is at 50)</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">gg</a></code></pre></div>
<p><img src="gitbook_files/figure-html/dux4-mono-AATCA-matched-histogram-1.png" width="672" /></p>
</div>
</div>
<div id="line1s-and-aatca" class="section level2">
<h2><span class="header-section-number">10.2</span> LINE1s and AATCA</h2>
<p>The question now is do L1s are enriched with <code>AATCA/TGATT</code> sequence? How many L1s contains <code>AATCA</code> or <code>TGATT</code>.</p>
<p><strong>Results:</strong> 60% of L1s contains <strong>AATCA</strong> or <strong>TGATT</strong> sequence.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="co"># which L1s have AATCA?</span></a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="co"># within these L1s, how many AATCA a L1 has and what&#39;s the average? </span></a>
<a class="sourceLine" id="cb26-3" data-line-number="3"><span class="co"># does the LINE1 element that has AATCA overlap with DUXC/DUX4 peaks?</span></a>
<a class="sourceLine" id="cb26-4" data-line-number="4"></a>
<a class="sourceLine" id="cb26-5" data-line-number="5"><span class="co"># prun the L1s to standard chromosome</span></a>
<a class="sourceLine" id="cb26-6" data-line-number="6">tidy_L1_rmsk &lt;-<span class="st"> </span><span class="kw">keepStandardChromosomes</span>(L1_rmsk, <span class="dt">pruning.mode=</span><span class="st">&quot;tidy&quot;</span>, <span class="dt">species=</span><span class="st">&quot;Canis_familiaris&quot;</span>)</a>
<a class="sourceLine" id="cb26-7" data-line-number="7">motif &lt;-<span class="st"> &quot;AATCA&quot;</span></a>
<a class="sourceLine" id="cb26-8" data-line-number="8">rev_motif &lt;-<span class="st"> &quot;TGATT&quot;</span></a>
<a class="sourceLine" id="cb26-9" data-line-number="9">L1_seq &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(<span class="dt">x=</span>BSgenome.Cfamiliaris.UCSC.canFam3, <span class="dt">names=</span>tidy_L1_rmsk)</a>
<a class="sourceLine" id="cb26-10" data-line-number="10">m_index &lt;-<span class="st"> </span><span class="kw">vmatchPattern</span>(motif, L1_seq)</a>
<a class="sourceLine" id="cb26-11" data-line-number="11">rev_m_index &lt;-<span class="st"> </span><span class="kw">vmatchPattern</span>(rev_motif, L1_seq)</a>
<a class="sourceLine" id="cb26-12" data-line-number="12"></a>
<a class="sourceLine" id="cb26-13" data-line-number="13">cnt_motif &lt;-<span class="st"> </span><span class="kw">vcountPattern</span>(motif, L1_seq)</a>
<a class="sourceLine" id="cb26-14" data-line-number="14">cnt_rev_motif &lt;-<span class="st"> </span><span class="kw">vcountPattern</span>(rev_motif, L1_seq)</a>
<a class="sourceLine" id="cb26-15" data-line-number="15">freq &lt;-<span class="st"> </span><span class="kw">sum</span>(cnt_motif <span class="op">+</span><span class="st"> </span>cnt_rev_motif <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)</a>
<a class="sourceLine" id="cb26-16" data-line-number="16">msg &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;%2.1f%% (%5.0f/%5.0f) L1s contain either AATCA or TGATT.&quot;</span>, </a>
<a class="sourceLine" id="cb26-17" data-line-number="17">               freq<span class="op">/</span><span class="kw">length</span>(cnt_motif) <span class="op">*</span><span class="st"> </span><span class="dv">100</span>, freq, <span class="kw">length</span>(tidy_L1_rmsk)) </a>
<a class="sourceLine" id="cb26-18" data-line-number="18">msg</a></code></pre></div>
<pre><code>## [1] &quot;60.0% (535563/892594) L1s contain either AATCA or TGATT.&quot;</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="co">#how many AATCA-contained L1s overlap with peaks?</span></a>
<a class="sourceLine" id="cb28-2" data-line-number="2"><span class="co">#which_has_AATCA &lt;- cnt_motif + cnt_rev_motif &gt; 0</span></a>
<a class="sourceLine" id="cb28-3" data-line-number="3"><span class="co">#L1_AATCA &lt;- tidy_L1_rmsk[which_has_AATCA]</span></a>
<a class="sourceLine" id="cb28-4" data-line-number="4"><span class="co">#ol_DUXC &lt;- join_overlap_inner(L1_AATCA, peaks_DUXC_mono, minoverlap=20L) </span></a>
<a class="sourceLine" id="cb28-5" data-line-number="5"><span class="co">#ol_hDUX4 &lt;- join_overlap_inner(L1_AATCA, peaks_hDUX4_mono, minoverlap=20L) </span></a></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="non-LTR.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["gitbook.pdf", "gitbook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
