<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Repeat elements | Conservation and ERVL activation of Canine DUXC</title>
  <meta name="description" content="Bioinformatics and statistics analysis" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Repeat elements | Conservation and ERVL activation of Canine DUXC" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Bioinformatics and statistics analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Repeat elements | Conservation and ERVL activation of Canine DUXC" />
  
  <meta name="twitter:description" content="Bioinformatics and statistics analysis" />
  

<meta name="author" content="Chao-Jen Wong" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="DUXC-early-embryo.html"/>
<link rel="next" href="LTR-activation.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DUXC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#datasets"><i class="fa fa-check"></i><b>0.1</b> Datasets</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#tips"><i class="fa fa-check"></i><b>0.2</b> Tips</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="chipseq.html"><a href="chipseq.html"><i class="fa fa-check"></i><b>1</b> ChIP-seq for the canine model</a><ul>
<li class="chapter" data-level="1.1" data-path="chipseq.html"><a href="chipseq.html#chip-seq-samples"><i class="fa fa-check"></i><b>1.1</b> ChIP-seq samples</a></li>
<li class="chapter" data-level="1.2" data-path="chipseq.html"><a href="chipseq.html#workflow"><i class="fa fa-check"></i><b>1.2</b> workflow</a></li>
<li class="chapter" data-level="1.3" data-path="chipseq.html"><a href="chipseq.html#peaks-and-functional-distribution"><i class="fa fa-check"></i><b>1.3</b> Peaks and functional distribution</a><ul>
<li class="chapter" data-level="1.3.1" data-path="chipseq.html"><a href="chipseq.html#functional-distribution"><i class="fa fa-check"></i><b>1.3.1</b> Functional distribution</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chipseq.html"><a href="chipseq.html#motifs-analysis-need-update"><i class="fa fa-check"></i><b>1.4</b> Motifs analysis NEED UPDATE</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="makeSE.html"><a href="makeSE.html"><i class="fa fa-check"></i><b>2</b> RNA-seq to DESeq2</a><ul>
<li class="chapter" data-level="2.1" data-path="makeSE.html"><a href="makeSE.html#rna-seq-preprocessing"><i class="fa fa-check"></i><b>2.1</b> RNA-seq preprocessing</a></li>
<li class="chapter" data-level="2.2" data-path="makeSE.html"><a href="makeSE.html#reads-counts"><i class="fa fa-check"></i><b>2.2</b> Reads counts</a><ul>
<li class="chapter" data-level="2.2.1" data-path="makeSE.html"><a href="makeSE.html#define-gene-features"><i class="fa fa-check"></i><b>2.2.1</b> Define gene features</a></li>
<li class="chapter" data-level="2.2.2" data-path="makeSE.html"><a href="makeSE.html#read-counts"><i class="fa fa-check"></i><b>2.2.2</b> Read counts</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="makeSE.html"><a href="makeSE.html#deseq2"><i class="fa fa-check"></i><b>2.3</b> DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html"><i class="fa fa-check"></i><b>3</b> DUXC transcriptome</a><ul>
<li class="chapter" data-level="3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#load-datasets-and-settings"><i class="fa fa-check"></i><b>3.1</b> Load datasets and settings</a></li>
<li class="chapter" data-level="3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cinc-duxc-expresison-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2</b> CinC: DUXC expresison in canine muscle cells</a></li>
<li class="chapter" data-level="3.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hinc-dux4-expression-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.3</b> HinC: DUX4 expression in canine muscle cells</a></li>
<li class="chapter" data-level="3.4" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#duxc-alt-expression---caltinc"><i class="fa fa-check"></i><b>3.4</b> DUXC-ALT expression - CALTinC</a><ul>
<li class="chapter" data-level="3.4.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#go-analysis-for-the-down-regulated-gene-set"><i class="fa fa-check"></i><b>3.4.1</b> GO analysis for the down-regulated gene set</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cross-species-models-dux4-expression-in-canine-and-murine"><i class="fa fa-check"></i><b>3.5</b> Cross-species models: DUX4 expression in canine and murine</a><ul>
<li class="chapter" data-level="3.5.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-duxc-expression-in-canine"><i class="fa fa-check"></i><b>3.5.1</b> DUX4 and DUXC expression in canine</a></li>
<li class="chapter" data-level="3.5.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-dux-expression-in-murine"><i class="fa fa-check"></i><b>3.5.2</b> DUX4 and Dux expression in murine</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html"><i class="fa fa-check"></i><b>4</b> DUXC and Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#duxc-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.1</b> DUXC transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs"><i class="fa fa-check"></i><b>4.1.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.1.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome"><i class="fa fa-check"></i><b>4.1.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#dux-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.2</b> Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.2.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs-1"><i class="fa fa-check"></i><b>4.2.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.2.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome-1"><i class="fa fa-check"></i><b>4.2.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html"><i class="fa fa-check"></i><b>5</b> DUXC transcriptome compared to early embryo transcriptome profile</a><ul>
<li class="chapter" data-level="5.1" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#mouse-2c-like-signature"><i class="fa fa-check"></i><b>5.1</b> Mouse 2C-like signature</a></li>
<li class="chapter" data-level="5.2" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#cleavage-stage-signature"><i class="fa fa-check"></i><b>5.2</b> cleavage-stage signature</a></li>
<li class="chapter" data-level="5.3" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#ipscs"><i class="fa fa-check"></i><b>5.3</b> iPSCs</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="repeats.html"><a href="repeats.html"><i class="fa fa-check"></i><b>6</b> Repeat elements</a><ul>
<li class="chapter" data-level="6.1" data-path="repeats.html"><a href="repeats.html#takeaway"><i class="fa fa-check"></i><b>6.1</b> Takeaway</a></li>
<li class="chapter" data-level="6.2" data-path="repeats.html"><a href="repeats.html#repeat-element-expression"><i class="fa fa-check"></i><b>6.2</b> Repeat element expression</a></li>
<li class="chapter" data-level="6.3" data-path="repeats.html"><a href="repeats.html#differential-analysis"><i class="fa fa-check"></i><b>6.3</b> Differential analysis</a><ul>
<li class="chapter" data-level="6.3.1" data-path="repeats.html"><a href="repeats.html#normalization"><i class="fa fa-check"></i><b>6.3.1</b> Normalization</a></li>
<li class="chapter" data-level="6.3.2" data-path="repeats.html"><a href="repeats.html#deseq2-results"><i class="fa fa-check"></i><b>6.3.2</b> DESeq2 results</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="repeats.html"><a href="repeats.html#repeat-family-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> Repeat family enrichment analysis</a><ul>
<li class="chapter" data-level="6.4.1" data-path="repeats.html"><a href="repeats.html#correlation-between-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>6.4.1</b> Correlation between HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="6.4.2" data-path="repeats.html"><a href="repeats.html#repeat-family-activation"><i class="fa fa-check"></i><b>6.4.2</b> Repeat family activation</a></li>
<li class="chapter" data-level="6.4.3" data-path="repeats.html"><a href="repeats.html#repeat-classs-activation"><i class="fa fa-check"></i><b>6.4.3</b> Repeat classs activation</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="repeats.html"><a href="repeats.html#ltr-retroelements-distribution-and-activation"><i class="fa fa-check"></i><b>6.5</b> LTR retroelements distribution and activation</a></li>
<li class="chapter" data-level="6.6" data-path="repeats.html"><a href="repeats.html#peaks-distribution-in-class-of-repeat-elements"><i class="fa fa-check"></i><b>6.6</b> Peaks distribution in class of repeat elements</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="LTR-activation.html"><a href="LTR-activation.html"><i class="fa fa-check"></i><b>7</b> LTR activation</a><ul>
<li class="chapter" data-level="7.1" data-path="LTR-activation.html"><a href="LTR-activation.html#takaway"><i class="fa fa-check"></i><b>7.1</b> Takaway</a></li>
<li class="chapter" data-level="7.2" data-path="LTR-activation.html"><a href="LTR-activation.html#background-in-murine-and-human"><i class="fa fa-check"></i><b>7.2</b> background in murine and human</a></li>
<li class="chapter" data-level="7.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-subfamily-activation-in-canine"><i class="fa fa-check"></i><b>7.3</b> ERVL subfamily activation in canine</a></li>
<li class="chapter" data-level="7.4" data-path="LTR-activation.html"><a href="LTR-activation.html#malr-activation-in-canine"><i class="fa fa-check"></i><b>7.4</b> MaLR activation in canine</a></li>
<li class="chapter" data-level="7.5" data-path="LTR-activation.html"><a href="LTR-activation.html#compare-wih-hinh"><i class="fa fa-check"></i><b>7.5</b> Compare wih HinH</a><ul>
<li class="chapter" data-level="7.5.1" data-path="LTR-activation.html"><a href="LTR-activation.html#copy-number"><i class="fa fa-check"></i><b>7.5.1</b> Copy number</a></li>
<li class="chapter" data-level="7.5.2" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-malr"><i class="fa fa-check"></i><b>7.5.2</b> ERVL-MaLR</a></li>
<li class="chapter" data-level="7.5.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl"><i class="fa fa-check"></i><b>7.5.3</b> ERVL</a></li>
<li class="chapter" data-level="7.5.4" data-path="LTR-activation.html"><a href="LTR-activation.html#erv1"><i class="fa fa-check"></i><b>7.5.4</b> ERV1</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="LTR-activation.html"><a href="LTR-activation.html#duxc-alt-and-ervl-transcriptions"><i class="fa fa-check"></i><b>7.6</b> DUXC-ALT and ERVL transcriptions</a><ul>
<li class="chapter" data-level="7.6.1" data-path="LTR-activation.html"><a href="LTR-activation.html#erv-subfamily-expression"><i class="fa fa-check"></i><b>7.6.1</b> ERV subfamily expression</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="LTR-activation.html"><a href="LTR-activation.html#LTR-activation"><i class="fa fa-check"></i><b>8</b> LTR activation</a><ul>
<li class="chapter" data-level="8.1" data-path="LTR-activation.html"><a href="LTR-activation.html#copy-number-1"><i class="fa fa-check"></i><b>8.1</b> Copy number</a></li>
<li class="chapter" data-level="8.2" data-path="LTR-activation.html"><a href="LTR-activation.html#sattillites-activiation"><i class="fa fa-check"></i><b>8.2</b> Sattillites activiation</a></li>
<li class="chapter" data-level="8.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ltrs-activation"><i class="fa fa-check"></i><b>8.3</b> LTRs activation</a><ul>
<li class="chapter" data-level="8.3.1" data-path="LTR-activation.html"><a href="LTR-activation.html#erv1-1"><i class="fa fa-check"></i><b>8.3.1</b> ERV1</a></li>
<li class="chapter" data-level="8.3.2" data-path="LTR-activation.html"><a href="LTR-activation.html#direct-binding-of-dux4-and-duxc"><i class="fa fa-check"></i><b>8.3.2</b> Direct binding of DUX4 and DUXC</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="LTR-activation.html"><a href="LTR-activation.html#line1-activation"><i class="fa fa-check"></i><b>8.4</b> LINE1 activation</a></li>
<li class="chapter" data-level="8.5" data-path="LTR-activation.html"><a href="LTR-activation.html#satellites"><i class="fa fa-check"></i><b>8.5</b> Satellites</a></li>
<li class="chapter" data-level="8.6" data-path="LTR-activation.html"><a href="LTR-activation.html#sines"><i class="fa fa-check"></i><b>8.6</b> SINEs</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="non-LTR.html"><a href="non-LTR.html"><i class="fa fa-check"></i><b>9</b> Non-LTR retroelements and satellites</a><ul>
<li class="chapter" data-level="9.1" data-path="non-LTR.html"><a href="non-LTR.html#line"><i class="fa fa-check"></i><b>9.1</b> LINE</a></li>
<li class="chapter" data-level="9.2" data-path="non-LTR.html"><a href="non-LTR.html#sine"><i class="fa fa-check"></i><b>9.2</b> SINE</a><ul>
<li class="chapter" data-level="9.2.1" data-path="non-LTR.html"><a href="non-LTR.html#distribution-of-sine-family"><i class="fa fa-check"></i><b>9.2.1</b> Distribution of SINE family</a></li>
<li class="chapter" data-level="9.2.2" data-path="non-LTR.html"><a href="non-LTR.html#expression-of-sines-in-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>9.2.2</b> Expression of SINEs in HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="9.2.3" data-path="non-LTR.html"><a href="non-LTR.html#trna-lys"><i class="fa fa-check"></i><b>9.2.3</b> tRNA-Lys</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="non-LTR.html"><a href="non-LTR.html#satellites-1"><i class="fa fa-check"></i><b>9.3</b> Satellites</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="peaks-line1.html"><a href="peaks-line1.html"><i class="fa fa-check"></i><b>10</b> DUXC peaks and LINE1s</a><ul>
<li class="chapter" data-level="10.1" data-path="peaks-line1.html"><a href="peaks-line1.html#peaks-of-cch-and-hdux4"><i class="fa fa-check"></i><b>10.1</b> Peaks of CCH and hDUX4</a></li>
<li class="chapter" data-level="10.2" data-path="peaks-line1.html"><a href="peaks-line1.html#overlapping-with-expressed-l1s"><i class="fa fa-check"></i><b>10.2</b> Overlapping with expressed L1s</a><ul>
<li class="chapter" data-level="10.2.1" data-path="peaks-line1.html"><a href="peaks-line1.html#peaks-overlap-with-l1s"><i class="fa fa-check"></i><b>10.2.1</b> Peaks overlap with L1s</a></li>
<li class="chapter" data-level="10.2.2" data-path="peaks-line1.html"><a href="peaks-line1.html#duxc-motif-discovery-using-l1-overlapped-peaks"><i class="fa fa-check"></i><b>10.2.2</b> DUXC Motif discovery using L1-overlapped peaks</a></li>
<li class="chapter" data-level="10.2.3" data-path="peaks-line1.html"><a href="peaks-line1.html#match-dux4-and-duxc-motif-to-l1-overapped-peaks"><i class="fa fa-check"></i><b>10.2.3</b> Match DUX4 and DUXC motif to L1-overapped peaks</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="peaks-line1.html"><a href="peaks-line1.html#line1-element-with-aatca"><i class="fa fa-check"></i><b>10.3</b> LINE1 element with AATCA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i># References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Conservation and ERVL activation of Canine DUXC</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="repeats" class="section level1">
<h1><span class="header-section-number">Chapter 6</span> Repeat elements</h1>
<p>In this chapter we performed bioinformatic profiling on repeat elements inferred repeat family and class activation in CinC and HinC transcriptome using gene set enrichment analysis (GSEA). The associated computational methods are:<br />
- counting reads that overlap with repeat elements<br />
- identifing DUX4/DUXC affected repeat elements by <em>DESeq2</em><br />
- comparing CinC to HinC transcriptome<br />
- analying family and class enrichment in repeat elements<br />
- estimating peaks distribution in class of repeat elements: show preference on binding to LTR.</p>
<p>the following chapters, 7 and 8, investigated more detials of retrotransposons and satellite repeats. (regulatory activities in our canine model and the impact in DUXC and DUX4 expressed canine muscles.)</p>
<div id="takeaway" class="section level2">
<h2><span class="header-section-number">6.1</span> Takeaway</h2>
<p>The RNA expression levels of repeat elements in DUXC expressed muscle cells is highly correlated with that in DUX4 (Pearson=0.782). We used the repeats annotation from UCSC Repeat Masker (RMSK) track on main chromosomes of canFam3 genome and applied an ad-hoc <em>RMSKstats</em> package (github URL) to estimate read counts on the annotated repeat elements (see Methods). With a threshold of adjusted <span class="math inline">\(p\)</span>-value &lt; 0.05 corresponding to the null hypothesis <span class="math inline">\(H_0: |log2FC| &lt; 1\)</span>, <em>DESeq2</em> identified 231 and 99 differentially expressed repeat elements in DUX4 and DUXC expressed canine skeletal cells, respectively, compared to cells expressing Luciferase. 86% (83/99) of the differentially expressed repeat elements in CinC is overlapped with that of in HinC.</p>
<p>DUX4 continued to show the trend of activating retrotransposons as it did in human skeletal muscle (Whidden 2017) and in the 2C-like embryonic stage (Hendrickson 2017). Similarly, DUXC followed the same trend in canine. In fact, by gene set enrichment analysis using hypergeometric tests (code shown below), LINE1, ERV1, ERVL, and ERVL-MaLR LTR subfamilies were enriched in both DUX4 and DUXC expressed canine skeletal muscle. Almost all the activated LTRs of DUXC were overlapping with that of DUX4. Moreover, the peak binding distribution in classes of repeat element confirmed the preference of DUX4 and DUXC in binding to LTRs and LINE1: LTRs and LINEs constitute 5% and 15% of the reference canine genome, respectively, whereas the frequency of the peak binding sites (within +/- 50 from the summit) on LTRs and LINEs are 14% ~ 17% and 21% ~ 35%, respectively. The enrichment of LTRs activation is statistically significant. But the evidence of DUXC’s direct activation to LINE1 remains subtle and requires further investigation (e.g. motifs of peaks that bind to LINE1).</p>
<p><strong>Load libraries and datasets</strong></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(TxDb.Cfamiliaris.UCSC.canFam3.ensGene))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(canFam3.rmsk))</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(tidyverse))</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(wesanderson) )</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(DESeq2))</a>
<a class="sourceLine" id="cb1-6" data-line-number="6">pkg_dir &lt;-<span class="st"> &quot;~/CompBio/canFam3.DuxFamily&quot;</span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7">fig_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;figures&quot;</span>)</a>
<a class="sourceLine" id="cb1-8" data-line-number="8">data_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>)</a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;peaks_list.rda&quot;</span>))</a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;rmsk.dds.rda&quot;</span>))</a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;CinC.ens.dds.rda&quot;</span>))</a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;HinC.ens.dds.rda&quot;</span>))</a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;CALTinC.ens.dds.rda&quot;</span>))</a></code></pre></div>
</div>
<div id="repeat-element-expression" class="section level2">
<h2><span class="header-section-number">6.2</span> Repeat element expression</h2>
<p>The repeat elements we used here were collected from UCSC’s Repeat Masker (RMSK) annotation track. We estimated the repeat elements expression by counting the reads that overlap the instances of repeat elements (featured by RMSK’s <code>repName</code>); each read count was adjusted by its number of reported multiple alignments (column NH, BAM format). The fractional counts were then rounded and saved as an <code>DESeqDataSet</code> instance (<code>/data/rmsk.dds.rda</code>)</p>
<p>The <code>countHits()</code> function, code shown beloww, performed the counting; it is only being tested for single-ended RNA-seq reads (see <code>/scripts/summarizeOverlaps.adjNH.R</code>).</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># summarizeOverlaps-adjNH(features, all.reads)</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">.countSubjectHits<span class="fl">.2</span> &lt;-<span class="st"> </span><span class="cf">function</span>(ov, features, reads) {</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">    fracReadCount &lt;-<span class="st"> </span>reads[<span class="kw">queryHits</span>(ov)]<span class="op">$</span>count<span class="op">/</span>reads[<span class="kw">queryHits</span>(ov)]<span class="op">$</span>NH</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">    cnt &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>, <span class="kw">length</span>(features))</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">    <span class="kw">names</span>(cnt) &lt;-<span class="st"> </span><span class="kw">names</span>(features)</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">    tmp &lt;-<span class="st"> </span><span class="kw">tapply</span>(fracReadCount, <span class="kw">subjectHits</span>(ov), sum)</a>
<a class="sourceLine" id="cb2-7" data-line-number="7">    cnt[<span class="kw">as.integer</span>(<span class="kw">names</span>(tmp))] &lt;-<span class="st"> </span>tmp</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">    <span class="kw">as.integer</span>(<span class="kw">round</span>(cnt))</a>
<a class="sourceLine" id="cb2-9" data-line-number="9">}</a>
<a class="sourceLine" id="cb2-10" data-line-number="10"></a>
<a class="sourceLine" id="cb2-11" data-line-number="11">countHits &lt;-<span class="st"> </span><span class="cf">function</span>(features, reads, <span class="dt">type=</span><span class="kw">c</span>(<span class="st">&quot;any&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>, <span class="st">&quot;within&quot;</span>),</a>
<a class="sourceLine" id="cb2-12" data-line-number="12">                  <span class="dt">ignore.strand=</span><span class="ot">TRUE</span>, <span class="dt">inter.feature=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb2-13" data-line-number="13">{   <span class="co">## feathers is GRangeList containing repName&#39;s range data (importd from the canFam.rmsk package)</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14">    <span class="co">## reads is GAlignment instance - see GenomicAlignments package</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15">    ov &lt;-<span class="st"> </span><span class="kw">findOverlaps</span>(reads, features, <span class="dt">type=</span><span class="kw">match.arg</span>(type),</a>
<a class="sourceLine" id="cb2-16" data-line-number="16">                       <span class="dt">ignore.strand=</span>ignore.strand)</a>
<a class="sourceLine" id="cb2-17" data-line-number="17">    <span class="cf">if</span> (inter.feature) {</a>
<a class="sourceLine" id="cb2-18" data-line-number="18">        <span class="co">## Remove ambigous reads.</span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19">        reads_to_keep &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">countQueryHits</span>(ov) <span class="op">==</span><span class="st"> </span>1L)</a>
<a class="sourceLine" id="cb2-20" data-line-number="20">        ov &lt;-<span class="st"> </span>ov[<span class="kw">queryHits</span>(ov) <span class="op">%in%</span><span class="st"> </span>reads_to_keep]</a>
<a class="sourceLine" id="cb2-21" data-line-number="21">    }</a>
<a class="sourceLine" id="cb2-22" data-line-number="22">    <span class="kw">.countSubjectHits.2</span>(ov, <span class="dt">features=</span>features, <span class="dt">reads=</span>reads)</a>
<a class="sourceLine" id="cb2-23" data-line-number="23">}</a></code></pre></div>
</div>
<div id="differential-analysis" class="section level2">
<h2><span class="header-section-number">6.3</span> Differential analysis</h2>
<p><strong>Aims</strong><br />
- Compare DUXC (CinC) and DUX4 (HinC) expressed cell lines with Luciferase expressed<br />
- Identify targets of DUXC and DUX4 in repeat elements</p>
<div id="normalization" class="section level3">
<h3><span class="header-section-number">6.3.1</span> Normalization</h3>
<p>Prior to differential analysis, we needed to set appropriate, unbiased size factor for each sample. DUXC and DUX4 tend to activate retrotransponsons and result much more repeat element transcripts than that of Luciferase. So merely using the number of repeat elements trasncripts for size factors would magnify the normalized expression induced by Luciferase. Therefore we used the size factors estimated by DESeq2 over the total transcripts (all gene features).</p>
</div>
<div id="deseq2-results" class="section level3">
<h3><span class="header-section-number">6.3.2</span> DESeq2 results</h3>
<p>Differentially expressed repeat elements (featured by <code>repName</code>) were called if adjusted <span class="math inline">\(p\)</span>-value <span class="math inline">\(&lt; 0.05\)</span>, corresponding to null hypothesis <span class="math inline">\(H_0:|log2FC| &lt; 1\)</span>. Among ~ 700 annotated repeat elements, the DESeq2 analysis resulted 231 and 99 differentially up-regulated repeat elements in HinC and CinC models, respectively. Code and MA plots are shown below.</p>
<p><strong>Adjust size factors</strong></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">new_rmsk_dds &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">names</span>(rmsk.dds), <span class="cf">function</span>(name) {</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">  <span class="co"># use size factor estimated over total transcripts</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3">  dds &lt;-<span class="st"> </span>rmsk.dds[[name]]</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">    </a>
<a class="sourceLine" id="cb3-5" data-line-number="5">    <span class="cf">if</span> (name <span class="op">==</span><span class="st"> &quot;CinC&quot;</span>)</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">      <span class="kw">sizeFactors</span>(dds) &lt;-<span class="st"> </span><span class="kw">sizeFactors</span>(CinC.ens.dds)</a>
<a class="sourceLine" id="cb3-7" data-line-number="7"></a>
<a class="sourceLine" id="cb3-8" data-line-number="8">    <span class="cf">if</span> (name <span class="op">==</span><span class="st"> &quot;HinC&quot;</span>)</a>
<a class="sourceLine" id="cb3-9" data-line-number="9">      <span class="kw">sizeFactors</span>(dds) &lt;-<span class="st"> </span><span class="kw">sizeFactors</span>(HinC.ens.dds)  </a>
<a class="sourceLine" id="cb3-10" data-line-number="10"></a>
<a class="sourceLine" id="cb3-11" data-line-number="11">    <span class="cf">if</span> (name <span class="op">==</span><span class="st"> &quot;CALTinC&quot;</span>)  </a>
<a class="sourceLine" id="cb3-12" data-line-number="12">      <span class="kw">sizeFactors</span>(dds) &lt;-<span class="st"> </span><span class="kw">sizeFactors</span>(CALTinC.ens.dds)</a>
<a class="sourceLine" id="cb3-13" data-line-number="13"></a>
<a class="sourceLine" id="cb3-14" data-line-number="14">    dds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds)</a>
<a class="sourceLine" id="cb3-15" data-line-number="15"></a>
<a class="sourceLine" id="cb3-16" data-line-number="16">    <span class="kw">return</span>(dds)</a>
<a class="sourceLine" id="cb3-17" data-line-number="17">})</a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="kw">names</span>(new_rmsk_dds) &lt;-<span class="st"> </span><span class="kw">names</span>(rmsk.dds)</a></code></pre></div>
<p><strong>Make MA plots</strong></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># MA plots for CinC and HinC</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw">lapply</span>(<span class="kw">names</span>(new_rmsk_dds[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]), <span class="cf">function</span>(name) {</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">    <span class="co"># make MA plots</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4">    dds &lt;-<span class="st"> </span>new_rmsk_dds[[name]]</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">    res &lt;-<span class="st"> </span><span class="kw">results</span>(dds, <span class="dt">alpha=</span><span class="fl">0.05</span>, <span class="dt">lfcThreshold =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">    summary &lt;-<span class="st"> </span><span class="kw">as</span>(res, <span class="st">&quot;data.frame&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="st">      </span><span class="kw">summarise</span>(<span class="dt">up =</span> <span class="kw">sum</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>), </a>
<a class="sourceLine" id="cb4-8" data-line-number="8">      <span class="dt">down =</span> <span class="kw">sum</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&lt;</span><span class="st"> </span><span class="dv">1</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb4-9" data-line-number="9"></a>
<a class="sourceLine" id="cb4-10" data-line-number="10">    file_name &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;rmsk_&quot;</span>, name, <span class="st">&quot;_MAplot.pdf&quot;</span>)</a>
<a class="sourceLine" id="cb4-11" data-line-number="11">    y_text &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">down=</span><span class="op">-</span><span class="dv">4</span>, <span class="dt">up=</span><span class="dv">4</span>)</a>
<a class="sourceLine" id="cb4-12" data-line-number="12"></a>
<a class="sourceLine" id="cb4-13" data-line-number="13">    <span class="cf">if</span> (name <span class="op">==</span><span class="st"> &quot;CinC&quot;</span>) y_text &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">down=</span><span class="op">-</span><span class="dv">2</span>, <span class="dt">up=</span><span class="dv">2</span>)</a>
<a class="sourceLine" id="cb4-14" data-line-number="14"></a>
<a class="sourceLine" id="cb4-15" data-line-number="15">    <span class="co">#pdf(file.path(fig_dir, file_name), height=4, width=4.5)</span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16">    <span class="kw">plotMA</span>(res, <span class="dt">alpha =</span> <span class="fl">0.05</span>, </a>
<a class="sourceLine" id="cb4-17" data-line-number="17">           <span class="dt">main=</span><span class="kw">paste0</span>(name, <span class="st">&quot;: &quot;</span>, <span class="kw">nrow</span>(res), <span class="st">&quot; annotated elements&quot;</span>))</a>
<a class="sourceLine" id="cb4-18" data-line-number="18">    label &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;LFC &gt; 1 (up) : %d, %1.1f%%&quot;</span>, </a>
<a class="sourceLine" id="cb4-19" data-line-number="19">                     <span class="kw">pull</span>(summary, up), <span class="kw">pull</span>(summary, up)<span class="op">/</span><span class="kw">nrow</span>(res) <span class="op">*</span><span class="dv">100</span>)           </a>
<a class="sourceLine" id="cb4-20" data-line-number="20">    <span class="kw">text</span>(<span class="dt">x=</span><span class="dv">80</span>, <span class="dt">y=</span>y_text[<span class="st">&quot;up&quot;</span>],   <span class="dt">labels=</span>label, <span class="dt">adj=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))</a>
<a class="sourceLine" id="cb4-21" data-line-number="21">    label &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;LFC &lt; 1 (down) : %d, %1.1f%%&quot;</span>, </a>
<a class="sourceLine" id="cb4-22" data-line-number="22">                     <span class="kw">pull</span>(summary, down), <span class="kw">pull</span>(summary, down)<span class="op">/</span><span class="kw">nrow</span>(res) <span class="op">*</span><span class="dv">100</span>)     </a>
<a class="sourceLine" id="cb4-23" data-line-number="23">    <span class="kw">text</span>(<span class="dt">x=</span><span class="dv">80</span>, <span class="dt">y=</span>y_text[<span class="st">&quot;down&quot;</span>], <span class="dt">labels=</span>label, <span class="dt">adj=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb4-24" data-line-number="24">    <span class="co">#dev.off()  </span></a>
<a class="sourceLine" id="cb4-25" data-line-number="25">})</a></code></pre></div>
<div class="figure"><span id="fig:rmsk-MAplot1"></span>
<img src="gitbook_files/figure-html/rmsk-MAplot-1.png" alt="MAPlots of DESeq2 results for HinC and CinC transcriptome." width="672" />
<p class="caption">
Figure 6.1: MAPlots of DESeq2 results for HinC and CinC transcriptome.
</p>
</div>
<div class="figure"><span id="fig:rmsk-MAplot2"></span>
<img src="gitbook_files/figure-html/rmsk-MAplot-2.png" alt="MAPlots of DESeq2 results for HinC and CinC transcriptome." width="672" />
<p class="caption">
Figure 6.2: MAPlots of DESeq2 results for HinC and CinC transcriptome.
</p>
</div>
<pre><code>## [[1]]
## NULL
## 
## [[2]]
## NULL</code></pre>
</div>
</div>
<div id="repeat-family-enrichment-analysis" class="section level2">
<h2><span class="header-section-number">6.4</span> Repeat family enrichment analysis</h2>
<p>To inferr which repeat family or class is enriched (compared with luciferase), we implemented gene set enrichment analysis (GSEA) using hypergeometric tests (see <code>/scripts/tools.R</code>). The <code>.rmsk_enrichment()</code> function below returned both the DESeq2 and GSEA results.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">.rmsk_enrichment &lt;-<span class="st"> </span><span class="cf">function</span>(rmsk_dds) {</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">    <span class="co"># DESeq2 threshold: p-value=0.05 correspondng to H_0: |lfc| &lt; 1 </span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3">    rowdata &lt;-<span class="st"> </span><span class="kw">as</span>(<span class="kw">rowData</span>(rmsk_dds), <span class="st">&quot;data.frame&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="st">      </span><span class="kw">rownames_to_column</span>(<span class="dt">var=</span><span class="st">&quot;repName&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">select</span>(repName, repClass, repFamily)</a>
<a class="sourceLine" id="cb6-6" data-line-number="6">    <span class="co"># results using designated threshold and null hypothesis</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7">    res_df &lt;-<span class="st"> </span><span class="kw">as</span>(<span class="kw">results</span>(rmsk_dds, <span class="dt">alpha=</span><span class="fl">0.05</span>, <span class="dt">lfcThreshold=</span><span class="dv">1</span>), <span class="st">&quot;data.frame&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="st">      </span><span class="kw">rownames_to_column</span>(<span class="dt">var=</span><span class="st">&quot;repName&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="st">      </span><span class="kw">left_join</span>(rowdata, <span class="dt">by=</span><span class="st">&quot;repName&quot;</span>) </a>
<a class="sourceLine" id="cb6-10" data-line-number="10"></a>
<a class="sourceLine" id="cb6-11" data-line-number="11">    universe &lt;-<span class="st"> </span><span class="kw">nrow</span>(res_df)</a>
<a class="sourceLine" id="cb6-12" data-line-number="12">    selected_up &lt;-<span class="st"> </span>res_df <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="st">      </span><span class="kw">summarise</span>(<span class="dt">up =</span> <span class="kw">sum</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="st">      </span><span class="kw">pull</span>(up)</a>
<a class="sourceLine" id="cb6-15" data-line-number="15">    selected_down &lt;-<span class="st"> </span>res_df <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="st">      </span><span class="kw">summarise</span>(<span class="dt">down =</span> <span class="kw">sum</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="st">      </span><span class="kw">pull</span>(down)</a>
<a class="sourceLine" id="cb6-18" data-line-number="18"></a>
<a class="sourceLine" id="cb6-19" data-line-number="19">    <span class="co"># repFamily enrichment/depletion summary</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20">    repFamily_summary &lt;-<span class="st"> </span>res_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(repFamily) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"><span class="st">      </span><span class="kw">summarise</span>(<span class="dt">total=</span><span class="kw">n</span>(), </a>
<a class="sourceLine" id="cb6-22" data-line-number="22">                <span class="dt">up =</span> <span class="kw">sum</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>), </a>
<a class="sourceLine" id="cb6-23" data-line-number="23">                <span class="dt">down =</span> <span class="kw">sum</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb6-24" data-line-number="24">                <span class="dt">enrichment_prob =</span> <span class="kw">dhyper</span>(<span class="dt">x=</span>up, <span class="dt">k=</span>total, <span class="dt">m=</span>selected_up, <span class="dt">n=</span>universe <span class="op">-</span><span class="st"> </span>selected_up),</a>
<a class="sourceLine" id="cb6-25" data-line-number="25">                <span class="dt">enrichment_mu =</span> total <span class="op">*</span><span class="st"> </span>selected_up<span class="op">/</span>universe)</a>
<a class="sourceLine" id="cb6-26" data-line-number="26">                <span class="co">#depletion_prob = dhyper(x=down, k=total, m=selected_down,</span></a>
<a class="sourceLine" id="cb6-27" data-line-number="27">                <span class="co">#                        n=universe-selected_down),</span></a>
<a class="sourceLine" id="cb6-28" data-line-number="28">                <span class="co">#depletion_mu = total * selected_down/universe)</span></a>
<a class="sourceLine" id="cb6-29" data-line-number="29"></a>
<a class="sourceLine" id="cb6-30" data-line-number="30">    <span class="co"># repClass enrichement/depletion summary</span></a>
<a class="sourceLine" id="cb6-31" data-line-number="31">    repClass_summary &lt;-<span class="st"> </span>res_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(repClass) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-32" data-line-number="32"><span class="st">       </span><span class="kw">summarise</span>(<span class="dt">total=</span><span class="kw">n</span>(), </a>
<a class="sourceLine" id="cb6-33" data-line-number="33">                <span class="dt">up =</span> <span class="kw">sum</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>), </a>
<a class="sourceLine" id="cb6-34" data-line-number="34">                <span class="dt">down =</span> <span class="kw">sum</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb6-35" data-line-number="35">                <span class="dt">enrichment_prob =</span> <span class="kw">dhyper</span>(<span class="dt">x=</span>up, <span class="dt">k=</span>total, <span class="dt">m=</span>selected_up, <span class="dt">n=</span>universe <span class="op">-</span><span class="st"> </span>selected_up),</a>
<a class="sourceLine" id="cb6-36" data-line-number="36">                <span class="dt">enrichment_mu =</span> total <span class="op">*</span><span class="st"> </span>selected_up<span class="op">/</span>universe)</a>
<a class="sourceLine" id="cb6-37" data-line-number="37">    <span class="kw">return</span>(<span class="kw">list</span>(<span class="dt">res_df=</span>res_df, </a>
<a class="sourceLine" id="cb6-38" data-line-number="38">                <span class="dt">repFamily_summary=</span>repFamily_summary, </a>
<a class="sourceLine" id="cb6-39" data-line-number="39">                <span class="dt">repClass_summary=</span>repClass_summary))               </a>
<a class="sourceLine" id="cb6-40" data-line-number="40">}</a></code></pre></div>
<p>Calling <code>.rmsk_enrichement()</code> and saving results <code>HinC_rmsk.rda</code> and <code>CinC_rmsk.rda</code>:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">source</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;scripts&quot;</span>, <span class="st">&quot;tools.R&quot;</span>)) <span class="co"># .rmsk_enrichment</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2">HinC_rmsk &lt;-<span class="st"> </span><span class="kw">.rmsk_enrichment</span>(new_rmsk_dds[[<span class="st">&quot;HinC&quot;</span>]])</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">CinC_rmsk &lt;-<span class="st"> </span><span class="kw">.rmsk_enrichment</span>(new_rmsk_dds[[<span class="st">&quot;CinC&quot;</span>]])</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">CALTinC_rmsk &lt;-<span class="st"> </span><span class="kw">.rmsk_enrichment</span>(new_rmsk_dds[[<span class="st">&quot;CALTinC&quot;</span>]])</a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#save(HinC_rmsk, file=file.path(data_dir, &quot;HinC_rmsk.rda&quot;))</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#save(CinC_rmsk, file=file.path(data_dir, &quot;CinC_rmsk.rda&quot;))</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#save(CALTinC_rmsk, file=file.path(pkg_dir, &quot;data&quot;, &quot;CALTinC_rmsk.rda&quot;))</span></a></code></pre></div>
<div id="correlation-between-hinc-and-cinc-transcriptome" class="section level3">
<h3><span class="header-section-number">6.4.1</span> Correlation between HinC and CinC transcriptome</h3>
<p>The CinC and HinC transcriptome are highly correlated - Pearson correlation is 0.782. 86% (83/99) of the differentially expressed repeat elements in CinC is overlapped with that of in HinC.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># (1c) scatter plot of HinC and CinC; highlight LTR: ERVL-MaLR</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2">comb &lt;-<span class="st"> </span><span class="kw">inner_join</span>(HinC_rmsk<span class="op">$</span>res_df, CinC_rmsk<span class="op">$</span>res_df, </a>
<a class="sourceLine" id="cb8-3" data-line-number="3">                   <span class="dt">by=</span><span class="st">&quot;repName&quot;</span>, <span class="dt">suffix=</span><span class="kw">c</span>(<span class="st">&quot;_HinC&quot;</span>, <span class="st">&quot;_CinC&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">show_name =</span> <span class="kw">as.character</span>(repFamily_HinC)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">show_name =</span> <span class="kw">ifelse</span>(<span class="kw">abs</span>(log2FoldChange_HinC<span class="op">-</span>log2FoldChange_CinC) <span class="op">&lt;</span><span class="st"> </span><span class="fl">1.5</span>, </a>
<a class="sourceLine" id="cb8-6" data-line-number="6">                                   <span class="st">&quot;&quot;</span>, show_name)) <span class="op">%&gt;%</span><span class="st">      </span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">DE_status =</span> <span class="st">&quot;neither&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">DE_status =</span> </a>
<a class="sourceLine" id="cb8-9" data-line-number="9">                  <span class="kw">replace</span>(DE_status, padj_HinC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">|</span><span class="st"> </span>padj_CinC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, <span class="st">&quot;either&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">DE_status =</span> <span class="kw">replace</span>(DE_status, padj_HinC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>padj_CinC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, <span class="st">&quot;both&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">DE_status =</span> <span class="kw">factor</span>(DE_status, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;neither&quot;</span>, <span class="st">&quot;either&quot;</span>, <span class="st">&quot;both&quot;</span>)))                                          </a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="co"># label repeat family if |HinC - CInC lfc| &gt; 1.5                </span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="kw">library</span>(ggrepel)</a>
<a class="sourceLine" id="cb8-14" data-line-number="14"></a>
<a class="sourceLine" id="cb8-15" data-line-number="15">gg1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(comb, <span class="kw">aes</span>(<span class="dt">x=</span>log2FoldChange_CinC, <span class="dt">y=</span>log2FoldChange_HinC)) <span class="op">+</span></a>
<a class="sourceLine" id="cb8-16" data-line-number="16"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">1</span>, <span class="dt">alpha=</span><span class="fl">0.7</span>, <span class="kw">aes</span>(<span class="dt">color=</span>DE_status)) <span class="op">+</span></a>
<a class="sourceLine" id="cb8-17" data-line-number="17"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#999999&quot;</span>,<span class="st">&quot;#E69F00&quot;</span>, <span class="st">&quot;#56B4E9&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb8-18" data-line-number="18"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="dt">method=</span><span class="st">&quot;lm&quot;</span>, <span class="dt">se=</span><span class="ot">FALSE</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>, <span class="dt">color=</span><span class="st">&quot;gray66&quot;</span>, <span class="dt">alpha=</span><span class="fl">0.5</span>,</a>
<a class="sourceLine" id="cb8-19" data-line-number="19">              <span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb8-20" data-line-number="20"><span class="st">  </span><span class="kw">geom_abline</span>(<span class="dt">intercept=</span><span class="dv">0</span>, <span class="dt">slope=</span><span class="dv">1</span>, <span class="dt">color=</span><span class="st">&quot;gray75&quot;</span>, <span class="dt">alpha=</span><span class="fl">0.2</span>, <span class="dt">size=</span><span class="fl">0.5</span>) <span class="op">+</span><span class="st">              </span></a>
<a class="sourceLine" id="cb8-21" data-line-number="21"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;RMSK: HinC vs CinC log2FC&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb8-22" data-line-number="22"><span class="st">  </span><span class="kw">theme_bw</span>()<span class="op">+</span></a>
<a class="sourceLine" id="cb8-23" data-line-number="23"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.88</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb8-24" data-line-number="24"><span class="st">  </span><span class="kw">geom_text_repel</span>(<span class="kw">aes</span>(<span class="dt">label=</span>show_name), <span class="dt">size=</span><span class="fl">2.5</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb8-25" data-line-number="25"></a>
<a class="sourceLine" id="cb8-26" data-line-number="26">df &lt;-<span class="st"> </span>comb <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(log2FoldChange_HinC, log2FoldChange_CinC) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-27" data-line-number="27"><span class="st">  </span><span class="kw">rename</span>(<span class="dt">y=</span>log2FoldChange_HinC, <span class="dt">x=</span>log2FoldChange_CinC)  </a>
<a class="sourceLine" id="cb8-28" data-line-number="28">gg1 &lt;-<span class="st"> </span>gg1 <span class="op">+</span><span class="st"> </span><span class="kw">geom_text</span>(<span class="dt">x =</span> <span class="dv">7</span>, <span class="dt">y =</span> <span class="fl">8.3</span>, <span class="dt">vjust=</span><span class="dv">0</span>, <span class="dt">hjust=</span><span class="fl">0.5</span>, </a>
<a class="sourceLine" id="cb8-29" data-line-number="29">  <span class="dt">label =</span> <span class="kw">lm_eqn</span>(df), <span class="dt">parse =</span> <span class="ot">TRUE</span>, <span class="dt">color=</span><span class="st">&quot;gray65&quot;</span>, <span class="dt">alpha=</span><span class="fl">0.5</span>)</a>
<a class="sourceLine" id="cb8-30" data-line-number="30">gg1 </a></code></pre></div>
<div class="figure"><span id="fig:rmsk-scatter-HinC-CinC"></span>
<img src="gitbook_files/figure-html/rmsk-scatter-HinC-CinC-1.png" alt="Scatter plot of repeat elements' (repNAME) logFC in HinC and CinC with linear regression line (dashed line). The Pearson correlation is 0.782. Each dot represents a repeat element; an element's repeat family is labelled if its difference between theå logFC in HinC and CinC is greater than 1.5." width="672" />
<p class="caption">
Figure 6.3: Scatter plot of repeat elements’ (repNAME) logFC in HinC and CinC with linear regression line (dashed line). The Pearson correlation is 0.782. Each dot represents a repeat element; an element’s repeat family is labelled if its difference between theå logFC in HinC and CinC is greater than 1.5.
</p>
</div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">pdf</span>(<span class="kw">file.path</span>(fig_dir, <span class="st">&quot;rmsk_HinC_CinC_log2FC_scatter.pdf&quot;</span>),</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">    <span class="dt">width=</span><span class="dv">6</span>, <span class="dt">height=</span><span class="dv">6</span>)  </a>
<a class="sourceLine" id="cb9-3" data-line-number="3">gg1</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="kw">dev.off</span>()</a></code></pre></div>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co">#</span></a></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(ggVennDiagram))</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">duxc_de &lt;-<span class="st"> </span>CinC_rmsk<span class="op">$</span>res_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">dux4_de &lt;-<span class="st"> </span>HinC_rmsk<span class="op">$</span>res_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">venn_data &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">CinC =</span> <span class="kw">as.character</span>(duxc_de<span class="op">$</span>repName), </a>
<a class="sourceLine" id="cb12-5" data-line-number="5">                  <span class="dt">HinC =</span> <span class="kw">as.character</span>(dux4_de<span class="op">$</span>repName))</a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="kw">ggVennDiagram</span>(venn_data) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7"><span class="st">  </span><span class="kw">scale_fill_gradient</span>(<span class="dt">low=</span><span class="st">&quot;#E1AF00&quot;</span>, <span class="dt">high=</span><span class="st">&quot;#3B9AB2&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-8" data-line-number="8"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;RMSK: differentially expressed elements&quot;</span>)</a></code></pre></div>
<div class="figure"><span id="fig:venn-diagram-HinC-CinC-rmsk"></span>
<img src="gitbook_files/figure-html/venn-diagram-HinC-CinC-rmsk-1.png" alt="Venn diagram of differentially expressed repeat elements in CinC and HinC." width="672" />
<p class="caption">
Figure 6.4: Venn diagram of differentially expressed repeat elements in CinC and HinC.
</p>
</div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="dt">filename=</span><span class="kw">file.path</span>(fig_dir, <span class="st">&quot;CinC_HinC_rmsk_overlapping_vinndiagram.pdf&quot;</span>),</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">       <span class="dt">width=</span><span class="dv">5</span>, <span class="dt">height=</span><span class="dv">3</span>)</a></code></pre></div>
</div>
<div id="repeat-family-activation" class="section level3">
<h3><span class="header-section-number">6.4.2</span> Repeat family activation</h3>
<p>Among 48 families of repeat elements, ERVL subfamilies and LINE1 are over-represented in CinC and HinC trasncriptome: the number of up-regulated repeat elements in the family is higher than expected (by hypergeometric tests).</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">family_res &lt;-<span class="st"> </span><span class="kw">full_join</span>(HinC_rmsk<span class="op">$</span>repFamily_summary, CinC_rmsk<span class="op">$</span>repFamily_summary, </a>
<a class="sourceLine" id="cb14-2" data-line-number="2">                        <span class="dt">by=</span><span class="st">&quot;repFamily&quot;</span>, <span class="dt">suffix=</span><span class="kw">c</span>(<span class="st">&quot;_HinC&quot;</span>, <span class="st">&quot;_CinC&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(repFamily, enrichment_prob_HinC, enrichment_prob_CinC,</a>
<a class="sourceLine" id="cb14-4" data-line-number="4">    up_HinC, up_CinC, enrichment_mu_HinC, enrichment_mu_CinC) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">gather</span>(<span class="dt">key=</span><span class="st">&quot;sample&quot;</span>, <span class="dt">value=</span><span class="st">&quot;pval&quot;</span>, <span class="op">-</span>repFamily, </a>
<a class="sourceLine" id="cb14-6" data-line-number="6">                <span class="op">-</span>up_HinC, <span class="op">-</span>up_CinC, <span class="op">-</span>enrichment_mu_HinC, <span class="op">-</span>enrichment_mu_CinC) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;HinC&quot;</span>, sample), <span class="st">&quot;HinC&quot;</span>, <span class="st">&quot;CinC&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">log10Pval =</span> <span class="dv">-10</span><span class="op">*</span><span class="kw">log10</span>(pval)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">up =</span> <span class="kw">ifelse</span>(sample<span class="op">==</span><span class="st">&quot;HinC&quot;</span>, up_HinC, up_CinC),</a>
<a class="sourceLine" id="cb14-10" data-line-number="10">                <span class="dt">mu =</span> <span class="kw">ifelse</span>(sample<span class="op">==</span><span class="st">&quot;HinC&quot;</span>, enrichment_mu_HinC, enrichment_mu_CinC)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-11" data-line-number="11"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">overrepresented =</span> pval <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>up <span class="op">&gt;</span><span class="st"> </span>mu)  </a>
<a class="sourceLine" id="cb14-12" data-line-number="12"></a>
<a class="sourceLine" id="cb14-13" data-line-number="13">gg &lt;-<span class="st"> </span><span class="kw">ggplot</span>(family_res, <span class="kw">aes</span>(<span class="dt">x=</span>repFamily, <span class="dt">y=</span>sample)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-14" data-line-number="14"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">size=</span>log10Pval, <span class="dt">color=</span>overrepresented, <span class="dt">fill=</span>overrepresented)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-15" data-line-number="15"><span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb14-16" data-line-number="16"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>, <span class="dt">vjust=</span><span class="fl">0.5</span>),</a>
<a class="sourceLine" id="cb14-17" data-line-number="17">        <span class="dt">axis.title.x=</span><span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb14-18" data-line-number="18">        <span class="dt">axis.title.y=</span><span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb14-19" data-line-number="19">        <span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-20" data-line-number="20"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="kw">expression</span>(<span class="st">&quot;Repeat Family GSEA: &quot;</span> <span class="op">~</span><span class="st"> </span><span class="dv">-10</span> <span class="op">*</span><span class="st"> </span>Log[<span class="dv">10</span>] <span class="op">~</span><span class="st"> &quot;(p-value)&quot;</span>),</a>
<a class="sourceLine" id="cb14-21" data-line-number="21">       <span class="dt">size =</span> <span class="kw">expression</span>(<span class="op">-</span><span class="dv">10</span> <span class="op">*</span><span class="st"> </span>Log[<span class="dv">10</span>] <span class="op">~</span><span class="st"> &quot;(p-value)&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-22" data-line-number="22"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">name =</span><span class="st">&quot;over-represented&quot;</span>,  <span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;ivory4&quot;</span>, <span class="st">&quot;firebrick4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-23" data-line-number="23"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">name=</span><span class="st">&quot;over-represented&quot;</span>,  <span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;ivory4&quot;</span>, <span class="st">&quot;firebrick4&quot;</span>))</a>
<a class="sourceLine" id="cb14-24" data-line-number="24">gg</a></code></pre></div>
<div class="figure"><span id="fig:rmsk-family-activation"></span>
<img src="gitbook_files/figure-html/rmsk-family-activation-1.png" alt="GSEA p-values in families of repeat elements. The p-value deteremines whether the number of up-regulated repeat elements in the family is different from expected; the size of dots reflect the scaled p-values, -10log10(p-value); red dots present over-represented families (p-value &lt; 0.1 and the number of up-regulated elements is higher than expected)." width="672" />
<p class="caption">
Figure 6.5: GSEA p-values in families of repeat elements. The p-value deteremines whether the number of up-regulated repeat elements in the family is different from expected; the size of dots reflect the scaled p-values, -10log10(p-value); red dots present over-represented families (p-value &lt; 0.1 and the number of up-regulated elements is higher than expected).
</p>
</div>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
</div>
<div id="repeat-classs-activation" class="section level3">
<h3><span class="header-section-number">6.4.3</span> Repeat classs activation</h3>
<p>Similarily, using the same method above, LTR is determined activated (over-represetend).</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">class_res &lt;-<span class="st"> </span><span class="kw">full_join</span>(HinC_rmsk<span class="op">$</span>repClass_summary, CinC_rmsk<span class="op">$</span>repClass_summary, </a>
<a class="sourceLine" id="cb16-2" data-line-number="2">                        <span class="dt">by=</span><span class="st">&quot;repClass&quot;</span>, <span class="dt">suffix=</span><span class="kw">c</span>(<span class="st">&quot;_HinC&quot;</span>, <span class="st">&quot;_CinC&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(repClass, enrichment_prob_HinC, enrichment_prob_CinC,</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">    up_HinC, up_CinC, enrichment_mu_HinC, enrichment_mu_CinC) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">gather</span>(<span class="dt">key=</span><span class="st">&quot;sample&quot;</span>, <span class="dt">value=</span><span class="st">&quot;pval&quot;</span>, <span class="op">-</span>repClass, </a>
<a class="sourceLine" id="cb16-6" data-line-number="6">                <span class="op">-</span>up_HinC, <span class="op">-</span>up_CinC, <span class="op">-</span>enrichment_mu_HinC, <span class="op">-</span>enrichment_mu_CinC) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb16-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;HinC&quot;</span>, sample), <span class="st">&quot;HinC&quot;</span>, <span class="st">&quot;CinC&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb16-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">log10Pval =</span> <span class="dv">-10</span><span class="op">*</span><span class="kw">log10</span>(pval)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb16-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">up =</span> <span class="kw">ifelse</span>(sample<span class="op">==</span><span class="st">&quot;HinC&quot;</span>, up_HinC, up_CinC),</a>
<a class="sourceLine" id="cb16-10" data-line-number="10">                <span class="dt">mu =</span> <span class="kw">ifelse</span>(sample<span class="op">==</span><span class="st">&quot;HinC&quot;</span>, enrichment_mu_HinC, enrichment_mu_CinC)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb16-11" data-line-number="11"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">overrepresented =</span> pval <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.1</span> <span class="op">&amp;</span><span class="st"> </span>up <span class="op">&gt;</span><span class="st"> </span>mu)                         </a>
<a class="sourceLine" id="cb16-12" data-line-number="12"></a>
<a class="sourceLine" id="cb16-13" data-line-number="13">gg &lt;-<span class="st"> </span><span class="kw">ggplot</span>(class_res, <span class="kw">aes</span>(<span class="dt">x=</span>repClass, <span class="dt">y=</span>sample)) <span class="op">+</span></a>
<a class="sourceLine" id="cb16-14" data-line-number="14"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">size=</span>log10Pval, <span class="dt">color=</span>overrepresented, <span class="dt">fill=</span>overrepresented), <span class="dt">show.legend=</span><span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb16-15" data-line-number="15"><span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb16-16" data-line-number="16"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>, <span class="dt">vjust=</span><span class="fl">0.5</span>),</a>
<a class="sourceLine" id="cb16-17" data-line-number="17">        <span class="dt">axis.title.x=</span><span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb16-18" data-line-number="18">        <span class="dt">axis.title.y=</span><span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb16-19" data-line-number="19">        <span class="dt">legend.position=</span><span class="st">&quot;bottom&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb16-20" data-line-number="20"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="kw">expression</span>(<span class="st">&quot;Repeat Class GSEA: &quot;</span> <span class="op">~</span><span class="st"> </span><span class="dv">-10</span> <span class="op">*</span><span class="st"> </span>Log[<span class="dv">10</span>] <span class="op">~</span><span class="st"> &quot;(p-value)&quot;</span>),</a>
<a class="sourceLine" id="cb16-21" data-line-number="21">       <span class="dt">size =</span> <span class="kw">expression</span>(<span class="op">-</span><span class="dv">10</span> <span class="op">*</span><span class="st"> </span>Log[<span class="dv">10</span>] <span class="op">~</span><span class="st"> &quot;(p-value)&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb16-22" data-line-number="22"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">name =</span><span class="st">&quot;over-represented&quot;</span>,  <span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;ivory4&quot;</span>, <span class="st">&quot;firebrick4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb16-23" data-line-number="23"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">name=</span><span class="st">&quot;over-represented&quot;</span>, <span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;ivory4&quot;</span>, <span class="st">&quot;firebrick4&quot;</span>))</a>
<a class="sourceLine" id="cb16-24" data-line-number="24">gg</a></code></pre></div>
<div class="figure"><span id="fig:rmsk-repClass-activation"></span>
<img src="gitbook_files/figure-html/rmsk-repClass-activation-1.png" alt="GSEA p-values in classes of repeat elements. The p-value deteremines whether the number of up-regulated repeat elements in the family is different from expected; the size of dots reflect the scaled p-values, -10log10(p-value); red dots present over-represented families (p-value &lt; 0.1 and the number of up-regulated elements is higher than expected)." width="672" />
<p class="caption">
Figure 6.6: GSEA p-values in classes of repeat elements. The p-value deteremines whether the number of up-regulated repeat elements in the family is different from expected; the size of dots reflect the scaled p-values, -10log10(p-value); red dots present over-represented families (p-value &lt; 0.1 and the number of up-regulated elements is higher than expected).
</p>
</div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw">pdf</span>(<span class="kw">file.path</span>(fig_dir, <span class="st">&quot;rmsk_repClass_log10Pval.pdf&quot;</span>), <span class="dt">width=</span><span class="fl">4.5</span>, <span class="dt">height=</span><span class="dv">2</span>)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">gg</a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="kw">dev.off</span>()</a></code></pre></div>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
</div>
</div>
<div id="ltr-retroelements-distribution-and-activation" class="section level2">
<h2><span class="header-section-number">6.5</span> LTR retroelements distribution and activation</h2>
<p>LTR retroelements were classified into ERV1, ERVL, ERVL-MaLR, Gypsy, others (non-classified ERV1?, ERVL?, Gypsy?, and LTR) families. ERVL-MaLR constitutes most of the LTR (44%), and next is ERVL (33%) and ERR1 (19%).</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(plyranges))</a>
<a class="sourceLine" id="cb19-2" data-line-number="2">percent &lt;-<span class="st"> </span><span class="cf">function</span>(prob) <span class="kw">paste0</span>(<span class="kw">format</span>(prob, <span class="dt">digit=</span><span class="dv">1</span>), <span class="st">&quot;%&quot;</span>)</a>
<a class="sourceLine" id="cb19-3" data-line-number="3"></a>
<a class="sourceLine" id="cb19-4" data-line-number="4">ltr &lt;-<span class="st"> </span>canFam3.rmsk <span class="op">%&gt;%</span><span class="st"> </span>plyranges<span class="op">::</span><span class="kw">filter</span>(repClass <span class="op">==</span><span class="st"> &quot;LTR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-5" data-line-number="5"><span class="st">  </span>plyranges<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">repFamily=</span><span class="kw">as.character</span>(repFamily)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-6" data-line-number="6"><span class="st">  </span>plyranges<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">repFamily=</span><span class="kw">ifelse</span>(repFamily <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Gypsy?&quot;</span>, <span class="st">&quot;ERVL?&quot;</span>, <span class="st">&quot;ERV1?&quot;</span>, <span class="st">&quot;LTR&quot;</span>), </a>
<a class="sourceLine" id="cb19-7" data-line-number="7">                                 <span class="st">&quot;Others&quot;</span>, repFamily)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-8" data-line-number="8"><span class="st">  </span>plyranges<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">rng_width =</span> <span class="kw">width</span>(.)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-9" data-line-number="9"><span class="st">  </span>plyranges<span class="op">::</span><span class="kw">group_by</span>(repFamily) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-10" data-line-number="10"><span class="st">  </span>plyranges<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">length =</span> <span class="kw">sum</span>(rng_width)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-11" data-line-number="11"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(<span class="kw">desc</span>(length)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-12" data-line-number="12"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">prop =</span> <span class="dv">100</span><span class="op">*</span><span class="st"> </span>(length<span class="op">/</span><span class="kw">sum</span>(length)))  <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-13" data-line-number="13"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">ypos =</span> <span class="kw">cumsum</span>(prop) <span class="op">-</span><span class="st"> </span><span class="fl">0.5</span> <span class="op">*</span><span class="st"> </span>prop) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb19-14" data-line-number="14"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">percent=</span><span class="kw">percent</span>(prop), <span class="dt">label =</span> <span class="kw">paste0</span>(repFamily, <span class="st">&quot; (&quot;</span>, percent, <span class="st">&quot;)&quot;</span>))</a>
<a class="sourceLine" id="cb19-15" data-line-number="15"></a>
<a class="sourceLine" id="cb19-16" data-line-number="16"><span class="co"># Basic piechart</span></a>
<a class="sourceLine" id="cb19-17" data-line-number="17"><span class="kw">ggplot</span>(ltr, <span class="kw">aes</span>(<span class="dt">x=</span><span class="st">&quot;&quot;</span>, <span class="dt">y=</span>prop, <span class="dt">fill=</span>repFamily)) <span class="op">+</span></a>
<a class="sourceLine" id="cb19-18" data-line-number="18"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">width=</span><span class="dv">1</span>, <span class="dt">color=</span><span class="st">&quot;white&quot;</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb19-19" data-line-number="19"><span class="st">  </span><span class="kw">coord_polar</span>(<span class="st">&quot;y&quot;</span>, <span class="dt">start=</span><span class="dv">0</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb19-20" data-line-number="20"><span class="st">  </span><span class="kw">theme_void</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb19-21" data-line-number="21"><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">y=</span>ypos, <span class="dt">x=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="fl">1.2</span>), <span class="dt">label =</span> label),</a>
<a class="sourceLine" id="cb19-22" data-line-number="22">            <span class="dt">color =</span> <span class="st">&quot;white&quot;</span>, <span class="dt">size=</span><span class="dv">4</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb19-23" data-line-number="23"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette=</span><span class="st">&quot;Set1&quot;</span>)</a></code></pre></div>
<div class="figure"><span id="fig:LTR-subfamily-distribution"></span>
<img src="gitbook_files/figure-html/LTR-subfamily-distribution-1.png" alt="Distribution of LTR retroelement family. LTRs are categorized into ERV1, ERVL, ERVL-MaLR, Gypsy and ambibuguos, unclassified Others (ERVL?, Gypsy? ERVL1? and LTR) families. Almost 44% of LTRs belong to MaLR family in canFam3 genome." width="672" />
<p class="caption">
Figure 6.7: Distribution of LTR retroelement family. LTRs are categorized into ERV1, ERVL, ERVL-MaLR, Gypsy and ambibuguos, unclassified Others (ERVL?, Gypsy? ERVL1? and LTR) families. Almost 44% of LTRs belong to MaLR family in canFam3 genome.
</p>
</div>
<p><strong>LTR repeat family activation (ERV1, ERVL, MaLR) by violin plot</strong>: violin plot of LTR retroelements expression in each family; each dot represents a repeat element; red dots present up-regulated elements; diamond shape is the median expression of the family.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">LTR &lt;-<span class="st"> </span>comb <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(repClass_HinC <span class="op">==</span><span class="st"> &quot;LTR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">repFamily_HinC =</span> <span class="kw">factor</span>(repFamily_HinC)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(repName, repFamily_HinC, repClass_HinC, </a>
<a class="sourceLine" id="cb20-4" data-line-number="4">                log2FoldChange_HinC, log2FoldChange_CinC, padj_HinC, padj_CinC) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">repFamily=</span>repFamily_HinC, <span class="dt">repClass=</span>repClass_HinC) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-6" data-line-number="6"><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">gather</span>(<span class="dt">key=</span><span class="st">&quot;sample&quot;</span>, <span class="dt">value=</span><span class="st">&quot;log2FC&quot;</span>, <span class="op">-</span>repName, <span class="op">-</span>repFamily, <span class="op">-</span>repClass, <span class="op">-</span>padj_HinC, <span class="op">-</span>padj_CinC) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">sample =</span> <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;HinC&quot;</span>, sample), <span class="st">&quot;HinC&quot;</span>, <span class="st">&quot;CinC&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">padj =</span> <span class="kw">ifelse</span>(sample <span class="op">==</span><span class="st"> &quot;HinC&quot;</span>, padj_HinC, padj_CinC)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">DE_status =</span> padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># fix repFamily: put *? to others</span></a>
<a class="sourceLine" id="cb20-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">repFamily=</span><span class="kw">as.character</span>(repFamily)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-11" data-line-number="11"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">repFamily =</span> </a>
<a class="sourceLine" id="cb20-12" data-line-number="12">                <span class="kw">replace</span>(repFamily, <span class="kw">which</span>(repFamily <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ERV1?&quot;</span>, <span class="st">&quot;ERVL?&quot;</span>, <span class="st">&quot;Gypsy?&quot;</span>, <span class="st">&quot;LTR&quot;</span>)), <span class="st">&quot;Others&quot;</span>))</a>
<a class="sourceLine" id="cb20-13" data-line-number="13"></a>
<a class="sourceLine" id="cb20-14" data-line-number="14">gg &lt;-<span class="st"> </span><span class="kw">ggplot</span>(LTR, <span class="kw">aes</span>(<span class="dt">x=</span>repFamily, <span class="dt">y=</span>log2FC)) <span class="op">+</span></a>
<a class="sourceLine" id="cb20-15" data-line-number="15"><span class="st">  </span><span class="co">#geom_boxplot(width=0.5, outlier.shape=NA) +</span></a>
<a class="sourceLine" id="cb20-16" data-line-number="16"><span class="st">  </span><span class="kw">geom_violin</span>(<span class="dt">width=</span><span class="fl">0.7</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb20-17" data-line-number="17"><span class="st">  </span><span class="co">#geom_jitter(size=0.6, alpha=0.7, aes(color=DE_status), show.legend=FALSE) +</span></a>
<a class="sourceLine" id="cb20-18" data-line-number="18"><span class="st">  </span><span class="kw">geom_dotplot</span>(<span class="dt">binaxis=</span><span class="st">&#39;y&#39;</span>, <span class="dt">stackdir=</span><span class="st">&#39;center&#39;</span>, <span class="dt">dotsize=</span><span class="fl">0.5</span>, </a>
<a class="sourceLine" id="cb20-19" data-line-number="19">               <span class="kw">aes</span>(<span class="dt">color=</span>DE_status, <span class="dt">fill=</span>DE_status), </a>
<a class="sourceLine" id="cb20-20" data-line-number="20">               <span class="dt">alpha=</span><span class="fl">0.9</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb20-21" data-line-number="21"><span class="st">  </span><span class="kw">stat_summary</span>(<span class="dt">fun.y=</span>mean, <span class="dt">geom=</span><span class="st">&quot;point&quot;</span>, <span class="dt">shape=</span><span class="dv">23</span>, <span class="dt">size=</span><span class="dv">2</span>, <span class="dt">color=</span><span class="st">&quot;red&quot;</span>) <span class="op">+</span><span class="st">               </span></a>
<a class="sourceLine" id="cb20-22" data-line-number="22"><span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb20-23" data-line-number="23"><span class="st">  </span><span class="kw">facet_wrap</span>( <span class="op">~</span><span class="st"> </span>sample, <span class="dt">nrow=</span><span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb20-24" data-line-number="24"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;ivory4&quot;</span>, <span class="st">&quot;firebrick4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb20-25" data-line-number="25"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;ivory4&quot;</span>, <span class="st">&quot;firebrick4&quot;</span>))  <span class="op">+</span></a>
<a class="sourceLine" id="cb20-26" data-line-number="26"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;Repeat families in the LTR class&quot;</span>, <span class="dt">x=</span><span class="st">&quot;family&quot;</span>)</a>
<a class="sourceLine" id="cb20-27" data-line-number="27">gg</a></code></pre></div>
<div class="figure"><span id="fig:rmsk-LTR-violinplot"></span>
<img src="gitbook_files/figure-html/rmsk-LTR-violinplot-1.png" alt="Violin plot of LTR retroelement expression in families of LTR. Each dot represents a repeat element; red dots present up-regulated elements; diamond shape is the median expression of the family." width="672" />
<p class="caption">
Figure 6.8: Violin plot of LTR retroelement expression in families of LTR. Each dot represents a repeat element; red dots present up-regulated elements; diamond shape is the median expression of the family.
</p>
</div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw">pdf</span>(<span class="kw">file.path</span>(fig_dir, <span class="st">&quot;rmsk_repFamily_violinplot.pdf&quot;</span>), <span class="dt">width=</span><span class="dv">4</span>, <span class="dt">height=</span><span class="dv">6</span>)</a>
<a class="sourceLine" id="cb21-2" data-line-number="2">gg</a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="kw">dev.off</span>()</a></code></pre></div>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
<p><strong>Frequency of up-regulated retroelements in LTR Family:</strong> DUX4 positively affects ~50% of retroelements in ERV1, ERVL and ERVL-MaLR families; DUXC affects 18% ~ 25% in these families.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">freq &lt;-<span class="st"> </span>LTR <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample, repFamily) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-2" data-line-number="2"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">n=</span>dplyr<span class="op">::</span><span class="kw">n</span>(), <span class="dt">n_up=</span><span class="kw">sum</span>(DE_status, <span class="dt">na.rm=</span><span class="ot">TRUE</span>), <span class="dt">prob=</span>n_up<span class="op">/</span>n)</a>
<a class="sourceLine" id="cb23-3" data-line-number="3"><span class="kw">ggplot</span>(freq, <span class="kw">aes</span>(<span class="dt">x=</span>sample, <span class="dt">y=</span>prob, <span class="dt">fill=</span>repFamily) )<span class="op">+</span></a>
<a class="sourceLine" id="cb23-4" data-line-number="4"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb23-5" data-line-number="5"><span class="st">  </span><span class="kw">facet_wrap</span>( <span class="op">~</span><span class="st"> </span>repFamily, <span class="dt">nrow=</span><span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb23-6" data-line-number="6"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette=</span><span class="st">&quot;Set1&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb23-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span><span class="kw">format</span>(prob, <span class="dt">digit=</span><span class="dv">2</span>)), <span class="dt">vjust=</span><span class="op">-</span><span class="fl">0.3</span>, <span class="dt">size=</span><span class="fl">3.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb23-8" data-line-number="8"><span class="st">  </span><span class="kw">theme_minimal</span>()</a></code></pre></div>
<div class="figure"><span id="fig:LTR-up-regulated-frequency"></span>
<img src="gitbook_files/figure-html/LTR-up-regulated-frequency-1.png" alt="Percentage of activation of LTRs in each familiy." width="672" />
<p class="caption">
Figure 6.9: Percentage of activation of LTRs in each familiy.
</p>
</div>
</div>
<div id="peaks-distribution-in-class-of-repeat-elements" class="section level2">
<h2><span class="header-section-number">6.6</span> Peaks distribution in class of repeat elements</h2>
<p>The peaks distribution suggests that both DUX4 and DUXC enriched LTR subfamilies. Can we show that activating LTR leads to re-programing nearby gene’s structure, like we did at Whidden’s paper?</p>
<p><strong>repeat class distribution in the canine geonome</strong>
The code below considered the repeat class distribution on standard chromsomes in canFam3 genome.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="co"># only keep the starndard chromosome</span></a>
<a class="sourceLine" id="cb24-2" data-line-number="2">canFam3.rmsk &lt;-<span class="st"> </span><span class="kw">keepStandardChromosomes</span>(canFam3.rmsk, <span class="dt">pruning.mode =</span> <span class="st">&quot;tidy&quot;</span>)</a>
<a class="sourceLine" id="cb24-3" data-line-number="3">seq_info &lt;-<span class="st"> </span><span class="kw">seqinfo</span>(<span class="kw">keepStandardChromosomes</span>(TxDb.Cfamiliaris.UCSC.canFam3.ensGene))</a>
<a class="sourceLine" id="cb24-4" data-line-number="4"></a>
<a class="sourceLine" id="cb24-5" data-line-number="5">genome_length &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">seqlengths</span>(seq_info))</a>
<a class="sourceLine" id="cb24-6" data-line-number="6">nonrepeat_length &lt;-<span class="st"> </span>genome_length <span class="op">-</span><span class="st"> </span><span class="kw">sum</span>(<span class="kw">width</span>(canFam3.rmsk))</a>
<a class="sourceLine" id="cb24-7" data-line-number="7">repeat_class &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;LTR&quot;</span>, <span class="st">&quot;SINE&quot;</span>, <span class="st">&quot;LINE&quot;</span>, <span class="st">&quot;DNA&quot;</span>, <span class="st">&quot;Simple_repeat&quot;</span>)</a>
<a class="sourceLine" id="cb24-8" data-line-number="8">canFam3.rmsk<span class="op">$</span>instance_width &lt;-<span class="st"> </span><span class="kw">width</span>(canFam3.rmsk)</a>
<a class="sourceLine" id="cb24-9" data-line-number="9">genome_repClass_freq &lt;-<span class="st"> </span>canFam3.rmsk <span class="op">%&gt;%</span><span class="st"> </span>plyranges<span class="op">::</span><span class="kw">group_by</span>(repClass) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb24-10" data-line-number="10"><span class="st">  </span><span class="kw">summarize</span>(<span class="dt">total_length =</span> <span class="kw">sum</span>(instance_width)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb24-11" data-line-number="11"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">tidy_repClass =</span> <span class="kw">as.character</span>(repClass)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb24-12" data-line-number="12"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">tidy_repClass =</span> <span class="kw">replace</span>(tidy_repClass, </a>
<a class="sourceLine" id="cb24-13" data-line-number="13">                                        <span class="kw">which</span>(<span class="op">!</span>tidy_repClass <span class="op">%in%</span><span class="st"> </span>repeat_class), <span class="st">&quot;Other repeats&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb24-14" data-line-number="14"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">add_row</span>(<span class="dt">repClass=</span><span class="st">&quot;Non-repetitive&quot;</span>, <span class="dt">total_length=</span>nonrepeat_length, </a>
<a class="sourceLine" id="cb24-15" data-line-number="15">                  <span class="dt">tidy_repClass =</span> <span class="st">&quot;Non-repetitive&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb24-16" data-line-number="16"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">tidy_repClass =</span> <span class="kw">factor</span>(tidy_repClass, </a>
<a class="sourceLine" id="cb24-17" data-line-number="17">                                           <span class="dt">levels=</span><span class="kw">c</span>(repeat_class, <span class="st">&quot;Other repeats&quot;</span>, <span class="st">&quot;Non-repetitive&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb24-18" data-line-number="18"><span class="st">  </span><span class="kw">group_by</span>(tidy_repClass) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb24-19" data-line-number="19"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">total_length =</span> <span class="kw">sum</span>(total_length)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb24-20" data-line-number="20"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">frequency =</span> total_length<span class="op">/</span>genome_length <span class="op">*</span><span class="st"> </span><span class="dv">100</span>)</a>
<a class="sourceLine" id="cb24-21" data-line-number="21">genome_repClass_freq</a></code></pre></div>
<pre><code>## # A tibble: 7 x 3
##   tidy_repClass  total_length frequency
##   &lt;fct&gt;                 &lt;dbl&gt;     &lt;dbl&gt;
## 1 LTR               114911883     4.94 
## 2 SINE              251299704    10.8  
## 3 LINE              493299304    21.2  
## 4 DNA                65806988     2.83 
## 5 Simple_repeat      26825828     1.15 
## 6 Other repeats      22743601     0.977
## 7 Non-repetitive   1352763403    58.1</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="co"># construction</span></a></code></pre></div>
<p><strong>Distribution of peaks in classes of repeat elements</strong></p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">repeat_class &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;LTR&quot;</span>, <span class="st">&quot;SINE&quot;</span>, <span class="st">&quot;LINE&quot;</span>, <span class="st">&quot;DNA&quot;</span>, <span class="st">&quot;Simple_repeat&quot;</span>, <span class="st">&quot;Non-repetitive&quot;</span>)</a>
<a class="sourceLine" id="cb27-2" data-line-number="2">rep_summary &lt;-<span class="st"> </span><span class="kw">map</span>(peaks_list, <span class="cf">function</span>(rng) {</a>
<a class="sourceLine" id="cb27-3" data-line-number="3">    total_peak &lt;-<span class="st"> </span><span class="kw">length</span>(rng)</a>
<a class="sourceLine" id="cb27-4" data-line-number="4">    df &lt;-<span class="st"> </span><span class="kw">as</span>(<span class="kw">mcols</span>(rng), <span class="st">&quot;data.frame&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="co">#tidy repClass</span></a>
<a class="sourceLine" id="cb27-5" data-line-number="5"><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">tidy_repClass =</span> <span class="kw">as.character</span>(repClass)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-6" data-line-number="6"><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">tidy_repClass =</span> <span class="kw">replace</span>(tidy_repClass, </a>
<a class="sourceLine" id="cb27-7" data-line-number="7">                                            <span class="kw">which</span>(<span class="kw">is.na</span>(tidy_repClass)), <span class="st">&quot;Non-repetitive&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-8" data-line-number="8"><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">tidy_repClass =</span> <span class="kw">replace</span>(tidy_repClass, </a>
<a class="sourceLine" id="cb27-9" data-line-number="9">                                            <span class="kw">which</span>(<span class="op">!</span>tidy_repClass <span class="op">%in%</span><span class="st"> </span>repeat_class), <span class="st">&quot;Other repeats&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-10" data-line-number="10"><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">tidy_repClass =</span> <span class="kw">factor</span>(tidy_repClass, </a>
<a class="sourceLine" id="cb27-11" data-line-number="11">                                           <span class="dt">levels=</span><span class="kw">c</span>(repeat_class[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>], <span class="st">&quot;Other repeats&quot;</span>, <span class="st">&quot;Non-repetitive&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-12" data-line-number="12"><span class="st">      </span><span class="kw">group_by</span>(tidy_repClass) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-13" data-line-number="13"><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">frequency =</span> <span class="dv">100</span> <span class="op">*</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">n</span>()<span class="op">/</span><span class="kw">length</span>(rng))</a>
<a class="sourceLine" id="cb27-14" data-line-number="14">}) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">bind_rows</span>(<span class="dt">.id=</span><span class="st">&quot;sample_name&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-15" data-line-number="15"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">trans_factor =</span> <span class="kw">sapply</span>(<span class="kw">strsplit</span>(sample_name, <span class="st">&quot;_&quot;</span>, <span class="dt">fixed=</span><span class="ot">TRUE</span>), <span class="st">&quot;[[&quot;</span>, <span class="dv">3</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-16" data-line-number="16"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">clone_type =</span> <span class="kw">sapply</span>(<span class="kw">strsplit</span>(sample_name, <span class="st">&quot;_&quot;</span>, <span class="dt">fixed=</span><span class="ot">TRUE</span>), <span class="st">&quot;[[&quot;</span>, <span class="dv">4</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-17" data-line-number="17"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Sample =</span> <span class="kw">paste0</span>(trans_factor, <span class="st">&quot;_&quot;</span>, clone_type)) </a>
<a class="sourceLine" id="cb27-18" data-line-number="18"></a>
<a class="sourceLine" id="cb27-19" data-line-number="19">tidy_genome_repClass_freq &lt;-<span class="st"> </span>genome_repClass_freq <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-20" data-line-number="20"><span class="st">  </span><span class="kw">add_column</span>(<span class="dt">sample_name =</span> <span class="st">&quot;whole_genome&quot;</span>, <span class="dt">.before=</span><span class="st">&quot;tidy_repClass&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-21" data-line-number="21"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>total_length) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-22" data-line-number="22"><span class="st">  </span><span class="kw">add_column</span>(<span class="dt">trans_factor =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">clone_type=</span><span class="st">&quot;none&quot;</span>, <span class="dt">Sample=</span><span class="st">&quot;Genome&quot;</span>)</a>
<a class="sourceLine" id="cb27-23" data-line-number="23">  </a>
<a class="sourceLine" id="cb27-24" data-line-number="24"><span class="co">## remove poly</span></a>
<a class="sourceLine" id="cb27-25" data-line-number="25">rep_summary &lt;-<span class="st"> </span>rep_summary <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">bind_rows</span>(tidy_genome_repClass_freq) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-26" data-line-number="26"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span>(Sample <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;hDUX4_poly&quot;</span>, <span class="st">&quot;CCH_poly&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-27" data-line-number="27"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Sample=</span><span class="kw">factor</span>(Sample, </a>
<a class="sourceLine" id="cb27-28" data-line-number="28">                              <span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;Genome&quot;</span>,<span class="st">&quot;hDUX4_mono&quot;</span>, <span class="st">&quot;CCH_mono&quot;</span>, <span class="st">&quot;CALTh_poly&quot;</span>)))</a>
<a class="sourceLine" id="cb27-29" data-line-number="29"></a>
<a class="sourceLine" id="cb27-30" data-line-number="30">col &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">wes_palette</span>(<span class="dt">n=</span><span class="dv">7</span>, <span class="dt">name=</span><span class="st">&quot;Darjeeling1&quot;</span>, <span class="dt">type=</span><span class="st">&quot;continuous&quot;</span>))</a>
<a class="sourceLine" id="cb27-31" data-line-number="31"></a>
<a class="sourceLine" id="cb27-32" data-line-number="32">gg &lt;-<span class="st"> </span><span class="kw">ggplot</span>(rep_summary, <span class="kw">aes</span>(<span class="dt">x=</span>Sample, <span class="dt">y=</span>frequency, <span class="dt">fill=</span>tidy_repClass)) <span class="op">+</span></a>
<a class="sourceLine" id="cb27-33" data-line-number="33"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb27-34" data-line-number="34"><span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb27-35" data-line-number="35"><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span><span class="kw">round</span>(frequency)), <span class="dt">position =</span> <span class="kw">position_stack</span>(<span class="dt">vjust =</span> <span class="fl">0.5</span>), </a>
<a class="sourceLine" id="cb27-36" data-line-number="36">            <span class="dt">color=</span><span class="st">&quot;gray10&quot;</span>, <span class="dt">size=</span><span class="fl">3.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb27-37" data-line-number="37"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span>col) <span class="op">+</span></a>
<a class="sourceLine" id="cb27-38" data-line-number="38"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">45</span>, <span class="dt">hjust =</span> <span class="dv">1</span>), </a>
<a class="sourceLine" id="cb27-39" data-line-number="39">        <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>())   <span class="op">+</span></a>
<a class="sourceLine" id="cb27-40" data-line-number="40"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">y=</span><span class="st">&quot;Frequency (%)&quot;</span>, <span class="dt">fill=</span><span class="st">&quot;Repeat class&quot;</span>)        </a>
<a class="sourceLine" id="cb27-41" data-line-number="41">gg</a></code></pre></div>
<div class="figure"><span id="fig:peaks-distribution-repClass"></span>
<img src="gitbook_files/figure-html/peaks-distribution-repClass-1.png" alt="Distribution of peaks in classes of repeat elements." width="672" />
<p class="caption">
Figure 6.10: Distribution of peaks in classes of repeat elements.
</p>
</div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="co"># take out the polyclonal cell lines, except CALTh_poly</span></a>
<a class="sourceLine" id="cb28-2" data-line-number="2"><span class="kw">pdf</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;figures&quot;</span>, <span class="st">&quot;rmsk_peaks_repClass_freq_bar.pdf&quot;</span>), <span class="dt">height=</span><span class="dv">6</span>, <span class="dt">width=</span><span class="fl">3.5</span>)</a>
<a class="sourceLine" id="cb28-3" data-line-number="3">gg</a>
<a class="sourceLine" id="cb28-4" data-line-number="4"><span class="kw">dev.off</span>()</a></code></pre></div>
<pre><code>## quartz_off_screen 
##                 2</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="DUXC-early-embryo.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="LTR-activation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["gitbook.pdf", "gitbook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
