# ChIP-seq for the canine model {#chipseq}
We performed ChIP-seq experiments to determind the transcription factor DUX4, DUXC and DUXC-ALT's binding sites and motifs in the canine genome. Below we described the sample information and ChIP-seq preprocessing pipeline, peaks calling, and motifs discovery. 

```{r load-lib-dataset-2}
suppressPackageStartupMessages(library(TxDb.Cfamiliaris.UCSC.canFam3.ensGene))
txdb <- TxDb.Cfamiliaris.UCSC.canFam3.ensGene
suppressPackageStartupMessages(library(canFam3.rmsk))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(wesanderson))
# eventially change to loading from github
pkg_dir <- "~/CompBio/canFam3.DuxFamily"
load(file.path(pkg_dir, "data", "peaks_list.rda")) # peaks list
load(file.path(pkg_dir, "data", "chipseq_si.rda")) # sample information
```

## ChIP-seq samples 
We mainly used the monoclonal antibodies treated samples for our manuscript, except DUXC-ALT. 

```{r sample-attributes, echo=FALSE}
as.data.frame(si) %>% dplyr::select(sample_name, lib_size) %>%
  add_column(transcription_factor = c("luciferas", "DUX4", "DUXC", "DUXC-ALT",
                                      "luciferas", "DUX4", "DUXC"),
             cell_type=c(rep("polyclonal", 4), rep("monoclonal", 3)))
```

## Analysis workflow
The pre-processing pipeline and peak calling of the ChIP-seq for the canine model follows a standard pipeline:

1. __Filter unqualified reads__: based on casava filter, filtered out reads whose filter flag, indicated in the identifier, equals to "Y".
2. __Trim adapter__: `AGATCGGAAGAGCACACGTCAGAACTCCAGTCACGCACTAATCTCGTATG`  
3. __Quality control__: quality control by _fastqc_ to confirm that there is no more adapter contemination  
4. __Alignement__: align read to the _canFam3_ genome build by _BWA-0.7.10_ (`/scripts/do_bwa.sh `)
5. __Peak calling__: peak calling made by macs2 (MACS2) with Luciferase monoclonal samples as negative controls (`Sample5_ImmCnMb_luc_mono.bam`, monoclonal cell line) 
```
macs2 callpeak -t $sampleName.bam -c $cnt -f BAM \
  --outdir $macsDir -g 2.42e9 -n $sampleName \
  -B -q 0.01
```
6. __Peak annotation__: annotate peaks using Bioconductor **ChIPseeker** [@ChIPseeker] package along with a customed `TxDb` (`TxDb.Cfamiliaris.UCSC.canFam3.ensGene`) annotation package derived from the Ensembl's gtf file in the canfam3 genome build.     
7. Prepare peaks FASTA files: select top 1000 peaks and flank up/downstream 50 bps around the peak summits (`/scripts/peaks.R`)
8. __Motif discovery__: prepare peaks FASTA files by selecting top 1000 peaks and flank up/downstream 50 bps around the peak summits (`/scripts/peaks.R`); discovery motifs by using MEME 5.1.1 (`/scripts/peaks.R`)
```
meme Sample6_ImmCnMb_hDUX4_mono_peaksSeq_100.fa \ 
  -dna -oc . -nostatus -time 18000 -mod zoops \ 
  -nmotifs 3 -minw 6 -maxw 11 -objfun classic \ 
  -revcomp -markov_order 0
```

## Peaks and functional distribution
The peaks file is located at  `../data/peaks_list.rda`: It is a `list` of `GRanges` instances containing peaks coordinates, summits, q-values, annotaion, overlapping staus with repeat elements, and orthologs. For example:
```{r peak-example}
peaks_list[["Sample7_ImmCnMb_CCH_mono"]]
```

### Number of peaks
```{r peaks_number}
n_peaks <- t(map_dfr(peaks_list, length)) %>% as.data.frame() %>%
  rownames_to_column(var="sample_name") %>%
  dplyr::rename(num_peaks=V1) %>%
  add_column(transcription_factor = c("DUX4", "DUXC", "DUXC_ALT",
                                       "DUX4", "DUXC"))
n_peaks[c(5, 4, 3), ]
```

### Functional discribution
Visualization of the genomic annotation by peaks' functional location. 
```{r simplify_annotation}
.simplify_annotation <- function(peaks_gr) {
  # Promoter, 5' UTR, Exon, Intron, 3' UTR, Distal Intergenic, 
  data.frame(detail = peaks_gr$annotation) %>%
    add_column(simplify="unknown") %>%
    dplyr::mutate(simplify = case_when(
      grepl("Promoter", detail) ~ "Promoter",
      grepl("5' UTR", detail) ~ "5' UTR",
      grepl("Exon", detail) ~ "Exon",
      grepl("Intron", detail) ~ "Intron",
      grepl("3' UTR", detail) ~ "3' UTR",
      grepl("Downstream", detail) ~ "Downstream",
      grepl("Distal Intergenic", detail) ~ "Distal Intergenic")
    ) %>%
    dplyr::mutate(simplify=factor(simplify, levels=c("Promoter", "5' UTR",
                                                     "Exon",
                                                     "Intron", "3' UTR",
                                                     "Downstream",
                                                     "Distal Intergenic")))
}
```
```{r bar-plot-functional-distribution, fig.cap="Functional distribution of peaks.", fig.width=3, fig.height=4, fig.align="center"}
CCH_df <- .simplify_annotation(peaks_list[["Sample7_ImmCnMb_CCH_mono"]]) %>%
  count(simplify) %>% 
  dplyr::mutate(freq=n/sum(n), transcription_factor="DUXC") 
  
hDUX4_df <- .simplify_annotation(peaks_list[["Sample6_ImmCnMb_hDUX4_mono"]]) %>%
  count(simplify) %>%
  dplyr::mutate(freq=n/sum(n), transcription_factor="DUX4") 

CALTH_df <- .simplify_annotation(peaks_list[["Sample4_ImmCnMb_CALTh_poly"]]) %>%
  count(simplify) %>%
  dplyr::mutate(freq=n/sum(n), transcription_factor="DUXC-ALT") 

df <- CCH_df %>% bind_rows(hDUX4_df) %>% bind_rows(CALTH_df) %>%
  dplyr::mutate(transcription_factor=factor(transcription_factor, 
                                            levels=c("DUX4", "DUXC",
                                                     "DUXC-ALT"))) %>%
  dplyr::mutate(percentage=freq * 100)

ggplot(df, aes(x=transcription_factor, y=percentage, fill=simplify)) +
   geom_bar(stat="identity") +
  theme_minimal() +
  labs(y="Percentage (%)") +
  theme(axis.title.x=element_blank(), legend.title = element_blank(),
        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  scale_fill_manual(values = wes_palette("Darjeeling2", n=7,
                                         type="continuous"))
  
ggsave(file.path(pkg_dir, "figures", "chipseq-peaks-functional-distribution.pdf"), width=3, height=4)
```

## Functinal enrichement analysis

```{r enriched-pathway}
library(clusterProfiler)
library(ReactomePA)
enrichPathWay()
```
## Overlap of peaks 

### DUXC and DUX4
```{r vennplot-DUXC-DUX4}
genes= lapply(peaks_list[c(4, 5)], function(gr) gr$geneId)
names(genes) <- c("DUX4", "DUXC")
vennplot(genes)
```

### DUXC and DUXC-ALT
```{r vennplot-DUXC-DUX4}
genes= lapply(peaks_list[c(3, 5)], function(gr) gr$geneId)
names(genes) <- c("DUXC-ALT", "DUXC")
vennplot(genes)
```

About 55% of DUXC (monoclone) peaks overlap with hDUX4 peaks.
```{r overlap-hDUX4-CCH}
ov <- findOverlaps(peaks_list[["Sample7_ImmCnMb_CCH_mono"]],
                   peaks_list[["Sample6_ImmCnMb_hDUX4_mono"]])
freq <- length(unique(queryHits(ov)))/length(peaks_list[["Sample7_ImmCnMb_CCH_mono"]])
freq
```
__Question:__ Do DUXC's A or G peaks overlap with hDUX4 peaks?
About 60% and 50% of A-peaks and G-peaks overlap with hDUX4 peaks, respectively.

```{r A-G-peaks-ov-hDUX4, eval=FALSE, echo=FALSE}
# 1. identify peaks has motifs
# 2. among those peaks, how many A and G peaks overlap with hDUX4 peaks?
source("~/tapscott/ChIP-Seq/R_scripts/getMotifs.R")
duxc_summits <- peaks_list[["Sample7_ImmCnMb_CCH_mono"]]
start(duxc_summits) <- duxc_summits$abs_summit - 50
end(duxc_summits) <- duxc_summits$abs_summit + 50
duxc_summits$seq <- BSgenome::getSeq(BSgenome.Cfamiliaris.UCSC.canFam3, 
                                    duxc_summits)
# which has TAATC and TAGTC
duxc_summits$A_peaks <- vcountPattern("TAATC", duxc_summits$seq) > 0 |
  vcountPattern(reverseComplement(DNAString("TAATC")), duxc_summits$seq) > 0
duxc_summits$G_peaks <- vcountPattern("TAGTC", duxc_summits$seq) > 0 |
  vcountPattern(reverseComplement(DNAString("TAGTC")), duxc_summits$seq) > 0
query <- duxc_summits %>% plyranges::filter(A_peaks)
ov_A <- findOverlaps(query, peaks_list[["Sample6_ImmCnMb_hDUX4_mono"]])
length(unique(queryHits(ov_A)))/length(query)

query <- duxc_summits %>% plyranges::filter(G_peaks)
ov_G <- findOverlaps(query, peaks_list[["Sample6_ImmCnMb_hDUX4_mono"]])
length(unique(queryHits(ov_G)))/length(query)


```

### Functional distribution
1. compare peaks location between CCH conditions
```{r share_peaks_CCH, eval=FALSE}
CCH_mono <- peaks_list[["Sample7_ImmCnMb_CCH_mono"]]
CCH_poly <- peaks_list[["Sample3_ImmCnMb_CCH_poly"]]
left <- join_overlap_inner(CCH_mono, CCH_poly) 
length(left)
right <- join_overlap_inner(CCH_poly, CCH_mono)
length(right)
```

2. compare peak locations between HinC condition
```{r share_peaks_hDUX4, eval=FALSE}
hDUX4_mono <- peaks_list[["Sample6_ImmCnMb_hDUX4_mono"]]
hDUX4_poly <- peaks_list[["Sample2_ImmCnMb_hDUX4_poly"]]
left <- join_overlap_inner(hDUX4_mono, hDUX4_poly) 
length(left)
right <- join_overlap_inner(hDUX4_poly, hDUX4_mono)
length(right)
```
3. Compare peak location between CCH and CALT conditions


## Motifs analysis NEED UPDATE
__Question:__ The CCH motif T(A/G)ATCCAATCA: does the position, A/G, associated with induced genes??

The A motif does not associated with induced genes.

```{r CCH_motif, echo=FALSE, eval=FALSE}
# run this on rhino01
setwd("~/tapscott/ChIP-Seq/canFam3.ImmCnMb.dux/MEME_5.1.1/Sample7_ImmCnMb_CCH_mono")
library(gdata)
library(BSgenome.Cfamiliaris.UCSC.canFam3)

block = read.table("motif1_block_format.txt", skip=4, stringsAsFactors=FALSE) %>%
  dplyr::rename(peak_id = V1, motif_start =V3, motif_seq=V4) %>%
  dplyr::mutate(motif_start=as.integer(str_remove(motif_start, "[)]")))
# all have unique peak (id)
block_G <- block[startsWith(block$motif_seq, "TGA"), ]
block_A <- block[startsWith(block$motif_seq, "TAA"), ]
peaks <- peaks_list[["Sample7_ImmCnMb_CCH_mono"]]
source("~/tapscott/ChIP-Seq/R_scripts/getMotifs.R")
# get top, order peaks (as described in peaks.R)
summits <- .filter_peaks(peaks, n=1000)
start(summits) <- end(summits) <- summits$abs_summit
top_peaks <- .aroundSummit(summits, L=100) ## 50 around the summit
top_peaks_seq <- BSgenome::getSeq(BSgenome.Cfamiliaris.UCSC.canFam3, top_peaks)
# top peaks of TAA
top_peaks_A <- top_peaks[block_A$peak_id]
top_peaks_G <- top_peaks[block_G$peak_id]
# What are the induced genes? up-regulated by DUXC? find it! how many overlap with the top_peaks
# findOverlaps(top_peaks_A, up-regulated of DUXC)
library(DESeq2)
load(file.path("~/tapscott/RNA-Seq/canFam3.DuxFamily/data", "CinC.ens.dds.rda"))
CinC_res <- results(CinC.ens.dds, lfcThreshold=1, alpha=0.05)
CinC_sig <- as(CinC_res, "data.frame") %>% 
  dplyr::filter(padj < 0.05, log2FoldChange > 0)
a <- subsetByOverlaps(top_peaks_A, CinC.ens.dds[rownames(CinC_sig)], ignore.strand=TRUE)

# get the promoter of the CinC up-regulated genes, instead of using the exons ...
library(TxDb.Cfamiliaris.UCSC.canFam3.ensGene)
genes <- genes(TxDb.Cfamiliaris.UCSC.canFam3.ensGene)
findOverlaps(top_peaks_A, promoters(genes[rownames(CinC_sig)], 
                                         upstream=3000, downstream=200), 
                                         ignore.strand=TRUE)
findOverlaps(top_peaks_A, genes[rownames(CinC_sig)], ignore.strand=TRUE)

findOverlaps(top_peaks_G, promoters(genes[rownames(CinC_sig)], 
                                         upstream=3000, downstream=200), 
                                         ignore.strand=TRUE)
findOverlaps(top_peaks_G, genes[rownames(CinC_sig)], ignore.strand=TRUE)

## double check that the order of peak.fa is the same as the peaks I made here:
# sample 7 CCH mono fa: ~//tapscott/ChIP-Seq/canFam3.ImmCnMb.dux/MEME/Sample7_ImmCnMb_CCH_mono_peaksSeq_100.fa
library(ShortRead)
for (i in 1:nrow(block_A)) {
  print(block_A[i, "peak_id"])
  print(subseq(top_peaks_seq[block_A[i, "peak_id"]], start=block_A[i, "motif_start"], width=11))
}

table(top_peaks_G$repFamily)
table(top_peaks_A$repFamily)
# A associated with ERVL-MaLR and G with ERVL
```


```{r gap-in-chr17-telemere, eval=FALSE, echo=FALSE}
gap_canFam4 <- read_delim(file.path(pkg_dir, "extdata", "gap_canFam4.txt"),
                delim="\t", col_name=FALSE) %>% 
  dplyr::filter(X2 == "chr17")
gap_canFam3 <- read_delim(file.path(pkg_dir, "extdata", "gap_canFam3.txt"),
                delim="\t", col_name=FALSE) %>% 
  dplyr::filter(X2 == "chr17")
```