# Non-LTR retrotrasposons and satellites {#non-LTR}

This chapter investigated the satellites and non-LTR retrotransposons transcriptional activites. The non-LTR retrotransposons include short interspersed nuclear elements (SINEs) and long interspersed nuclear elements (LINEs).

__Summary: non-LTR tretrotrasponses and satellites activiation in canine__  
Studies ([@shadle2019] and [@whidden2017]) have shown that in human myoblast, DUX4 enriched HSATII and families of non-LTR RNA transcripts, such as Alu family of SINE and LINE1 (L1). Therefore, we were interested in examing the conservation of the non-LTR activation by DUX4 in our canine model. Our analysis showed that DUX4 up-regulated 53% of L1s (52/97) and 28.5% of DUX4 binding sites overlap with L1s. Given that L1s constitute 17% of the canine genome, DUX4, whether directly interacted with L1s or not, induced the L1 family's enrichment in canine myoblasts. On the other hand, the enrichment of L1s by DUXC was rather unimpressive - only 16% of L1s (16/97) exhibited differential expression, and 15.6% DUXC peaks overlap with L1s (figure \@ref(fig:LINE-up-regulation-freq)).  

Unlike human SINEs derived from 7SL-RNA, canine SINEs share no homology with humans (Bentolila 1999). They evolved from tRNA genes (Bentolila 1999) and are categorized by RMSK into tRNA-Lys (72%, SINEC_#), MIR (27%), Deu (0.2%, AmnSINE1, AmnSINE2), and tRNA (0.2%, LFSINE_Vert, MamSINE1) families (supplemental figure). Six out of 11 tRNA-Lys elements (54%) are up-regulated in DUX4 expressed cell line; in fact, all tRNA-Lys have logFC > 1. That is, the tRNA-Lys family is enriched in HinC transcriptome. Though none of the tRNA-Lys was called up-regulated in DUXC cells, their expression is highly correlated with that of in DUX4 expressed cell line (figure \@ref(fig:sine-distribution)).

Canine (canFam3) has 5 major satellites and satellite-like family: Bs, Carnivore Satellite (CarSat1, CarSat2), Satellite canis Familiaris (SAT1_CF, SAT2_CF, SAT3_CF, SAT4_CF, and SAT6_CF), (CATTC)n, (GAATG)n and SUBTEL_sa. Bs, CarSat1, and SAT1_CF were differentially up-regulated in DUX4 expressed cells, whereas in DUXC expressed cells, Bs element was called up-regulated, and CarSat1 was moderately affected (adjusted p-value = 0.14). 

__Load and datasets and libraries__
```{r load-datasets-library-08}
# define directory
pkg_dir <- "~/CompBio/canFam3.DuxFamily"
fig_dir <- file.path(pkg_dir, "figures")
data_dir <- file.path(pkg_dir, "data")
# load library
suppressPackageStartupMessages(library(DESeq2))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(tidyr))
suppressPackageStartupMessages(library(plyranges))
suppressPackageStartupMessages(library(ggrepel))
suppressPackageStartupMessages(library(purrr))
suppressPackageStartupMessages(library(canFam3.rmsk))
load(file.path("~/CompBio/hg38/hg38.rmsk.rda"))
suppressPackageStartupMessages(library(wesanderson)) # modern palettes

# load datasets and local functions
source(file.path(pkg_dir, "scripts", "tools.R"))
load(file.path(pkg_dir, "data", "HinC_rmsk.rda")) # made in chapter 6
load(file.path(pkg_dir, "data", "CinC_rmsk.rda")) # made in chapter 6
load(file.path(pkg_dir, "data", "CALTinC_rmsk.rda")) # made in chapter 6
load(file.path(pkg_dir, "data", "HinH_rmsk.rda"))
load(file.path(data_dir, "peaks_list.rda"))
```

Tidy up the differential analysis results on the repeats of HinC and CinC transcriptome into one _data.frame_ instance.

```{r comb_repeats}
comb <- inner_join(HinC_rmsk$res_df, CinC_rmsk$res_df, 
                   by="repName", suffix=c("_HinC", "_CinC")) %>%
  dplyr::mutate(DE_status = "neither") %>%
  dplyr::mutate(DE_status = 
                  replace(DE_status, padj_HinC < 0.05 | padj_CinC < 0.05, "either")) %>%
  dplyr::mutate(DE_status = replace(DE_status, padj_HinC < 0.05 & padj_CinC < 0.05, "both")) %>%
  dplyr::mutate(DE_status = factor(DE_status, levels=c("neither", "either", "both"))) 
```
reference: Analysis of major repetitive DNA sequences in the dog (Canis familiaris) genome, Betolila 1999.

## LINEs

__Background:__ LINEs constitute 20% of canine genome; LINE1 (L1) and LINE2 (L2) families derive 81% and 16% of the LINE repeat class, respectively. Whidden et al. [@whidden2017] showed that DUX4 up-regulated, directly or indirectly, impressive numbers of LINE1s in human myoblast. In this project, we examed whether DUX4 and DUXC transcription activities lead to similar LINE1s activation in canine myblast cells.

__Results:__        

- DUX4 up-regulated > 50% of L1s and L2s in canine cells          
- 35% DUX4 peaks binds to LINEs  (monoclonal cell line)           
- GSEA suggested LINE1 family is enriched in DUX4 transcriptome in canine cells          
- DUX4 up-regulated many carnivore-specific L1s and some Half-L1s (HAL1s, ancient L1-like)     
- DUXC up-regulated some LINEs (16% in LINE1 family); GESA does not suggest L1 enrichment in DUXC canine cells     
- Both DUX4 and DUXC up-regulate L1M2a (L1) and X6B_LINE (CR1) with impressive fold-change (~ 6 and 4)      
- DUXC’s L1-overlapped peaks have a palindromic motif - AATCAATTGATT (243/691); DUX4’s have TAA(G/T/C)TCAATCA (431/999) ([Chapter 9](#peaks-line1))
- Peaks overlapping with L1s have enriched __AATCA__ sequence ([Chapter 9](#peaks-line1))   

```{r line-distribution, warning=FALSE, fig.cap="Distribution of LINE family.", echo=FALSE}
line <- canFam3.rmsk %>% plyranges::filter(repClass=="LINE")
percent <- function(prob) paste0(format(prob, digit=0), "%")
line_freq <- line %>% 
  plyranges::mutate(rng_width = width(.)) %>%
  plyranges::group_by(repFamily) %>%
  plyranges::summarise(length = sum(rng_width)) %>% as.data.frame() %>%
  dplyr::arrange(desc(length)) %>%
  #dplyr::mutate(repFamily=factor(repFamily, levels=as.character(repFamily))) %>%
  dplyr::mutate(prop = 100 * (length/sum(length)))  %>%
  dplyr::mutate(ypos = cumsum(prop) - 0.5 * prop) %>%
  dplyr::mutate(percent=percent(prop), label = paste0(repFamily, " (", percent, ")"))
line_freq
```


DUX4 up-regulates > 50% of L1 and L2 retroelements in canine cells, whereas DUXC only up-regulated 16% of L1s. LFC of HinC and CinC are highly correlated.

```{r LINE-up-regulation-freq, echo=FALSE, fig.cap="Percentage of up-regulation of LINE retroelements in each familiy. Others include unclassified L1, Dong-R4, RTE-BovB and RTR-X."}
freq <- comb %>% dplyr::filter(repClass_HinC == "LINE") %>%
  dplyr::mutate(repFamily_HinC = factor(repFamily_HinC)) %>%
  dplyr::select(repName, repFamily_HinC, repClass_HinC, 
                log2FoldChange_HinC, log2FoldChange_CinC, padj_HinC, padj_CinC) %>%
  dplyr::rename(repFamily=repFamily_HinC, repClass=repClass_HinC) %>%
  tidyr::gather(key="sample", value="log2FC", -repName, -repFamily, -repClass, -padj_HinC, -padj_CinC) %>%
  dplyr::mutate(sample = ifelse(grepl("HinC", sample), "HinC", "CinC")) %>%
  dplyr::mutate(padj = ifelse(sample == "HinC", padj_HinC, padj_CinC)) %>%
  dplyr::mutate(DE_status = padj < 0.05 & log2FC > 0) %>% # fix repFamily: put *? to others
  dplyr::mutate(repFamily=as.character(repFamily)) %>%
  dplyr::mutate(repFamily = 
                replace(repFamily, 
                        which(repFamily %in% c("L1?", "Dong-R4", "RTE-BovB", "RTE-X")),
                        "Others")) %>%
  group_by(sample, repFamily) %>%
  summarise(n=dplyr::n(), n_up=sum(DE_status, na.rm=TRUE), prob=n_up/n)

ggplot(freq, aes(x=sample, y=prob, fill=repFamily) )+
  geom_bar(stat="identity", show.legend=FALSE) +
  facet_wrap( ~ repFamily, nrow=1) +
  scale_fill_brewer(palette="Set1") +
  geom_text(aes(label=format(prob, digit=1)), vjust=-0.3, size=3.5) +
  theme_minimal() +
  labs(title="Up-regulation % of LINE retroelements in each family")
```


```{r LINE-transcripts, warning=FALSE, message=FALSE, fig.cap="LINEs expression (LFC) in HinC and CinC.", echo=FALSE}
LINE <- comb %>% dplyr::filter(repClass_HinC == "LINE")
ggplot(LINE, aes(x=log2FoldChange_CinC, y=log2FoldChange_HinC)) +
  geom_point(size=1, alpha=0.7, aes(color=DE_status)) +
  scale_color_manual(values=c("#999999","#E69F00", "#56B4E9")) +
  geom_smooth(method="lm", se=FALSE, show.legend=FALSE, color="gray66", alpha=0.5,
              linetype="dashed") +
  geom_abline(intercept=0, slope=1, color="gray75", alpha=0.2, size=0.5) +              
  labs(title="LINEs: HinC vs CinC log2FC") +
  theme_bw()+
  theme(legend.position = c(0.1, 0.8)) 
  #geom_text_repel(aes(label=repName), size=2.5, show.legend=FALSE)

```

### L1s activation
__Question:__ Which subfamilies of L1s were up-regulated by DUX4 in canine cells? 

DUX4 up-regulated many mammalian-specific, carnivore-specific L1s and some HAL1s. 
```{r L1s, echo=FALSE, warning=FALSE, message=FALSE, fig.cap="LINE1 up-regulation in DUX4/DUXC expressed canine cells and DUX4 expressed human cells."}
activated_L1s <- inner_join(HinC_rmsk$res_df, CinC_rmsk$res_df, 
                   by="repName", suffix=c("_HinC", "_CinC")) %>%
  left_join(dplyr::select(HinH_rmsk$res_df, repName, log2FoldChange, padj), by="repName") %>%
  rename(repFamily = repFamily_CinC, log2FoldChange_HinH = log2FoldChange, padj_HinH = padj) %>%
  dplyr::filter(repFamily %in% c("L1")) %>%
  dplyr::filter(log2FoldChange_HinC > 0 | log2FoldChange_CinC > 0 | log2FoldChange_HinH > 0) %>%
  dplyr::filter(padj_HinC < 0.05 | padj_CinC < 0.05 | padj_HinH < 0.05) %>%
  dplyr::select(repName, repFamily, 
                log2FoldChange_HinC, padj_HinC, log2FoldChange_CinC, padj_CinC,
                log2FoldChange_HinH, padj_HinH)
      
logfc_gather <- gather(activated_L1s, key=tr_factor, value=logFC, 
                       log2FoldChange_HinC, log2FoldChange_CinC, log2FoldChange_HinH, 
                       -repName, -repFamily) %>%
  dplyr::mutate(padj = padj_HinC) %>%
  dplyr::mutate(padj = ifelse(tr_factor == "log2FoldChange_CinC", padj_CinC, padj)) %>%
  dplyr::mutate(padj = ifelse(tr_factor == "log2FoldChange_HinH", padj_HinH, padj)) %>%
  dplyr::select(repName, repFamily, tr_factor, logFC, padj) %>%
  dplyr::mutate(sig = ifelse(padj < 0.05, TRUE, FALSE)) %>%
  dplyr::mutate(tr_factor = sapply(strsplit(tr_factor, "_", fixed=TRUE), "[[", 2))
```



```{r plot-L1-elements, echo=FALSE, fig.height=8, fig.cap="L1s activation in DUXC and DUX4 in canine cells. Only display significently expressio L1s in either CinC, HinC, or HinH."}
gg3 <- ggplot(logfc_gather, aes(x = repName, y=logFC, fill=sig)) +
  geom_bar(stat="identity") +
  facet_wrap ( ~ tr_factor) +
  coord_flip() +
  theme_minimal() + 
  labs(title="L1 logFC") + 
  scale_fill_manual(values=c("#999999", "#E69F00")) +
  theme(legend.position = "none")
gg3
```

### LINE1 and DUXC/DUX4 binding sites

__This section is a summary of [Chapter 9](#peaks-line1).__

Our previous study ([@whidden2017]) and the canine model suggested that L1 subfamily is enriched in DUX4 expressed human and canine myoblasts. But does DUX4 (or DUXC) directly interact L1s? Are the binding sites on L1s artifacts? To answer the question, we want to first identify the motifs of peaks that overlap with the differentially expressed LINE1 elements (LFC > 1 with adjusted p-value < 0.05) by:     
     
1. use the narrowed peaks  (+/- 50 bps from summit) and select the subset that overlap with the differentially expression L1s         
2. get sequence of the selected peaks        
3. use _MEME_ to find the DUXC/DUX4 motif of these peaks          
4. use _BioStrings::matchPattern()_ to find the motif enrichment          

__Results:__ 
20.5% (21,888 out of 85,825) of DUX4 narrowed peaks overlap with L1s (25.50%); 14% of DUXC narrowed peaks overlap with L1s. Based on the top 1,000 of these L1-overlapped narrowed peaks of DUXC and DUX4 (on monoclonal cell lines), MEME identified a palindromic motif __AATCAATTGATT__ (243/691) in DUXC cells and a motif __TAAGTCAATCAG__ (431/999) in DUX4 cells. That is L1s do not have DUX4 nor DUXC motif enrichment; most likely that L1's up-regulated are directly affected by DUX4 or DUXC in canine cells. Interestingly, however, both motifs of L1-overlapped peaks in DUX4 and DUXC cells obtain a motif __AATCA__, which is the canonical, second part of DUX4, Dux and DUXC motifs. This implies that L1s are enriched with __AATCA__ sequences: 42.8% and 46.6% of DUXC and DUX4's L1-overlapped narrowed peaks have __AATCA__ sequences, mostly distributed near the center of the summits (technical details in [Chapter 9](#peaks-line1)).

__Take away:__
DUXC and DUX4 seem not directly regulated L1s.

## SINE
__Background:__ The short interspersed repetitive elements (SINEs) constitute approximately 10% of canine genome. Canine SINEs evolved from tRNA genes whereas human SINEs evolved from 7SL-RNA, which gave rise to human Alu family [@bentolila]; they do not share homology. However, known that human Alu family are enriched by DUX4 transcriptome in human myoblast [@shaddle2019], we are interested to know whether canine SINEs are also enriched by DUX4 and DUXC in canine myoblast and why? 

__Results:__ RMSK categorized canine SINEs into tRNA-Lys (72%, `SINEC_*`), MIR (27%), Deu (0.2%, `AmnSINE1`, `AmnSINE2`), and tRNA (0.2%, `LFSINE_Vert`, `MamSINE1`) families (Figure \@ref(fig:sine-distribution)) with total 20 name/group and 1,610,013 instances of sequence. Six out of 11 tRNA-Lys (54%, named `SINEC_*`) are up-regulated in DUX4 expressed cell line, and all tRNA-Lys have logFC > 1. This means the tRNA-Lys family is enriched in HinC transcriptome. Though none of tRNA-Lys elements were called up-regulated by DUXC, their expression is highly correlated with that of in DUX4 expressed cell line.  

__Note:__ Murine SINEs are derived from both 7SL-RNA (B1 family) and tRNA [@bentolila].

### Distribution of SINE family
```{r sine-distribution, message=FALSE, warning=FALSE, fig.cap="Distribution of SINE family.", echo=FALSE}
sine <- canFam3.rmsk %>% plyranges::filter(repClass=="SINE")
percent <- function(prob) paste0(format(prob, digit=1), "%")
sine_freq <- sine %>% 
  plyranges::mutate(rng_width = width(.)) %>%
  plyranges::group_by(repFamily) %>%
  plyranges::summarise(length = sum(rng_width)) %>% as.data.frame() %>%
  dplyr::arrange(desc(length)) %>%
  dplyr::mutate(prop = 100* (length/sum(length)))  %>%
  dplyr::mutate(ypos = cumsum(prop) - 0.5 * prop) %>%
  dplyr::mutate(percent=percent(prop), label = paste0(repFamily, " (", percent, ")"))

# Basic piechart
ggplot(sine_freq, aes(x="", y=prop, fill=repFamily)) +
  geom_bar(stat="identity", width=1, color="white", show.legend=FALSE) +
  coord_polar("y", start=0) +
  theme_void() +
  geom_text(aes(y=ypos, x=c(1, 1, 1, 1.2), label = label),
            color = "white", size=4) +
  scale_fill_brewer(palette="Set1") +
  labs(title="Distribution of SINE family")

```

### Expression of SINEs in HinC and CinC transcriptome
```{r SINE-transcripts, warning=FALSE, message=FALSE, fig.cap="SINEs expression in HinC and CinC.", echo=FALSE}
SINE <- comb %>% dplyr::filter(repClass_HinC == "SINE")
ggplot(SINE, aes(x=log2FoldChange_CinC, y=log2FoldChange_HinC)) +
  geom_point(size=1, alpha=0.7, aes(color=DE_status)) +
  scale_color_manual(values=c("#999999","#E69F00", "#56B4E9")) +
  geom_smooth(method="lm", se=FALSE, show.legend=FALSE, color="gray66", alpha=0.5,
              linetype="dashed") +
  geom_abline(intercept=0, slope=1, color="gray75", alpha=0.2, size=0.5) +              
  labs(title="SINE: HinC vs CinC log2FC") +
  theme_bw()+
  theme(legend.position = c(0.1, 0.88)) +
  geom_text_repel(aes(label=repName), size=2.5, show.legend=FALSE)

```
### tRNA-Lys
Table below listed number of up-regulated repeats (name/group) in each SINE family:
```{r tRNA-Lys-regulation}
SINE %>% dplyr::group_by(factor(repFamily_HinC)) %>%
  dplyr::summarise(n = dplyr::n(), 
                   HinC_sig = sum(padj_HinC < 0.05), CinC_sig=sum(padj_CinC < 0.05)) %>%
  knitr::kable(caption="Number of up-regulated repeats (name/group) in each SINE family")
```

## Satellites
__Background:__ Satellites are consisted with short tandem repeats, and Canine satellite DNA may be more complicated and not restricted to the tandem repeats (Bentolila 1999). Based on RMAK, canine genome (canFam3) has 5 major satellites and satellite-like DNA -  Bs, Carnivore Satellite (CarSat1, CarSat2), Satellite canis Familiaris (SAT1_CF, SAT2_CF, SAT3_CF, SAT4_CF and SAT6_CF), `(CATTC)n`, `(GAATG)n` and SUBTEL_sa. Bs, CarSat1, Carsat2, and SAT1_CF are the four satellites (repName) show some expression in DUX4 and DUXC expressed canine cell lines and therefore were analyzed. (CarSat1 and SAT1_CF are cetromeric satellite families.)

__Results:__ Bs, CarSat1, and SAT1_CF were up-regulated in HinC. Bs was up-regulated in CinC; CarSat1 was moderately affected ($padj = 0.14$).

How does SAT1_CF or CarSat compare to HSETII and GSETII?

```{r satellite}
comb %>% dplyr::filter(repClass_HinC == "Satellite") %>%
  dplyr::select(repName, log2FoldChange_HinC, log2FoldChange_CinC, padj_HinC, padj_CinC) %>%
  knitr::kable(caption="Significant satallites in HinC and CinC transcriptome.")
```
