<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 ChIP-seq for the canine model | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</title>
  <meta name="description" content="Bioinformatics and statistics analysis" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 ChIP-seq for the canine model | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Bioinformatics and statistics analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 ChIP-seq for the canine model | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD" />
  
  <meta name="twitter:description" content="Bioinformatics and statistics analysis" />
  

<meta name="author" content="Chao-Jen Wong" />


<meta name="date" content="2021-11-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="makeSE.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DUXC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#datasets"><i class="fa fa-check"></i><b>0.1</b> Datasets</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#annotation"><i class="fa fa-check"></i><b>0.2</b> Annotation</a><ul>
<li class="chapter" data-level="0.2.1" data-path="index.html"><a href="index.html#canfam3-ensembl-gtf"><i class="fa fa-check"></i><b>0.2.1</b> canFam3 ensembl GTF</a></li>
</ul></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>0.3</b> Software</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="chipseq.html"><a href="chipseq.html"><i class="fa fa-check"></i><b>1</b> ChIP-seq for the canine model</a><ul>
<li class="chapter" data-level="1.1" data-path="chipseq.html"><a href="chipseq.html#chip-seq-samples"><i class="fa fa-check"></i><b>1.1</b> ChIP-seq samples</a></li>
<li class="chapter" data-level="1.2" data-path="chipseq.html"><a href="chipseq.html#analysis-workflow"><i class="fa fa-check"></i><b>1.2</b> Analysis workflow</a></li>
<li class="chapter" data-level="1.3" data-path="chipseq.html"><a href="chipseq.html#peaks-and-functional-distribution"><i class="fa fa-check"></i><b>1.3</b> Peaks and functional distribution</a><ul>
<li class="chapter" data-level="1.3.1" data-path="chipseq.html"><a href="chipseq.html#number-of-peaks"><i class="fa fa-check"></i><b>1.3.1</b> Number of peaks</a></li>
<li class="chapter" data-level="1.3.2" data-path="chipseq.html"><a href="chipseq.html#functional-discribution"><i class="fa fa-check"></i><b>1.3.2</b> Functional discribution</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chipseq.html"><a href="chipseq.html#overlap-of-peaks"><i class="fa fa-check"></i><b>1.4</b> Overlap of peaks</a><ul>
<li class="chapter" data-level="1.4.1" data-path="chipseq.html"><a href="chipseq.html#duxc-and-dux4"><i class="fa fa-check"></i><b>1.4.1</b> DUXC and DUX4</a></li>
<li class="chapter" data-level="1.4.2" data-path="chipseq.html"><a href="chipseq.html#duxc-and-duxc-alt"><i class="fa fa-check"></i><b>1.4.2</b> DUXC and DUXC-ALT</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="chipseq.html"><a href="chipseq.html#motifs-analysis"><i class="fa fa-check"></i><b>1.5</b> Motifs analysis</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="makeSE.html"><a href="makeSE.html"><i class="fa fa-check"></i><b>2</b> RNA-seq to DESeq2</a><ul>
<li class="chapter" data-level="2.1" data-path="makeSE.html"><a href="makeSE.html#rna-seq-preprocessing"><i class="fa fa-check"></i><b>2.1</b> RNA-seq preprocessing</a></li>
<li class="chapter" data-level="2.2" data-path="makeSE.html"><a href="makeSE.html#reads-counts"><i class="fa fa-check"></i><b>2.2</b> Reads counts</a><ul>
<li class="chapter" data-level="2.2.1" data-path="makeSE.html"><a href="makeSE.html#define-gene-features"><i class="fa fa-check"></i><b>2.2.1</b> Define gene features</a></li>
<li class="chapter" data-level="2.2.2" data-path="makeSE.html"><a href="makeSE.html#read-counts"><i class="fa fa-check"></i><b>2.2.2</b> Read counts</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="makeSE.html"><a href="makeSE.html#deseq2"><i class="fa fa-check"></i><b>2.3</b> DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html"><i class="fa fa-check"></i><b>3</b> DUXC transcriptome</a><ul>
<li class="chapter" data-level="3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#load-datasets-and-settings"><i class="fa fa-check"></i><b>3.1</b> Load datasets and settings</a></li>
<li class="chapter" data-level="3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cinc-duxc-expresison-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2</b> CinC: DUXC expresison in canine muscle cells</a></li>
<li class="chapter" data-level="3.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hinc-dux4-expression-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.3</b> HinC: DUX4 expression in canine muscle cells</a></li>
<li class="chapter" data-level="3.4" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#duxc-alt-expression---caltinc"><i class="fa fa-check"></i><b>3.4</b> DUXC-ALT expression - CALTinC</a><ul>
<li class="chapter" data-level="3.4.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#go-analysis-for-the-down-regulated-gene-set"><i class="fa fa-check"></i><b>3.4.1</b> GO analysis for the down-regulated gene set</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cross-species-models-dux4-expression-in-canine-and-murine"><i class="fa fa-check"></i><b>3.5</b> Cross-species models: DUX4 expression in canine and murine</a><ul>
<li class="chapter" data-level="3.5.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-duxc-expression-in-canine"><i class="fa fa-check"></i><b>3.5.1</b> DUX4 and DUXC expression in canine</a></li>
<li class="chapter" data-level="3.5.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-dux-expression-in-murine"><i class="fa fa-check"></i><b>3.5.2</b> DUX4 and Dux expression in murine</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html"><i class="fa fa-check"></i><b>4</b> DUXC and Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#duxc-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.1</b> DUXC transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs"><i class="fa fa-check"></i><b>4.1.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.1.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome"><i class="fa fa-check"></i><b>4.1.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#dux-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.2</b> Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.2.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs-1"><i class="fa fa-check"></i><b>4.2.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.2.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome-1"><i class="fa fa-check"></i><b>4.2.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html"><i class="fa fa-check"></i><b>5</b> DUXC transcriptome compared to early embryo transcriptome profile</a><ul>
<li class="chapter" data-level="5.1" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#mouse-2c-like-signature"><i class="fa fa-check"></i><b>5.1</b> Mouse 2C-like signature</a></li>
<li class="chapter" data-level="5.2" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#cleavage-stage-signature"><i class="fa fa-check"></i><b>5.2</b> cleavage-stage signature</a></li>
<li class="chapter" data-level="5.3" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#ipscs"><i class="fa fa-check"></i><b>5.3</b> iPSCs</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="repeats.html"><a href="repeats.html"><i class="fa fa-check"></i><b>6</b> Repeat elements</a><ul>
<li class="chapter" data-level="6.1" data-path="repeats.html"><a href="repeats.html#takeaway"><i class="fa fa-check"></i><b>6.1</b> Takeaway</a></li>
<li class="chapter" data-level="6.2" data-path="repeats.html"><a href="repeats.html#repeat-element-expression"><i class="fa fa-check"></i><b>6.2</b> Repeat element expression</a></li>
<li class="chapter" data-level="6.3" data-path="repeats.html"><a href="repeats.html#differential-analysis"><i class="fa fa-check"></i><b>6.3</b> Differential analysis</a><ul>
<li class="chapter" data-level="6.3.1" data-path="repeats.html"><a href="repeats.html#normalization"><i class="fa fa-check"></i><b>6.3.1</b> Normalization</a></li>
<li class="chapter" data-level="6.3.2" data-path="repeats.html"><a href="repeats.html#deseq2-results"><i class="fa fa-check"></i><b>6.3.2</b> DESeq2 results</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="repeats.html"><a href="repeats.html#repeat-family-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> Repeat family enrichment analysis</a><ul>
<li class="chapter" data-level="6.4.1" data-path="repeats.html"><a href="repeats.html#correlation-between-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>6.4.1</b> Correlation between HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="6.4.2" data-path="repeats.html"><a href="repeats.html#repeat-family-activation"><i class="fa fa-check"></i><b>6.4.2</b> Repeat family activation</a></li>
<li class="chapter" data-level="6.4.3" data-path="repeats.html"><a href="repeats.html#repeat-classs-activation"><i class="fa fa-check"></i><b>6.4.3</b> Repeat classs activation</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="repeats.html"><a href="repeats.html#ltr-retroelements-distribution-and-activation"><i class="fa fa-check"></i><b>6.5</b> LTR retroelements distribution and activation</a></li>
<li class="chapter" data-level="6.6" data-path="repeats.html"><a href="repeats.html#peaks-distribution-in-class-of-repeat-elements"><i class="fa fa-check"></i><b>6.6</b> Peaks distribution in class of repeat elements</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="LTR-activation.html"><a href="LTR-activation.html"><i class="fa fa-check"></i><b>7</b> LTR activation</a><ul>
<li class="chapter" data-level="7.1" data-path="LTR-activation.html"><a href="LTR-activation.html#takaway"><i class="fa fa-check"></i><b>7.1</b> Takaway</a></li>
<li class="chapter" data-level="7.2" data-path="LTR-activation.html"><a href="LTR-activation.html#background-in-murine-and-human"><i class="fa fa-check"></i><b>7.2</b> background in murine and human</a></li>
<li class="chapter" data-level="7.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-subfamily-activation-in-canine"><i class="fa fa-check"></i><b>7.3</b> ERVL subfamily activation in canine</a></li>
<li class="chapter" data-level="7.4" data-path="LTR-activation.html"><a href="LTR-activation.html#malr-activation-in-canine"><i class="fa fa-check"></i><b>7.4</b> MaLR activation in canine</a></li>
<li class="chapter" data-level="7.5" data-path="LTR-activation.html"><a href="LTR-activation.html#compare-wih-hinh"><i class="fa fa-check"></i><b>7.5</b> Compare wih HinH</a><ul>
<li class="chapter" data-level="7.5.1" data-path="LTR-activation.html"><a href="LTR-activation.html#copy-number"><i class="fa fa-check"></i><b>7.5.1</b> Copy number</a></li>
<li class="chapter" data-level="7.5.2" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-malr"><i class="fa fa-check"></i><b>7.5.2</b> ERVL-MaLR</a></li>
<li class="chapter" data-level="7.5.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl"><i class="fa fa-check"></i><b>7.5.3</b> ERVL</a></li>
<li class="chapter" data-level="7.5.4" data-path="LTR-activation.html"><a href="LTR-activation.html#erv1"><i class="fa fa-check"></i><b>7.5.4</b> ERV1</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="LTR-activation.html"><a href="LTR-activation.html#duxc-alt-and-ervl-transcriptions"><i class="fa fa-check"></i><b>7.6</b> DUXC-ALT and ERVL transcriptions</a><ul>
<li class="chapter" data-level="7.6.1" data-path="LTR-activation.html"><a href="LTR-activation.html#erv-subfamily-expression"><i class="fa fa-check"></i><b>7.6.1</b> ERV subfamily expression</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="non-LTR.html"><a href="non-LTR.html"><i class="fa fa-check"></i><b>8</b> Non-LTR retrotrasposons and satellites</a><ul>
<li class="chapter" data-level="8.1" data-path="non-LTR.html"><a href="non-LTR.html#lines"><i class="fa fa-check"></i><b>8.1</b> LINEs</a><ul>
<li class="chapter" data-level="8.1.1" data-path="non-LTR.html"><a href="non-LTR.html#l1s-activation"><i class="fa fa-check"></i><b>8.1.1</b> L1s activation</a></li>
<li class="chapter" data-level="8.1.2" data-path="non-LTR.html"><a href="non-LTR.html#line1-and-duxcdux4-binding-sites"><i class="fa fa-check"></i><b>8.1.2</b> LINE1 and DUXC/DUX4 binding sites</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="non-LTR.html"><a href="non-LTR.html#sine"><i class="fa fa-check"></i><b>8.2</b> SINE</a><ul>
<li class="chapter" data-level="8.2.1" data-path="non-LTR.html"><a href="non-LTR.html#distribution-of-sine-family"><i class="fa fa-check"></i><b>8.2.1</b> Distribution of SINE family</a></li>
<li class="chapter" data-level="8.2.2" data-path="non-LTR.html"><a href="non-LTR.html#expression-of-sines-in-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>8.2.2</b> Expression of SINEs in HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="8.2.3" data-path="non-LTR.html"><a href="non-LTR.html#trna-lys"><i class="fa fa-check"></i><b>8.2.3</b> tRNA-Lys</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="non-LTR.html"><a href="non-LTR.html#satellites"><i class="fa fa-check"></i><b>8.3</b> Satellites</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="peaks-line1.html"><a href="peaks-line1.html"><i class="fa fa-check"></i><b>9</b> DUXC peaks and LINE1s</a><ul>
<li class="chapter" data-level="9.1" data-path="peaks-line1.html"><a href="peaks-line1.html#narrow-peaks-and-expressed-l1s"><i class="fa fa-check"></i><b>9.1</b> Narrow Peaks and expressed L1s</a><ul>
<li class="chapter" data-level="9.1.1" data-path="peaks-line1.html"><a href="peaks-line1.html#motif-discovery-using-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.1</b> Motif discovery using L1-overlapped peaks</a></li>
<li class="chapter" data-level="9.1.2" data-path="peaks-line1.html"><a href="peaks-line1.html#match-dux4-and-duxc-motif-to-l1-overapped-peaks"><i class="fa fa-check"></i><b>9.1.2</b> Match DUX4 and DUXC motif to L1-overapped peaks</a></li>
<li class="chapter" data-level="9.1.3" data-path="peaks-line1.html"><a href="peaks-line1.html#location-of-aatca-on-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.3</b> Location of AATCA on L1-overlapped peaks</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="peaks-line1.html"><a href="peaks-line1.html#line1s-and-aatca"><i class="fa fa-check"></i><b>9.2</b> LINE1s and AATCA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i># References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chipseq" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> ChIP-seq for the canine model</h1>
<p>We performed ChIP-seq experiments to determind the transcription factor DUX4, DUXC and DUXC-ALT’s binding sites and motifs in the canine genome. DUX4 and DUXC were using monoclonal antiboy whereas DUXC-ALT using polyclonol Below we described the sample information and ChIP-seq preprocessing pipeline, peaks calling, and motifs discovery.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(TxDb.Cfamiliaris.UCSC.canFam3.ensGene))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">txdb &lt;-<span class="st"> </span>TxDb.Cfamiliaris.UCSC.canFam3.ensGene</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(canFam3.rmsk))</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(tidyverse))</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(wesanderson))</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(ggVennDiagram))</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co"># eventially change to loading from github</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8">pkg_dir &lt;-<span class="st"> &quot;~/CompBio/canFam3.DuxFamily&quot;</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;peaks_list.rda&quot;</span>)) <span class="co"># peaks list</span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;chipseq_si.rda&quot;</span>)) <span class="co"># sample information</span></a></code></pre></div>
<div id="chip-seq-samples" class="section level2">
<h2><span class="header-section-number">1.1</span> ChIP-seq samples</h2>
<pre><code>##                  sample_name lib_size transcription_factor  cell_type
## 1   Sample1_ImmCnMb_luc_poly 16561563            luciferas polyclonal
## 2 Sample4_ImmCnMb_CALTh_poly 17995356             DUXC-ALT polyclonal
## 3   Sample5_ImmCnMb_luc_mono 14149968            luciferas monoclonal
## 4 Sample6_ImmCnMb_hDUX4_mono 15326657                 DUX4 monoclonal
## 5   Sample7_ImmCnMb_CCH_mono 15316644                 DUXC monoclonal</code></pre>
</div>
<div id="analysis-workflow" class="section level2">
<h2><span class="header-section-number">1.2</span> Analysis workflow</h2>
<p>We used starndard pre-processing pipeline and <em>MACS2</em> to call peaks</p>
<ol style="list-style-type: decimal">
<li><strong>Filter unqualified reads</strong>: based on casava filter, filtered out reads whose filter flag, indicated in the identifier, equals to “Y”.</li>
<li><strong>Trim adapter</strong>: <code>AGATCGGAAGAGCACACGTCAGAACTCCAGTCACGCACTAATCTCGTATG</code><br />
</li>
<li><strong>Quality control</strong>: quality control by <em>fastqc</em> to confirm that there is no more adapter contemination<br />
</li>
<li><strong>Alignement</strong>: align read to the <em>canFam3</em> genome build by <em>BWA-0.7.10</em> (<code>/scripts/do_bwa.sh</code>)</li>
<li><strong>Peak calling</strong>: peak calling made by macs2 (MACS2) with Luciferase monoclonal samples as negative controls; monoclonal cell line <code>Sample5_ImmCnMb_luc_mono</code> is used for monoclonal DUXC/DUX4 samples and the polyclonal cell line <code>Sample1_ImmCnMb_luc_poly</code> for <code>Sample4_ImmCnMb_CALTh_poly</code>)</li>
</ol>
<pre><code>macs2 callpeak -t $sampleName.bam -c $cnt -f BAM \
  --outdir $macsDir -g 2.42e9 -n $sampleName \
  -B -q 0.01</code></pre>
<ol start="6" style="list-style-type: decimal">
<li><strong>Peak annotation</strong>: annotate peaks using Bioconductor <strong>ChIPseeker</strong> <span class="citation">(Yu, Wang, and He <a href="#ref-ChIPseeker">2015</a>)</span> package along with a customed <code>TxDb</code> (<code>TxDb.Cfamiliaris.UCSC.canFam3.ensGene</code>) annotation package derived from the Ensembl’s gtf file in the canfam3 genome build.<br />
</li>
<li><strong>Motif discovery</strong>: prepare peaks FASTA files by selecting top 1000 peaks and flank up/downstream 50 bps around the peak summits (<code>/scripts/peaks.R</code>); discovery motifs by using MEME 5.1.1 (<code>/scripts/peaks.R</code>)</li>
</ol>
<pre><code>meme Sample6_ImmCnMb_hDUX4_mono_peaksSeq_100.fa \ 
  -dna -oc . -nostatus -time 18000 -mod zoops \ 
  -nmotifs 3 -minw 6 -maxw 11 -objfun classic \ 
  -revcomp -markov_order 0</code></pre>
</div>
<div id="peaks-and-functional-distribution" class="section level2">
<h2><span class="header-section-number">1.3</span> Peaks and functional distribution</h2>
<p>The peaks file is located at <code>../data/peaks_list.rda</code>: It is a <code>list</code> of <code>GRanges</code> instances containing peaks coordinates, summits, q-values, annotaion, overlapping staus with repeat elements, and orthologs. For example:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">peaks_list[[<span class="st">&quot;Sample7_ImmCnMb_CCH_mono&quot;</span>]]</a></code></pre></div>
<pre><code>## GRanges object with 6031 ranges and 24 metadata columns:
##          seqnames              ranges strand |    length abs_summit    pileup
##             &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;  &lt;integer&gt; &lt;numeric&gt;
##      [1]     chr1             285-544      * |       260        519       135
##      [2]     chr1           2696-3055      * |       360       2847       244
##      [3]     chr1           4078-4790      * |       713       4602       367
##      [4]     chr1           5785-6158      * |       374       6004       385
##      [5]     chr1           6764-7193      * |       430       6946       473
##      ...      ...                 ...    ... .       ...        ...       ...
##   [6027]     chrX 122767363-122767625      * |       263  122767497        12
##   [6028]     chrX 122902787-122903048      * |       262  122902820        13
##   [6029]     chrX 123168859-123169208      * |       350  123169031        14
##   [6030]     chrX 123213575-123213866      * |       292  123213776        13
##   [6031]     chrX 123686002-123686277      * |       276  123686087        13
##          X.log10.pvalue. fold_enrichment X.log10.qvalue.
##                &lt;numeric&gt;       &lt;numeric&gt;       &lt;numeric&gt;
##      [1]         7.04597         1.60741         3.91784
##      [2]         7.02327          1.4164         3.89587
##      [3]         6.33763         1.30306         3.33661
##      [4]         7.20888         1.32224         4.06735
##      [5]        14.09717         1.45134        10.16508
##      ...             ...             ...             ...
##   [6027]         9.90244         6.31739         6.41459
##   [6028]          9.6226         5.93351         6.15395
##   [6029]         9.94595         5.91592         6.45397
##   [6030]        11.11847         6.86022         7.49343
##   [6031]         7.43072         4.60644         4.26318
##                                        name
##                                 &lt;character&gt;
##      [1]    Sample7_ImmCnMb_CCH_mono_peak_1
##      [2]    Sample7_ImmCnMb_CCH_mono_peak_2
##      [3]    Sample7_ImmCnMb_CCH_mono_peak_3
##      [4]    Sample7_ImmCnMb_CCH_mono_peak_4
##      [5]    Sample7_ImmCnMb_CCH_mono_peak_5
##      ...                                ...
##   [6027] Sample7_ImmCnMb_CCH_mono_peak_6027
##   [6028] Sample7_ImmCnMb_CCH_mono_peak_6028
##   [6029] Sample7_ImmCnMb_CCH_mono_peak_6029
##   [6030] Sample7_ImmCnMb_CCH_mono_peak_6030
##   [6031] Sample7_ImmCnMb_CCH_mono_peak_6031
##                                                                   annotation
##                                                                  &lt;character&gt;
##      [1]                                                   Distal Intergenic
##      [2]                                                   Distal Intergenic
##      [3]                                                   Distal Intergenic
##      [4]                                                   Distal Intergenic
##      [5]                                                   Distal Intergenic
##      ...                                                                 ...
##   [6027]   Intron (ENSCAFT00000044495.2/ENSCAFG00000019613.4, intron 1 of 6)
##   [6028] Intron (ENSCAFT00000031203.3/ENSCAFG00000019631.3, intron 25 of 25)
##   [6029]  Intron (ENSCAFT00000031213.3/ENSCAFG00000019638.3, intron 6 of 15)
##   [6030]                                                    Promoter (2-3kb)
##   [6031]                                                   Distal Intergenic
##              geneChr   geneStart     geneEnd  geneLength  geneStrand
##          &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;
##      [1]           1      270005      275409        5405           1
##      [2]           1      270005      275409        5405           1
##      [3]           1      270005      275409        5405           1
##      [4]           1      270005      275409        5405           1
##      [5]           1      270005      275409        5405           1
##      ...         ...         ...         ...         ...         ...
##   [6027]          39   122556508   122779142      222635           2
##   [6028]          39   122912936   122913286         351           1
##   [6029]          39   123162497   123205171       42675           1
##   [6030]          39   123211029   123230857       19829           1
##   [6031]          39   123695785   123707849       12065           2
##                        geneId         transcriptId distanceToTSS overlap.LTR
##                   &lt;character&gt;          &lt;character&gt;   &lt;character&gt;   &lt;logical&gt;
##      [1] ENSCAFG00000030108.1 ENSCAFT00000043967.1       -269461       FALSE
##      [2] ENSCAFG00000030108.1 ENSCAFT00000043967.1       -266950       FALSE
##      [3] ENSCAFG00000030108.1 ENSCAFT00000043967.1       -265215       FALSE
##      [4] ENSCAFG00000030108.1 ENSCAFT00000043967.1       -263847       FALSE
##      [5] ENSCAFG00000030108.1 ENSCAFT00000043967.1       -262812       FALSE
##      ...                  ...                  ...           ...         ...
##   [6027] ENSCAFG00000019613.4 ENSCAFT00000044495.2         11517       FALSE
##   [6028] ENSCAFG00000031844.1 ENSCAFT00000037773.2         -9888       FALSE
##   [6029] ENSCAFG00000019638.3 ENSCAFT00000031213.3          6362       FALSE
##   [6030] ENSCAFG00000019639.3 ENSCAFT00000031215.3          2546       FALSE
##   [6031] ENSCAFG00000033806.1 ENSCAFT00000054225.1         21572       FALSE
##                            repRange     repName    repClass   repFamily
##                         &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;
##      [1]           chr1:100-11597:-          Bs   Satellite   Satellite
##      [2]           chr1:100-11597:-          Bs   Satellite   Satellite
##      [3]           chr1:100-11597:-          Bs   Satellite   Satellite
##      [4]           chr1:100-11597:-          Bs   Satellite   Satellite
##      [5]           chr1:100-11597:-          Bs   Satellite   Satellite
##      ...                        ...         ...         ...         ...
##   [6027] chrX:122767214-122767812:+   L1_Canid_        LINE          L1
##   [6028] chrX:122902887-122903062:-       L1MEd        LINE          L1
##   [6029] chrX:123169121-123169286:+    SINEC_a1        SINE    tRNA-Lys
##   [6030] chrX:123213504-123213705:-        L1M5        LINE          L1
##   [6031] chrX:123685809-123686326:-   L1_Canis1        LINE          L1
##          external_gene_name hsapiens_homolog_associated_gene_name
##                 &lt;character&gt;                           &lt;character&gt;
##      [1]                                               AP002962.1
##      [2]                                               AP002962.1
##      [3]                                               AP002962.1
##      [4]                                               AP002962.1
##      [5]                                               AP002962.1
##      ...                ...                                   ...
##   [6027]               GAB3                                  GAB3
##   [6028]                                                   H2AFB1
##   [6029]                                                         
##   [6030]               VBP1                                  VBP1
##   [6031]                                                         
##          mmusculus_homolog_associated_gene_name
##                                     &lt;character&gt;
##      [1]                                       
##      [2]                                       
##      [3]                                       
##      [4]                                       
##      [5]                                       
##      ...                                    ...
##   [6027]                                   Gab3
##   [6028]                                 H2afb2
##   [6029]                                       
##   [6030]                                   Vbp1
##   [6031]                                       
##   -------
##   seqinfo: 40 sequences from an unspecified genome; no seqlengths</code></pre>
<div id="number-of-peaks" class="section level3">
<h3><span class="header-section-number">1.3.1</span> Number of peaks</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">n_peaks &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">map_dfr</span>(peaks_list, length)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var=</span><span class="st">&quot;sample_name&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">num_peaks=</span>V1) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="st">  </span><span class="kw">add_column</span>(<span class="dt">transcription_factor =</span> <span class="kw">c</span>(<span class="st">&quot;DUX4&quot;</span>, <span class="st">&quot;DUXC&quot;</span>, <span class="st">&quot;DUXC_ALT&quot;</span>,</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">                                       <span class="st">&quot;DUX4&quot;</span>, <span class="st">&quot;DUXC&quot;</span>))</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">n_peaks[<span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">3</span>), ]</a></code></pre></div>
<pre><code>##                  sample_name num_peaks transcription_factor
## 5   Sample7_ImmCnMb_CCH_mono      6031                 DUXC
## 4 Sample6_ImmCnMb_hDUX4_mono     85825                 DUX4
## 3 Sample4_ImmCnMb_CALTh_poly       561             DUXC_ALT</code></pre>
</div>
<div id="functional-discribution" class="section level3">
<h3><span class="header-section-number">1.3.2</span> Functional discribution</h3>
<p>Visualization of the genomic annotation by peaks’ functional location.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">.simplify_annotation &lt;-<span class="st"> </span><span class="cf">function</span>(peaks_gr) {</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">  <span class="co"># Promoter, 5&#39; UTR, Exon, Intron, 3&#39; UTR, Distal Intergenic, </span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3">  <span class="kw">data.frame</span>(<span class="dt">detail =</span> peaks_gr<span class="op">$</span>annotation) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="st">    </span><span class="kw">add_column</span>(<span class="dt">simplify=</span><span class="st">&quot;unknown&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">simplify =</span> <span class="kw">case_when</span>(</a>
<a class="sourceLine" id="cb9-6" data-line-number="6">      <span class="kw">grepl</span>(<span class="st">&quot;Promoter&quot;</span>, detail) <span class="op">~</span><span class="st"> &quot;Promoter&quot;</span>,</a>
<a class="sourceLine" id="cb9-7" data-line-number="7">      <span class="kw">grepl</span>(<span class="st">&quot;5&#39; UTR&quot;</span>, detail) <span class="op">~</span><span class="st"> &quot;5&#39; UTR&quot;</span>,</a>
<a class="sourceLine" id="cb9-8" data-line-number="8">      <span class="kw">grepl</span>(<span class="st">&quot;Exon&quot;</span>, detail) <span class="op">~</span><span class="st"> &quot;Exon&quot;</span>,</a>
<a class="sourceLine" id="cb9-9" data-line-number="9">      <span class="kw">grepl</span>(<span class="st">&quot;Intron&quot;</span>, detail) <span class="op">~</span><span class="st"> &quot;Intron&quot;</span>,</a>
<a class="sourceLine" id="cb9-10" data-line-number="10">      <span class="kw">grepl</span>(<span class="st">&quot;3&#39; UTR&quot;</span>, detail) <span class="op">~</span><span class="st"> &quot;3&#39; UTR&quot;</span>,</a>
<a class="sourceLine" id="cb9-11" data-line-number="11">      <span class="kw">grepl</span>(<span class="st">&quot;Downstream&quot;</span>, detail) <span class="op">~</span><span class="st"> &quot;Downstream&quot;</span>,</a>
<a class="sourceLine" id="cb9-12" data-line-number="12">      <span class="kw">grepl</span>(<span class="st">&quot;Distal Intergenic&quot;</span>, detail) <span class="op">~</span><span class="st"> &quot;Distal Intergenic&quot;</span>)</a>
<a class="sourceLine" id="cb9-13" data-line-number="13">    ) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">simplify=</span><span class="kw">factor</span>(simplify, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;Promoter&quot;</span>, <span class="st">&quot;5&#39; UTR&quot;</span>,</a>
<a class="sourceLine" id="cb9-15" data-line-number="15">                                                     <span class="st">&quot;Exon&quot;</span>,</a>
<a class="sourceLine" id="cb9-16" data-line-number="16">                                                     <span class="st">&quot;Intron&quot;</span>, <span class="st">&quot;3&#39; UTR&quot;</span>,</a>
<a class="sourceLine" id="cb9-17" data-line-number="17">                                                     <span class="st">&quot;Downstream&quot;</span>,</a>
<a class="sourceLine" id="cb9-18" data-line-number="18">                                                     <span class="st">&quot;Distal Intergenic&quot;</span>)))</a>
<a class="sourceLine" id="cb9-19" data-line-number="19">}</a></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">CCH_df &lt;-<span class="st"> </span><span class="kw">.simplify_annotation</span>(peaks_list[[<span class="st">&quot;Sample7_ImmCnMb_CCH_mono&quot;</span>]]) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="st">  </span><span class="kw">count</span>(simplify) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">freq=</span>n<span class="op">/</span><span class="kw">sum</span>(n), <span class="dt">transcription_factor=</span><span class="st">&quot;DUXC&quot;</span>) </a>
<a class="sourceLine" id="cb10-4" data-line-number="4">  </a>
<a class="sourceLine" id="cb10-5" data-line-number="5">hDUX4_df &lt;-<span class="st"> </span><span class="kw">.simplify_annotation</span>(peaks_list[[<span class="st">&quot;Sample6_ImmCnMb_hDUX4_mono&quot;</span>]]) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="st">  </span><span class="kw">count</span>(simplify) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">freq=</span>n<span class="op">/</span><span class="kw">sum</span>(n), <span class="dt">transcription_factor=</span><span class="st">&quot;DUX4&quot;</span>) </a>
<a class="sourceLine" id="cb10-8" data-line-number="8"></a>
<a class="sourceLine" id="cb10-9" data-line-number="9">CALTH_df &lt;-<span class="st"> </span><span class="kw">.simplify_annotation</span>(peaks_list[[<span class="st">&quot;Sample4_ImmCnMb_CALTh_poly&quot;</span>]]) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10"><span class="st">  </span><span class="kw">count</span>(simplify) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-11" data-line-number="11"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">freq=</span>n<span class="op">/</span><span class="kw">sum</span>(n), <span class="dt">transcription_factor=</span><span class="st">&quot;DUXC-ALT&quot;</span>) </a>
<a class="sourceLine" id="cb10-12" data-line-number="12"></a>
<a class="sourceLine" id="cb10-13" data-line-number="13">df &lt;-<span class="st"> </span>CCH_df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">bind_rows</span>(hDUX4_df) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">bind_rows</span>(CALTH_df) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-14" data-line-number="14"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">transcription_factor=</span><span class="kw">factor</span>(transcription_factor, </a>
<a class="sourceLine" id="cb10-15" data-line-number="15">                                            <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;DUX4&quot;</span>, <span class="st">&quot;DUXC&quot;</span>,</a>
<a class="sourceLine" id="cb10-16" data-line-number="16">                                                     <span class="st">&quot;DUXC-ALT&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-17" data-line-number="17"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">percentage=</span>freq <span class="op">*</span><span class="st"> </span><span class="dv">100</span>)</a>
<a class="sourceLine" id="cb10-18" data-line-number="18"></a>
<a class="sourceLine" id="cb10-19" data-line-number="19"><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(<span class="dt">x=</span>transcription_factor, <span class="dt">y=</span>percentage, <span class="dt">fill=</span>simplify)) <span class="op">+</span></a>
<a class="sourceLine" id="cb10-20" data-line-number="20"><span class="st">   </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb10-21" data-line-number="21"><span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb10-22" data-line-number="22"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">y=</span><span class="st">&quot;Percentage (%)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb10-23" data-line-number="23"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.title.x=</span><span class="kw">element_blank</span>(), <span class="dt">legend.title =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb10-24" data-line-number="24">        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">hjust=</span><span class="dv">1</span>))<span class="op">+</span></a>
<a class="sourceLine" id="cb10-25" data-line-number="25"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">wes_palette</span>(<span class="st">&quot;Darjeeling2&quot;</span>, <span class="dt">n=</span><span class="dv">7</span>,</a>
<a class="sourceLine" id="cb10-26" data-line-number="26">                                         <span class="dt">type=</span><span class="st">&quot;continuous&quot;</span>))</a></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:bar-plot-functional-distribution"></span>
<img src="gitbook_files/figure-html/bar-plot-functional-distribution-1.png" alt="Functional distribution of peaks." width="288" />
<p class="caption">
Figure 1.1: Functional distribution of peaks.
</p>
</div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;figures&quot;</span>, <span class="st">&quot;chipseq-peaks-functional-distribution.pdf&quot;</span>), <span class="dt">width=</span><span class="dv">3</span>, <span class="dt">height=</span><span class="dv">4</span>)</a></code></pre></div>
</div>
</div>
<div id="overlap-of-peaks" class="section level2">
<h2><span class="header-section-number">1.4</span> Overlap of peaks</h2>
<div id="duxc-and-dux4" class="section level3">
<h3><span class="header-section-number">1.4.1</span> DUXC and DUX4</h3>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">genes=<span class="st"> </span><span class="kw">lapply</span>(peaks_list[<span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">5</span>)], <span class="cf">function</span>(gr) gr<span class="op">$</span>geneId)</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="kw">names</span>(genes) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;DUX4&quot;</span>, <span class="st">&quot;DUXC&quot;</span>)</a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="co">#vennplot(genes)</span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="kw">ggVennDiagram</span>(genes)</a></code></pre></div>
<p><img src="gitbook_files/figure-html/vennplot-DUXC-DUX4-ov-1.png" width="672" /></p>
</div>
<div id="duxc-and-duxc-alt" class="section level3">
<h3><span class="header-section-number">1.4.2</span> DUXC and DUXC-ALT</h3>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">genes=<span class="st"> </span><span class="kw">lapply</span>(peaks_list[<span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">5</span>)], <span class="cf">function</span>(gr) gr<span class="op">$</span>geneId)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="kw">names</span>(genes) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;DUXC-ALT&quot;</span>, <span class="st">&quot;DUXC&quot;</span>)</a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="kw">ggVennDiagram</span>(genes)</a></code></pre></div>
<p><img src="gitbook_files/figure-html/vennplot-DUXC-ALT-DUX4-ov-1.png" width="672" /></p>
</div>
</div>
<div id="motifs-analysis" class="section level2">
<h2><span class="header-section-number">1.5</span> Motifs analysis</h2>
<p><strong>Quesition:</strong> Do DUXC’s A or G peaks overlap with hDUX4 peaks?
About 60% and 50% of A-peaks and G-peaks overlap with hDUX4 peaks, respectively.</p>
<p><strong>Question:</strong> As for DUXC motif T(A/G)ATCCAATCA, does the position, A/G, associated with induced genes?
No, neither A nor G motif is associated with induced genes.</p>

</div>
</div>
<h3># References</h3>
<div id="refs" class="references">
<div id="ref-ChIPseeker">
<p>Yu, Guangchuang, Li-Gen Wang, and Qing-Yu He. 2015. “ChIPseeker: An R/Bioconductor Package for Chip Peak Annotation, Comparison and Visualization.” <em>Bioinformatics</em> 31 (14): 2382–3. <a href="https://doi.org/10.1093/bioinformatics/btv145">https://doi.org/10.1093/bioinformatics/btv145</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="makeSE.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/FredHutch/canFam3.DuxFamily/tree/master/gitbook/01-ChIPseq-Peaks-and-Motifs.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/FredHutch/canFam3.DuxFamily/tree/master/gitbook/01-ChIPseq-Peaks-and-Motifs.Rmd",
"text": null
},
"download": ["gitbook.pdf", "gitbook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
