<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 DUXC and Dux transcriptome compared to that of FSHD | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</title>
  <meta name="description" content="Bioinformatics and statistics analysis" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 DUXC and Dux transcriptome compared to that of FSHD | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Bioinformatics and statistics analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 DUXC and Dux transcriptome compared to that of FSHD | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD" />
  
  <meta name="twitter:description" content="Bioinformatics and statistics analysis" />
  

<meta name="author" content="Chao-Jen Wong" />


<meta name="date" content="2021-11-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="DUXC-transcriptome.html"/>
<link rel="next" href="DUXC-early-embryo.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DUXC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#datasets"><i class="fa fa-check"></i><b>0.1</b> Datasets</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#annotation"><i class="fa fa-check"></i><b>0.2</b> Annotation</a><ul>
<li class="chapter" data-level="0.2.1" data-path="index.html"><a href="index.html#canfam3-ensembl-gtf"><i class="fa fa-check"></i><b>0.2.1</b> canFam3 ensembl GTF</a></li>
</ul></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>0.3</b> Software</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="chipseq.html"><a href="chipseq.html"><i class="fa fa-check"></i><b>1</b> ChIP-seq for the canine model</a><ul>
<li class="chapter" data-level="1.1" data-path="chipseq.html"><a href="chipseq.html#chip-seq-samples"><i class="fa fa-check"></i><b>1.1</b> ChIP-seq samples</a></li>
<li class="chapter" data-level="1.2" data-path="chipseq.html"><a href="chipseq.html#analysis-workflow"><i class="fa fa-check"></i><b>1.2</b> Analysis workflow</a></li>
<li class="chapter" data-level="1.3" data-path="chipseq.html"><a href="chipseq.html#peaks-and-functional-distribution"><i class="fa fa-check"></i><b>1.3</b> Peaks and functional distribution</a><ul>
<li class="chapter" data-level="1.3.1" data-path="chipseq.html"><a href="chipseq.html#number-of-peaks"><i class="fa fa-check"></i><b>1.3.1</b> Number of peaks</a></li>
<li class="chapter" data-level="1.3.2" data-path="chipseq.html"><a href="chipseq.html#functional-discribution"><i class="fa fa-check"></i><b>1.3.2</b> Functional discribution</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chipseq.html"><a href="chipseq.html#overlap-of-peaks"><i class="fa fa-check"></i><b>1.4</b> Overlap of peaks</a><ul>
<li class="chapter" data-level="1.4.1" data-path="chipseq.html"><a href="chipseq.html#duxc-and-dux4"><i class="fa fa-check"></i><b>1.4.1</b> DUXC and DUX4</a></li>
<li class="chapter" data-level="1.4.2" data-path="chipseq.html"><a href="chipseq.html#duxc-and-duxc-alt"><i class="fa fa-check"></i><b>1.4.2</b> DUXC and DUXC-ALT</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="chipseq.html"><a href="chipseq.html#motifs-analysis"><i class="fa fa-check"></i><b>1.5</b> Motifs analysis</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="makeSE.html"><a href="makeSE.html"><i class="fa fa-check"></i><b>2</b> RNA-seq to DESeq2</a><ul>
<li class="chapter" data-level="2.1" data-path="makeSE.html"><a href="makeSE.html#rna-seq-preprocessing"><i class="fa fa-check"></i><b>2.1</b> RNA-seq preprocessing</a></li>
<li class="chapter" data-level="2.2" data-path="makeSE.html"><a href="makeSE.html#reads-counts"><i class="fa fa-check"></i><b>2.2</b> Reads counts</a><ul>
<li class="chapter" data-level="2.2.1" data-path="makeSE.html"><a href="makeSE.html#define-gene-features"><i class="fa fa-check"></i><b>2.2.1</b> Define gene features</a></li>
<li class="chapter" data-level="2.2.2" data-path="makeSE.html"><a href="makeSE.html#read-counts"><i class="fa fa-check"></i><b>2.2.2</b> Read counts</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="makeSE.html"><a href="makeSE.html#deseq2"><i class="fa fa-check"></i><b>2.3</b> DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html"><i class="fa fa-check"></i><b>3</b> DUXC transcriptome</a><ul>
<li class="chapter" data-level="3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#load-datasets-and-settings"><i class="fa fa-check"></i><b>3.1</b> Load datasets and settings</a></li>
<li class="chapter" data-level="3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#deseq2-1"><i class="fa fa-check"></i><b>3.2</b> DESeq2</a><ul>
<li class="chapter" data-level="3.2.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hypothesis-settings"><i class="fa fa-check"></i><b>3.2.1</b> hypothesis settings</a></li>
<li class="chapter" data-level="3.2.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cinc-duxc-expresison-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2.2</b> CinC: DUXC expresison in canine muscle cells</a></li>
<li class="chapter" data-level="3.2.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hinc-dux4-expression-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2.3</b> HinC: DUX4 expression in canine muscle cells</a></li>
<li class="chapter" data-level="3.2.4" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#duxc-alt-expression---caltinc"><i class="fa fa-check"></i><b>3.2.4</b> DUXC-ALT expression - CALTinC</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cross-species-models-dux4-expression-in-canine-and-murine"><i class="fa fa-check"></i><b>3.3</b> Cross-species models: DUX4 expression in canine and murine</a><ul>
<li class="chapter" data-level="3.3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-duxc-expression-in-canine"><i class="fa fa-check"></i><b>3.3.1</b> DUX4 and DUXC expression in canine</a></li>
<li class="chapter" data-level="3.3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-dux-expression-in-murine"><i class="fa fa-check"></i><b>3.3.2</b> DUX4 and Dux expression in murine</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html"><i class="fa fa-check"></i><b>4</b> DUXC and Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#duxc-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.1</b> DUXC transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs"><i class="fa fa-check"></i><b>4.1.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.1.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome"><i class="fa fa-check"></i><b>4.1.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#dux-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.2</b> Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.2.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs-1"><i class="fa fa-check"></i><b>4.2.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.2.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome-1"><i class="fa fa-check"></i><b>4.2.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html"><i class="fa fa-check"></i><b>5</b> DUXC transcriptome compared to early embryo transcriptome profile</a><ul>
<li class="chapter" data-level="5.1" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#mouse-2c-like-signature"><i class="fa fa-check"></i><b>5.1</b> Mouse 2C-like signature</a></li>
<li class="chapter" data-level="5.2" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#cleavage-stage-signature"><i class="fa fa-check"></i><b>5.2</b> cleavage-stage signature</a></li>
<li class="chapter" data-level="5.3" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#ipscs"><i class="fa fa-check"></i><b>5.3</b> iPSCs</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="repeats.html"><a href="repeats.html"><i class="fa fa-check"></i><b>6</b> Repeat elements</a><ul>
<li class="chapter" data-level="6.1" data-path="repeats.html"><a href="repeats.html#takeaway"><i class="fa fa-check"></i><b>6.1</b> Takeaway</a></li>
<li class="chapter" data-level="6.2" data-path="repeats.html"><a href="repeats.html#repeat-element-expression"><i class="fa fa-check"></i><b>6.2</b> Repeat element expression</a></li>
<li class="chapter" data-level="6.3" data-path="repeats.html"><a href="repeats.html#differential-analysis"><i class="fa fa-check"></i><b>6.3</b> Differential analysis</a><ul>
<li class="chapter" data-level="6.3.1" data-path="repeats.html"><a href="repeats.html#normalization"><i class="fa fa-check"></i><b>6.3.1</b> Normalization</a></li>
<li class="chapter" data-level="6.3.2" data-path="repeats.html"><a href="repeats.html#deseq2-results"><i class="fa fa-check"></i><b>6.3.2</b> DESeq2 results</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="repeats.html"><a href="repeats.html#repeat-family-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> Repeat family enrichment analysis</a><ul>
<li class="chapter" data-level="6.4.1" data-path="repeats.html"><a href="repeats.html#correlation-between-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>6.4.1</b> Correlation between HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="6.4.2" data-path="repeats.html"><a href="repeats.html#repeat-family-activation"><i class="fa fa-check"></i><b>6.4.2</b> Repeat family activation</a></li>
<li class="chapter" data-level="6.4.3" data-path="repeats.html"><a href="repeats.html#repeat-classs-activation"><i class="fa fa-check"></i><b>6.4.3</b> Repeat classs activation</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="repeats.html"><a href="repeats.html#ltr-retroelements-distribution-and-activation"><i class="fa fa-check"></i><b>6.5</b> LTR retroelements distribution and activation</a></li>
<li class="chapter" data-level="6.6" data-path="repeats.html"><a href="repeats.html#peaks-distribution-in-class-of-repeat-elements"><i class="fa fa-check"></i><b>6.6</b> Peaks distribution in class of repeat elements</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="LTR-activation.html"><a href="LTR-activation.html"><i class="fa fa-check"></i><b>7</b> LTR activation</a><ul>
<li class="chapter" data-level="7.1" data-path="LTR-activation.html"><a href="LTR-activation.html#takaway"><i class="fa fa-check"></i><b>7.1</b> Takaway</a></li>
<li class="chapter" data-level="7.2" data-path="LTR-activation.html"><a href="LTR-activation.html#background-in-murine-and-human"><i class="fa fa-check"></i><b>7.2</b> background in murine and human</a></li>
<li class="chapter" data-level="7.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-subfamily-activation-in-canine"><i class="fa fa-check"></i><b>7.3</b> ERVL subfamily activation in canine</a></li>
<li class="chapter" data-level="7.4" data-path="LTR-activation.html"><a href="LTR-activation.html#malr-activation-in-canine"><i class="fa fa-check"></i><b>7.4</b> MaLR activation in canine</a></li>
<li class="chapter" data-level="7.5" data-path="LTR-activation.html"><a href="LTR-activation.html#compare-wih-hinh"><i class="fa fa-check"></i><b>7.5</b> Compare wih HinH</a><ul>
<li class="chapter" data-level="7.5.1" data-path="LTR-activation.html"><a href="LTR-activation.html#copy-number"><i class="fa fa-check"></i><b>7.5.1</b> Copy number</a></li>
<li class="chapter" data-level="7.5.2" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-malr"><i class="fa fa-check"></i><b>7.5.2</b> ERVL-MaLR</a></li>
<li class="chapter" data-level="7.5.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl"><i class="fa fa-check"></i><b>7.5.3</b> ERVL</a></li>
<li class="chapter" data-level="7.5.4" data-path="LTR-activation.html"><a href="LTR-activation.html#erv1"><i class="fa fa-check"></i><b>7.5.4</b> ERV1</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="LTR-activation.html"><a href="LTR-activation.html#duxc-alt-and-ervl-transcriptions"><i class="fa fa-check"></i><b>7.6</b> DUXC-ALT and ERVL transcriptions</a><ul>
<li class="chapter" data-level="7.6.1" data-path="LTR-activation.html"><a href="LTR-activation.html#erv-subfamily-expression"><i class="fa fa-check"></i><b>7.6.1</b> ERV subfamily expression</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="non-LTR.html"><a href="non-LTR.html"><i class="fa fa-check"></i><b>8</b> Non-LTR retrotrasposons and satellites</a><ul>
<li class="chapter" data-level="8.1" data-path="non-LTR.html"><a href="non-LTR.html#lines"><i class="fa fa-check"></i><b>8.1</b> LINEs</a><ul>
<li class="chapter" data-level="8.1.1" data-path="non-LTR.html"><a href="non-LTR.html#l1s-activation"><i class="fa fa-check"></i><b>8.1.1</b> L1s activation</a></li>
<li class="chapter" data-level="8.1.2" data-path="non-LTR.html"><a href="non-LTR.html#line1-and-duxcdux4-binding-sites"><i class="fa fa-check"></i><b>8.1.2</b> LINE1 and DUXC/DUX4 binding sites</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="non-LTR.html"><a href="non-LTR.html#sine"><i class="fa fa-check"></i><b>8.2</b> SINE</a><ul>
<li class="chapter" data-level="8.2.1" data-path="non-LTR.html"><a href="non-LTR.html#distribution-of-sine-family"><i class="fa fa-check"></i><b>8.2.1</b> Distribution of SINE family</a></li>
<li class="chapter" data-level="8.2.2" data-path="non-LTR.html"><a href="non-LTR.html#expression-of-sines-in-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>8.2.2</b> Expression of SINEs in HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="8.2.3" data-path="non-LTR.html"><a href="non-LTR.html#trna-lys"><i class="fa fa-check"></i><b>8.2.3</b> tRNA-Lys</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="non-LTR.html"><a href="non-LTR.html#satellites"><i class="fa fa-check"></i><b>8.3</b> Satellites</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="peaks-line1.html"><a href="peaks-line1.html"><i class="fa fa-check"></i><b>9</b> DUXC peaks and LINE1s</a><ul>
<li class="chapter" data-level="9.1" data-path="peaks-line1.html"><a href="peaks-line1.html#narrow-peaks-and-expressed-l1s"><i class="fa fa-check"></i><b>9.1</b> Narrow Peaks and expressed L1s</a><ul>
<li class="chapter" data-level="9.1.1" data-path="peaks-line1.html"><a href="peaks-line1.html#motif-discovery-using-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.1</b> Motif discovery using L1-overlapped peaks</a></li>
<li class="chapter" data-level="9.1.2" data-path="peaks-line1.html"><a href="peaks-line1.html#match-dux4-and-duxc-motif-to-l1-overapped-peaks"><i class="fa fa-check"></i><b>9.1.2</b> Match DUX4 and DUXC motif to L1-overapped peaks</a></li>
<li class="chapter" data-level="9.1.3" data-path="peaks-line1.html"><a href="peaks-line1.html#location-of-aatca-on-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.3</b> Location of AATCA on L1-overlapped peaks</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="peaks-line1.html"><a href="peaks-line1.html#line1s-and-aatca"><i class="fa fa-check"></i><b>9.2</b> LINE1s and AATCA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i># References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="DUXC-FSHD-transcriptome" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> DUXC and Dux transcriptome compared to that of FSHD</h1>
<p>We used a gene set enrichment method to exam whether the FSHD profile was enriched by DUXC transcriptome in the canine model. The results suggested that DUXC induced transcription program similar to FSHD muscle cells. We employed the FSHD transcriptome profile discovered by Jagannathan et al. <span class="citation">(Sujatha Jagannathan and Tapscott <a href="#ref-suja2016">2016</a>)</span> - there were 666 mis-expressed genes in FSHD muscle cell lines. Out of 666, 520 have human-canine paired homologous (by Ensembl orthologs database V88); we used hypergeometric tests to exam whether the FSHD transcription program is enriched in canine.</p>
<p><strong>Results.</strong> Based on the 520 human-to-canine homologous, the Pearson correlation between DUXC transcriptome and FSHD transcriptome is 0.35, moderately higher than that of between mouse Dux and FSHD transcriptome (0.29). Out of the 520 homologous, 146 (29%) were positively affected by DUXC in canine; the hypergeometric test yielded <span class="math inline">\(1e-12\)</span> <span class="math inline">\(p\)</span>-value, suggesting that that the mis-expressed gene set of FSHD was enriched.</p>
<p>Below are the codes implementing the statistics methods.</p>
<p><strong>Load datasets</strong></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">pkg_dir &lt;-<span class="st"> &quot;~/CompBio/canFam3.DuxFamily&quot;</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">fig_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;figures&quot;</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">data_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">pkg_dir &lt;-<span class="st"> &quot;~/CompBio/canFam3.DuxFamily&quot;</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">source</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;scripts&quot;</span>, <span class="st">&quot;load_lib_dataset.R&quot;</span>))</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;canine_human_mouse_ensOrtholog.rda&quot;</span>))</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;human_mouse_ensOrtholog.rda&quot;</span>))</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(ggplot2))</a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(latex2exp))</a></code></pre></div>
<div id="duxc-transcriptome-compared-to-that-of-fshd" class="section level2">
<h2><span class="header-section-number">4.1</span> DUXC transcriptome compared to that of FSHD</h2>
<div id="get-canine-human-homologs" class="section level3">
<h3><span class="header-section-number">4.1.1</span> Get canine-human homologs</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># read the table made by Jagannathan 2016</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">HinH_de &lt;-<span class="st"> </span><span class="kw">read_csv</span>(<span class="dt">file=</span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;extdata&quot;</span>, </a>
<a class="sourceLine" id="cb2-3" data-line-number="3">                                   <span class="st">&quot;Jagannathan-2016-sup-table-1-lfc.csv&quot;</span>), <span class="dt">skip =</span> <span class="dv">2</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(iDUX4_logFC <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>, iDUX4_pval <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(ENSEMBL, iDUX4_logFC, iDUX4_pval) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="st">  </span><span class="kw">rename</span>(<span class="dt">HUMAN_ENSEMBL=</span>ENSEMBL, <span class="dt">HinH_logFC=</span>iDUX4_logFC, <span class="dt">HinH_padj=</span>iDUX4_pval) </a>
<a class="sourceLine" id="cb2-7" data-line-number="7"></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co"># create canine-to-human ortholog data.frame, restrict to hDUX4 transcriptome (FSHD profile)</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9">canine_human_homolog &lt;-<span class="st"> </span>canine_human_mouse_ensOrtholog <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(ensembl_gene_id, hsapiens_homolog_ensembl_gene, hsapiens_homolog_associated_gene_name) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="st">  </span><span class="kw">rename</span>(<span class="dt">CANINE_ENSEMBL=</span>ensembl_gene_id, <span class="dt">HUMAN_ENSEMBL=</span>hsapiens_homolog_ensembl_gene) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># filter out empty string</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(HUMAN_ENSEMBL <span class="op">!=</span><span class="st"> &quot;&quot;</span>, <span class="op">!</span><span class="kw">duplicated</span>(HUMAN_ENSEMBL), <span class="op">!</span><span class="kw">duplicated</span>(CANINE_ENSEMBL)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="st">  </span><span class="kw">inner_join</span>(HinH_de, <span class="dt">by=</span><span class="st">&quot;HUMAN_ENSEMBL&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="st">  </span><span class="kw">left_join</span>(CinC, <span class="dt">by=</span><span class="st">&quot;CANINE_ENSEMBL&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="st">  </span><span class="kw">left_join</span>(HinC, <span class="dt">by=</span><span class="st">&quot;CANINE_ENSEMBL&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">CinC_up =</span> <span class="kw">ifelse</span>(CinC_padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>CinC_logFC <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="st">&quot;YES&quot;</span>, <span class="st">&quot;NO&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">HinC_up =</span> <span class="kw">ifelse</span>(HinC_padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>HinC_logFC <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="st">&quot;YES&quot;</span>, <span class="st">&quot;NO&quot;</span>))</a></code></pre></div>
</div>
<div id="gsea-enrichment-of-fshd-transcriptome" class="section level3">
<h3><span class="header-section-number">4.1.2</span> GSEA: enrichment of FSHD transcriptome</h3>
<p>Code below calculates the correlation (Pearson) and implements GSEA inferring FSHD transcriptome enrichment.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># GSEA: universe=; selected=; FSHD set=1735; picked=363;</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">white_drawn =<span class="st"> </span><span class="kw">sum</span>(canine_human_homolog<span class="op">$</span>CinC_up<span class="op">==</span><span class="st">&quot;YES&quot;</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>) <span class="co"># up-regulated in FSHD profile</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3">white_ball  =<span class="st"> </span>CinC <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(CinC_logFC <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, CinC_padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">nrow</span>() <span class="co"># up-regulated </span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">black_ball  =<span class="st"> </span><span class="kw">nrow</span>(CinC) <span class="op">-</span><span class="st"> </span>white_ball <span class="co"># universe - up-regulated</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5">ball_drawn  =<span class="st"> </span><span class="kw">nrow</span>(canine_human_homolog)  <span class="co"># how many are in FSHD profile</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6">prob &lt;-<span class="st"> </span><span class="kw">dhyper</span>(<span class="dt">x=</span>white_drawn, </a>
<a class="sourceLine" id="cb3-7" data-line-number="7">               <span class="dt">m=</span>white_ball, <span class="dt">n=</span>black_ball, <span class="dt">k=</span>ball_drawn, <span class="dt">log =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co"># correlation: pearson</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9">pearson &lt;-<span class="st"> </span>canine_human_homolog <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(CinC_logFC)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">pearson =</span> <span class="kw">cor</span>(HinH_logFC, CinC_logFC)) </a>
<a class="sourceLine" id="cb3-11" data-line-number="11">label &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;Up-regulated in CinC = %2.0f / %1.0f </span><span class="ch">\n</span><span class="st">pearson = %0.2f </span><span class="ch">\n</span><span class="st">GSEA p-value &lt; 1e-12&quot;</span>, </a>
<a class="sourceLine" id="cb3-12" data-line-number="12">                 <span class="kw">sum</span>(canine_human_homolog<span class="op">$</span>CinC_up<span class="op">==</span><span class="st">&quot;YES&quot;</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>), <span class="kw">nrow</span>(canine_human_homolog), pearson)</a>
<a class="sourceLine" id="cb3-13" data-line-number="13"></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="kw">ggplot</span>(canine_human_homolog, <span class="kw">aes</span>(<span class="dt">x=</span>CinC_logFC, <span class="dt">y=</span>HinH_logFC, <span class="dt">color=</span>CinC_up)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="st">  </span><span class="co">#geom_abline(intercept=0, slope=1, color=&quot;gray75&quot;, alpha=0.2) +</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">se=</span><span class="ot">FALSE</span>, <span class="dt">color=</span><span class="st">&quot;gray75&quot;</span>, <span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha=</span><span class="fl">0.7</span>, <span class="dt">size=</span><span class="fl">1.2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">x=</span><span class="op">-</span><span class="dv">2</span>, <span class="dt">y=</span><span class="fl">8.8</span>, <span class="dt">label=</span>label, <span class="dt">hjust=</span><span class="dv">0</span>, <span class="dt">vjust=</span><span class="dv">1</span>, <span class="dt">color=</span><span class="st">&quot;gray25&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;ivory4&quot;</span>, <span class="st">&quot;firebrick4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;FSHD signature homologs: CinC vs. HinH&quot;</span>,</a>
<a class="sourceLine" id="cb3-23" data-line-number="23">       <span class="dt">y=</span><span class="kw">TeX</span>(<span class="st">&quot;HinH: $</span><span class="ch">\\</span><span class="st">log_2$(DUX4/no DUX4) in human&quot;</span>),</a>
<a class="sourceLine" id="cb3-24" data-line-number="24">       <span class="dt">x=</span><span class="kw">TeX</span>(<span class="st">&quot;CinC: $</span><span class="ch">\\</span><span class="st">log_2$(DUXC/luciferase) in canine&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>, <span class="dt">face=</span><span class="st">&quot;bold&quot;</span>))</a></code></pre></div>
<pre><code>## Warning: Removed 35 rows containing non-finite values (stat_smooth).</code></pre>
<pre><code>## Warning: Removed 35 rows containing missing values (geom_point).</code></pre>
<div class="figure"><span id="fig:viz-canine-human-homolog-CinC"></span>
<img src="gitbook_files/figure-html/viz-canine-human-homolog-CinC-1.png" alt="Scatter plot of 520 homologs expression in  DUXC expressed canine myobrast and FSHD muscle cells. Dashlines represents the linear regression." width="672" />
<p class="caption">
Figure 4.1: Scatter plot of 520 homologs expression in DUXC expressed canine myobrast and FSHD muscle cells. Dashlines represents the linear regression.
</p>
</div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="dt">filename=</span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;figures&quot;</span>,</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">                          <span class="st">&quot;HinH_CinC_FSHD_signature_homologs.pdf&quot;</span>),</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">       <span class="dt">width=</span><span class="dv">5</span>, <span class="dt">height=</span><span class="fl">3.75</span>)</a></code></pre></div>
<pre><code>## Warning: Removed 35 rows containing non-finite values (stat_smooth).

## Warning: Removed 35 rows containing missing values (geom_point).</code></pre>
</div>
</div>
<div id="dux-transcriptome-compared-to-that-of-fshd" class="section level2">
<h2><span class="header-section-number">4.2</span> Dux transcriptome compared to that of FSHD</h2>
<div id="get-canine-human-homologs-1" class="section level3">
<h3><span class="header-section-number">4.2.1</span> Get canine-human homologs</h3>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">human_mouse_homolog &lt;-<span class="st"> </span>human_mouse_ensOrtholog <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(ensembl_gene_id, external_gene_name, </a>
<a class="sourceLine" id="cb8-3" data-line-number="3">                mmusculus_homolog_ensembl_gene, mmusculus_homolog_associated_gene_name) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="st">  </span><span class="kw">rename</span>(<span class="dt">HUMAN_ENSEMBL=</span>ensembl_gene_id, <span class="dt">HUMAN_gene_name=</span>external_gene_name ,</a>
<a class="sourceLine" id="cb8-5" data-line-number="5">         <span class="dt">MOUSE_ENSEMBL=</span>mmusculus_homolog_ensembl_gene,   </a>
<a class="sourceLine" id="cb8-6" data-line-number="6">         <span class="dt">MOUSE_gene_name=</span>mmusculus_homolog_associated_gene_name) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(HUMAN_ENSEMBL <span class="op">!=</span><span class="st"> &quot;&quot;</span>, <span class="op">!</span><span class="kw">duplicated</span>(HUMAN_ENSEMBL), <span class="op">!</span><span class="kw">duplicated</span>(MOUSE_ENSEMBL)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="st">  </span><span class="kw">inner_join</span>(HinH_de, <span class="dt">by=</span><span class="st">&quot;HUMAN_ENSEMBL&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="st">  </span><span class="kw">left_join</span>(MinM, <span class="dt">by=</span><span class="st">&quot;MOUSE_ENSEMBL&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="st">  </span><span class="kw">left_join</span>(HinM, <span class="dt">by=</span><span class="st">&quot;MOUSE_ENSEMBL&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">MinM_up =</span> <span class="kw">ifelse</span>(MinM_padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>MinM_logFC <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="st">&quot;YES&quot;</span>, <span class="st">&quot;NO&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">HinM_up =</span> <span class="kw">ifelse</span>(HinM_padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>HinM_logFC <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="st">&quot;YES&quot;</span>, <span class="st">&quot;NO&quot;</span>))</a></code></pre></div>
</div>
<div id="gsea-enrichment-of-fshd-transcriptome-1" class="section level3">
<h3><span class="header-section-number">4.2.2</span> GSEA: enrichment of FSHD transcriptome</h3>
<p>Out of 666 mis-expressed genes in FSHD muscle cells, 543 has homologs in mouse; 97 out of 534 homologs were up-regulated in Dux expressed mouse cell line. The correlation of 534 homologs expression between Dux expressed mouse cell line and FSHD muscle cells is 0.29. The GSEA by hypergeometric test gave <span class="math inline">\(p\)</span>-value &lt; <span class="math inline">\(1e-12\)</span> suggesting that Dux also induced similar transcription activity similar to FSHD muscle cells.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co"># GSEA</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">white_drawn =<span class="st"> </span><span class="kw">sum</span>(human_mouse_homolog<span class="op">$</span>MinM_up<span class="op">==</span><span class="st">&quot;YES&quot;</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>) <span class="co"># up-regulated in FSHD profile</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3">white_ball  =<span class="st"> </span>MinM <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(MinM_logFC <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, MinM_padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">nrow</span>() <span class="co"># up-regulated </span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4">black_ball  =<span class="st"> </span><span class="kw">nrow</span>(MinM) <span class="op">-</span><span class="st"> </span>white_ball <span class="co"># universe - up-regulated</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5">ball_drawn  =<span class="st"> </span><span class="kw">nrow</span>(human_mouse_homolog)  <span class="co"># how many are in FSHD profile</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6">prob &lt;-<span class="st"> </span><span class="kw">dhyper</span>(<span class="dt">x=</span>white_drawn, </a>
<a class="sourceLine" id="cb9-7" data-line-number="7">               <span class="dt">m=</span>white_ball, <span class="dt">n=</span>black_ball, <span class="dt">k=</span>ball_drawn, <span class="dt">log =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co"># correlation: pearson</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9">pearson &lt;-<span class="st"> </span>human_mouse_homolog <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(MinM_logFC)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">pearson =</span> <span class="kw">cor</span>(HinH_logFC, MinM_logFC)) </a>
<a class="sourceLine" id="cb9-11" data-line-number="11">label &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">&quot;Up-regulated in MinM = %2.0f / %1.0f </span><span class="ch">\n</span><span class="st">pearson = %0.2f </span><span class="ch">\n</span><span class="st">GSEA p-value &lt; 1e-12&quot;</span>, </a>
<a class="sourceLine" id="cb9-12" data-line-number="12">                 <span class="kw">sum</span>(human_mouse_homolog<span class="op">$</span>MinM_up<span class="op">==</span><span class="st">&quot;YES&quot;</span>, <span class="dt">na.rm=</span><span class="ot">TRUE</span>), <span class="kw">nrow</span>(human_mouse_homolog), pearson)</a>
<a class="sourceLine" id="cb9-13" data-line-number="13"></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="kw">ggplot</span>(human_mouse_homolog, <span class="kw">aes</span>(<span class="dt">x=</span>MinM_logFC, <span class="dt">y=</span>HinH_logFC, <span class="dt">color=</span>MinM_up)) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="st">  </span><span class="co">#geom_abline(intercept=0, slope=1, color=&quot;gray75&quot;, alpha=0.2) +</span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="dt">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="dt">se=</span><span class="ot">FALSE</span>, <span class="dt">color=</span><span class="st">&quot;gray75&quot;</span>, <span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha=</span><span class="fl">0.7</span>, <span class="dt">size=</span><span class="fl">1.2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">x=</span><span class="op">-</span><span class="fl">4.5</span>, <span class="dt">y=</span><span class="fl">7.5</span>, <span class="dt">label=</span>label, <span class="dt">hjust=</span><span class="dv">0</span>, <span class="dt">vjust=</span><span class="dv">1</span>, </a>
<a class="sourceLine" id="cb9-19" data-line-number="19">            <span class="kw">aes</span>(<span class="dt">fontface=</span><span class="st">&quot;plain&quot;</span>), <span class="dt">color=</span><span class="st">&quot;gray25&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-20" data-line-number="20"><span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb9-21" data-line-number="21"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-22" data-line-number="22"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;ivory4&quot;</span>, <span class="st">&quot;firebrick4&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-23" data-line-number="23"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;FSHD signature homologs: MinM vs. HinH&quot;</span>,</a>
<a class="sourceLine" id="cb9-24" data-line-number="24">       <span class="dt">y=</span><span class="kw">TeX</span>(<span class="st">&quot;HinH: $</span><span class="ch">\\</span><span class="st">log_2$(DUX4/no DUX4) in human&quot;</span>),</a>
<a class="sourceLine" id="cb9-25" data-line-number="25">       <span class="dt">x=</span><span class="kw">TeX</span>(<span class="st">&quot;MinM: $</span><span class="ch">\\</span><span class="st">log_2$(Dux/no Dux) in mouse&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-26" data-line-number="26"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>, <span class="dt">face=</span><span class="st">&quot;bold&quot;</span>))</a></code></pre></div>
<pre><code>## Warning: Removed 75 rows containing non-finite values (stat_smooth).</code></pre>
<pre><code>## Warning: Removed 91 rows containing missing values (geom_point).</code></pre>
<div class="figure"><span id="fig:viz-human-mouse-homolog-MinM"></span>
<img src="gitbook_files/figure-html/viz-human-mouse-homolog-MinM-1.png" alt="Scatter plot of 534 homologs expression in Dux expressed mouse myobrast and FSHD muscle cells. Dashlines represents the linear regression." width="672" />
<p class="caption">
Figure 4.2: Scatter plot of 534 homologs expression in Dux expressed mouse myobrast and FSHD muscle cells. Dashlines represents the linear regression.
</p>
</div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="dt">filename=</span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;figures&quot;</span>,</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">                          <span class="st">&quot;HinH_MinM_FSHD_signature_homologs.pdf&quot;</span>),</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">       <span class="dt">width=</span><span class="dv">5</span>, <span class="dt">height=</span><span class="fl">3.75</span>)</a></code></pre></div>
<pre><code>## Warning: Removed 75 rows containing non-finite values (stat_smooth).

## Warning: Removed 91 rows containing missing values (geom_point).</code></pre>

</div>
</div>
</div>
<h3># References</h3>
<div id="refs" class="references">
<div id="ref-suja2016">
<p>Sujatha Jagannathan, Rebecca Resnick, Sean C. Shadle, and Stephen J. Tapscott. 2016. Model Systems of Dux4 Expression Recapitulate the Transcriptional Profile of Fshd Cells. <em>Human Molecular Genetics</em>. <a href="https://doi.org/doi: 10.1093/hmg/ddw271">https://doi.org/doi: 10.1093/hmg/ddw271</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="DUXC-transcriptome.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="DUXC-early-embryo.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/FredHutch/canFam3.DuxFamily/tree/master/gitbook/04-DUXC-compared-to-FSHD-profile.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/FredHutch/canFam3.DuxFamily/tree/master/gitbook/04-DUXC-compared-to-FSHD-profile.Rmd",
"text": null
},
"download": ["gitbook.pdf", "gitbook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
