<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 DUXC transcriptome | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</title>
  <meta name="description" content="Bioinformatics and statistics analysis" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 DUXC transcriptome | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Bioinformatics and statistics analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 DUXC transcriptome | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD" />
  
  <meta name="twitter:description" content="Bioinformatics and statistics analysis" />
  

<meta name="author" content="Chao-Jen Wong" />


<meta name="date" content="2021-11-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="makeSE.html"/>
<link rel="next" href="DUXC-FSHD-transcriptome.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DUXC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#datasets"><i class="fa fa-check"></i><b>0.1</b> Datasets</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#annotation"><i class="fa fa-check"></i><b>0.2</b> Annotation</a><ul>
<li class="chapter" data-level="0.2.1" data-path="index.html"><a href="index.html#canfam3-ensembl-gtf"><i class="fa fa-check"></i><b>0.2.1</b> canFam3 ensembl GTF</a></li>
</ul></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>0.3</b> Software</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="chipseq.html"><a href="chipseq.html"><i class="fa fa-check"></i><b>1</b> ChIP-seq for the canine model</a><ul>
<li class="chapter" data-level="1.1" data-path="chipseq.html"><a href="chipseq.html#chip-seq-samples"><i class="fa fa-check"></i><b>1.1</b> ChIP-seq samples</a></li>
<li class="chapter" data-level="1.2" data-path="chipseq.html"><a href="chipseq.html#analysis-workflow"><i class="fa fa-check"></i><b>1.2</b> Analysis workflow</a></li>
<li class="chapter" data-level="1.3" data-path="chipseq.html"><a href="chipseq.html#peaks-and-functional-distribution"><i class="fa fa-check"></i><b>1.3</b> Peaks and functional distribution</a><ul>
<li class="chapter" data-level="1.3.1" data-path="chipseq.html"><a href="chipseq.html#number-of-peaks"><i class="fa fa-check"></i><b>1.3.1</b> Number of peaks</a></li>
<li class="chapter" data-level="1.3.2" data-path="chipseq.html"><a href="chipseq.html#functional-discribution"><i class="fa fa-check"></i><b>1.3.2</b> Functional discribution</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chipseq.html"><a href="chipseq.html#overlap-of-peaks"><i class="fa fa-check"></i><b>1.4</b> Overlap of peaks</a><ul>
<li class="chapter" data-level="1.4.1" data-path="chipseq.html"><a href="chipseq.html#duxc-and-dux4"><i class="fa fa-check"></i><b>1.4.1</b> DUXC and DUX4</a></li>
<li class="chapter" data-level="1.4.2" data-path="chipseq.html"><a href="chipseq.html#duxc-and-duxc-alt"><i class="fa fa-check"></i><b>1.4.2</b> DUXC and DUXC-ALT</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="chipseq.html"><a href="chipseq.html#motifs-analysis"><i class="fa fa-check"></i><b>1.5</b> Motifs analysis</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="makeSE.html"><a href="makeSE.html"><i class="fa fa-check"></i><b>2</b> RNA-seq to DESeq2</a><ul>
<li class="chapter" data-level="2.1" data-path="makeSE.html"><a href="makeSE.html#rna-seq-preprocessing"><i class="fa fa-check"></i><b>2.1</b> RNA-seq preprocessing</a></li>
<li class="chapter" data-level="2.2" data-path="makeSE.html"><a href="makeSE.html#reads-counts"><i class="fa fa-check"></i><b>2.2</b> Reads counts</a><ul>
<li class="chapter" data-level="2.2.1" data-path="makeSE.html"><a href="makeSE.html#define-gene-features"><i class="fa fa-check"></i><b>2.2.1</b> Define gene features</a></li>
<li class="chapter" data-level="2.2.2" data-path="makeSE.html"><a href="makeSE.html#read-counts"><i class="fa fa-check"></i><b>2.2.2</b> Read counts</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="makeSE.html"><a href="makeSE.html#deseq2"><i class="fa fa-check"></i><b>2.3</b> DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html"><i class="fa fa-check"></i><b>3</b> DUXC transcriptome</a><ul>
<li class="chapter" data-level="3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#load-datasets-and-settings"><i class="fa fa-check"></i><b>3.1</b> Load datasets and settings</a></li>
<li class="chapter" data-level="3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#deseq2-1"><i class="fa fa-check"></i><b>3.2</b> DESeq2</a><ul>
<li class="chapter" data-level="3.2.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hypothesis-settings"><i class="fa fa-check"></i><b>3.2.1</b> hypothesis settings</a></li>
<li class="chapter" data-level="3.2.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cinc-duxc-expresison-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2.2</b> CinC: DUXC expresison in canine muscle cells</a></li>
<li class="chapter" data-level="3.2.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hinc-dux4-expression-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2.3</b> HinC: DUX4 expression in canine muscle cells</a></li>
<li class="chapter" data-level="3.2.4" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#duxc-alt-expression---caltinc"><i class="fa fa-check"></i><b>3.2.4</b> DUXC-ALT expression - CALTinC</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cross-species-models-dux4-expression-in-canine-and-murine"><i class="fa fa-check"></i><b>3.3</b> Cross-species models: DUX4 expression in canine and murine</a><ul>
<li class="chapter" data-level="3.3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-duxc-expression-in-canine"><i class="fa fa-check"></i><b>3.3.1</b> DUX4 and DUXC expression in canine</a></li>
<li class="chapter" data-level="3.3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-dux-expression-in-murine"><i class="fa fa-check"></i><b>3.3.2</b> DUX4 and Dux expression in murine</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html"><i class="fa fa-check"></i><b>4</b> DUXC and Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#duxc-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.1</b> DUXC transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs"><i class="fa fa-check"></i><b>4.1.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.1.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome"><i class="fa fa-check"></i><b>4.1.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#dux-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.2</b> Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.2.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs-1"><i class="fa fa-check"></i><b>4.2.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.2.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome-1"><i class="fa fa-check"></i><b>4.2.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html"><i class="fa fa-check"></i><b>5</b> DUXC transcriptome compared to early embryo transcriptome profile</a><ul>
<li class="chapter" data-level="5.1" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#mouse-2c-like-signature"><i class="fa fa-check"></i><b>5.1</b> Mouse 2C-like signature</a></li>
<li class="chapter" data-level="5.2" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#cleavage-stage-signature"><i class="fa fa-check"></i><b>5.2</b> cleavage-stage signature</a></li>
<li class="chapter" data-level="5.3" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#ipscs"><i class="fa fa-check"></i><b>5.3</b> iPSCs</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="repeats.html"><a href="repeats.html"><i class="fa fa-check"></i><b>6</b> Repeat elements</a><ul>
<li class="chapter" data-level="6.1" data-path="repeats.html"><a href="repeats.html#takeaway"><i class="fa fa-check"></i><b>6.1</b> Takeaway</a></li>
<li class="chapter" data-level="6.2" data-path="repeats.html"><a href="repeats.html#repeat-element-expression"><i class="fa fa-check"></i><b>6.2</b> Repeat element expression</a></li>
<li class="chapter" data-level="6.3" data-path="repeats.html"><a href="repeats.html#differential-analysis"><i class="fa fa-check"></i><b>6.3</b> Differential analysis</a><ul>
<li class="chapter" data-level="6.3.1" data-path="repeats.html"><a href="repeats.html#normalization"><i class="fa fa-check"></i><b>6.3.1</b> Normalization</a></li>
<li class="chapter" data-level="6.3.2" data-path="repeats.html"><a href="repeats.html#deseq2-results"><i class="fa fa-check"></i><b>6.3.2</b> DESeq2 results</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="repeats.html"><a href="repeats.html#repeat-family-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> Repeat family enrichment analysis</a><ul>
<li class="chapter" data-level="6.4.1" data-path="repeats.html"><a href="repeats.html#correlation-between-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>6.4.1</b> Correlation between HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="6.4.2" data-path="repeats.html"><a href="repeats.html#repeat-family-activation"><i class="fa fa-check"></i><b>6.4.2</b> Repeat family activation</a></li>
<li class="chapter" data-level="6.4.3" data-path="repeats.html"><a href="repeats.html#repeat-classs-activation"><i class="fa fa-check"></i><b>6.4.3</b> Repeat classs activation</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="repeats.html"><a href="repeats.html#ltr-retroelements-distribution-and-activation"><i class="fa fa-check"></i><b>6.5</b> LTR retroelements distribution and activation</a></li>
<li class="chapter" data-level="6.6" data-path="repeats.html"><a href="repeats.html#peaks-distribution-in-class-of-repeat-elements"><i class="fa fa-check"></i><b>6.6</b> Peaks distribution in class of repeat elements</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="LTR-activation.html"><a href="LTR-activation.html"><i class="fa fa-check"></i><b>7</b> LTR activation</a><ul>
<li class="chapter" data-level="7.1" data-path="LTR-activation.html"><a href="LTR-activation.html#takaway"><i class="fa fa-check"></i><b>7.1</b> Takaway</a></li>
<li class="chapter" data-level="7.2" data-path="LTR-activation.html"><a href="LTR-activation.html#background-in-murine-and-human"><i class="fa fa-check"></i><b>7.2</b> background in murine and human</a></li>
<li class="chapter" data-level="7.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-subfamily-activation-in-canine"><i class="fa fa-check"></i><b>7.3</b> ERVL subfamily activation in canine</a></li>
<li class="chapter" data-level="7.4" data-path="LTR-activation.html"><a href="LTR-activation.html#malr-activation-in-canine"><i class="fa fa-check"></i><b>7.4</b> MaLR activation in canine</a></li>
<li class="chapter" data-level="7.5" data-path="LTR-activation.html"><a href="LTR-activation.html#compare-wih-hinh"><i class="fa fa-check"></i><b>7.5</b> Compare wih HinH</a><ul>
<li class="chapter" data-level="7.5.1" data-path="LTR-activation.html"><a href="LTR-activation.html#copy-number"><i class="fa fa-check"></i><b>7.5.1</b> Copy number</a></li>
<li class="chapter" data-level="7.5.2" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-malr"><i class="fa fa-check"></i><b>7.5.2</b> ERVL-MaLR</a></li>
<li class="chapter" data-level="7.5.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl"><i class="fa fa-check"></i><b>7.5.3</b> ERVL</a></li>
<li class="chapter" data-level="7.5.4" data-path="LTR-activation.html"><a href="LTR-activation.html#erv1"><i class="fa fa-check"></i><b>7.5.4</b> ERV1</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="LTR-activation.html"><a href="LTR-activation.html#duxc-alt-and-ervl-transcriptions"><i class="fa fa-check"></i><b>7.6</b> DUXC-ALT and ERVL transcriptions</a><ul>
<li class="chapter" data-level="7.6.1" data-path="LTR-activation.html"><a href="LTR-activation.html#erv-subfamily-expression"><i class="fa fa-check"></i><b>7.6.1</b> ERV subfamily expression</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="non-LTR.html"><a href="non-LTR.html"><i class="fa fa-check"></i><b>8</b> Non-LTR retrotrasposons and satellites</a><ul>
<li class="chapter" data-level="8.1" data-path="non-LTR.html"><a href="non-LTR.html#lines"><i class="fa fa-check"></i><b>8.1</b> LINEs</a><ul>
<li class="chapter" data-level="8.1.1" data-path="non-LTR.html"><a href="non-LTR.html#l1s-activation"><i class="fa fa-check"></i><b>8.1.1</b> L1s activation</a></li>
<li class="chapter" data-level="8.1.2" data-path="non-LTR.html"><a href="non-LTR.html#line1-and-duxcdux4-binding-sites"><i class="fa fa-check"></i><b>8.1.2</b> LINE1 and DUXC/DUX4 binding sites</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="non-LTR.html"><a href="non-LTR.html#sine"><i class="fa fa-check"></i><b>8.2</b> SINE</a><ul>
<li class="chapter" data-level="8.2.1" data-path="non-LTR.html"><a href="non-LTR.html#distribution-of-sine-family"><i class="fa fa-check"></i><b>8.2.1</b> Distribution of SINE family</a></li>
<li class="chapter" data-level="8.2.2" data-path="non-LTR.html"><a href="non-LTR.html#expression-of-sines-in-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>8.2.2</b> Expression of SINEs in HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="8.2.3" data-path="non-LTR.html"><a href="non-LTR.html#trna-lys"><i class="fa fa-check"></i><b>8.2.3</b> tRNA-Lys</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="non-LTR.html"><a href="non-LTR.html#satellites"><i class="fa fa-check"></i><b>8.3</b> Satellites</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="peaks-line1.html"><a href="peaks-line1.html"><i class="fa fa-check"></i><b>9</b> DUXC peaks and LINE1s</a><ul>
<li class="chapter" data-level="9.1" data-path="peaks-line1.html"><a href="peaks-line1.html#narrow-peaks-and-expressed-l1s"><i class="fa fa-check"></i><b>9.1</b> Narrow Peaks and expressed L1s</a><ul>
<li class="chapter" data-level="9.1.1" data-path="peaks-line1.html"><a href="peaks-line1.html#motif-discovery-using-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.1</b> Motif discovery using L1-overlapped peaks</a></li>
<li class="chapter" data-level="9.1.2" data-path="peaks-line1.html"><a href="peaks-line1.html#match-dux4-and-duxc-motif-to-l1-overapped-peaks"><i class="fa fa-check"></i><b>9.1.2</b> Match DUX4 and DUXC motif to L1-overapped peaks</a></li>
<li class="chapter" data-level="9.1.3" data-path="peaks-line1.html"><a href="peaks-line1.html#location-of-aatca-on-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.3</b> Location of AATCA on L1-overlapped peaks</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="peaks-line1.html"><a href="peaks-line1.html#line1s-and-aatca"><i class="fa fa-check"></i><b>9.2</b> LINE1s and AATCA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i># References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="DUXC-transcriptome" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> DUXC transcriptome</h1>
<p>In this chapter, we present<br />
(1) the hypothesis setting for determining differentailly expressed genes in the canine model<br />
(2) MA plots of DUXC, DUX4 and DUXC-ALT expression in canine<br />
(3) GO analysis for the DUXC-ALT down-regulated genes<br />
(4) analysis for the cross-species models: DUX4 expression in canine and murine</p>
<div id="load-datasets-and-settings" class="section level2">
<h2><span class="header-section-number">3.1</span> Load datasets and settings</h2>
<p>First, we load the <code>DESeqDataSet</code> instances (container of gene expression matrix and metadata) for all canine, mouse, and human models:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(DESeq2))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(tidyverse))</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(ggVennDiagram))</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(org.Cf.eg.db))</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(TxDb.Cfamiliaris.UCSC.canFam3.ensGene))</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(goseq))</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(latex2exp))</a>
<a class="sourceLine" id="cb1-8" data-line-number="8">pkg_dir &lt;-<span class="st"> &quot;~/CompBio/canFam3.DuxFamily&quot;</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9">data_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>)</a>
<a class="sourceLine" id="cb1-10" data-line-number="10">fig_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;figures&quot;</span>)</a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="co"># load datasets </span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="kw">source</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;scripts&quot;</span>, <span class="st">&quot;tools.R&quot;</span>))</a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="kw">load</span>(<span class="kw">file.path</span>(data_dir, <span class="st">&quot;CinC.ens.dds.rda&quot;</span>)) <span class="co"># DUXC in canine</span></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="kw">load</span>(<span class="kw">file.path</span>(data_dir, <span class="st">&quot;CALTinC.ens.dds.rda&quot;</span>)) <span class="co"># DUXC-ALT in canine </span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="kw">load</span>(<span class="kw">file.path</span>(data_dir, <span class="st">&quot;HinC.ens.dds.rda&quot;</span>)) <span class="co"># DUX4 in canine</span></a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="kw">load</span>(<span class="kw">file.path</span>(data_dir, <span class="st">&quot;C2C12.ens.ddsl2.rda&quot;</span>)) <span class="co">#DUX4 and Dux in murine</span></a>
<a class="sourceLine" id="cb1-17" data-line-number="17"></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="kw">rownames</span>(HinC.ens.dds) &lt;-<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(HinC.ens.dds), <span class="dt">fixed=</span><span class="ot">TRUE</span>, <span class="st">&quot;.&quot;</span>), <span class="st">&quot;[&quot;</span>, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="kw">rownames</span>(CinC.ens.dds) &lt;-<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(CinC.ens.dds), <span class="dt">fixed=</span><span class="ot">TRUE</span>, <span class="st">&quot;.&quot;</span>), <span class="st">&quot;[&quot;</span>, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="kw">rownames</span>(CALTinC.ens.dds) &lt;-<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">strsplit</span>(<span class="kw">rownames</span>(CALTinC.ens.dds), <span class="dt">fixed=</span><span class="ot">TRUE</span>, <span class="st">&quot;.&quot;</span>), <span class="st">&quot;[&quot;</span>, <span class="dv">1</span>)</a></code></pre></div>
</div>
<div id="deseq2-1" class="section level2">
<h2><span class="header-section-number">3.2</span> DESeq2</h2>
<div id="hypothesis-settings" class="section level3">
<h3><span class="header-section-number">3.2.1</span> hypothesis settings</h3>
<p><span class="math inline">\(H_0: |log2FoldChange| &lt; 1\)</span> with adjusted p-value threshold <span class="math inline">\(0.05\)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">CinC_res &lt;-<span class="st"> </span><span class="kw">results</span>(CinC.ens.dds, <span class="dt">lfcThreshold=</span><span class="dv">1</span>, <span class="dt">alpha=</span><span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">HinC_res &lt;-<span class="st"> </span><span class="kw">results</span>(HinC.ens.dds, <span class="dt">lfcThreshold=</span><span class="dv">1</span>, <span class="dt">alpha=</span><span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">CALTinC_res &lt;-<span class="st"> </span><span class="kw">results</span>(CALTinC.ens.dds, <span class="dt">lfcThreshold=</span><span class="dv">1</span>, <span class="dt">alpha=</span><span class="fl">0.05</span>)</a></code></pre></div>
</div>
<div id="cinc-duxc-expresison-in-canine-muscle-cells" class="section level3">
<h3><span class="header-section-number">3.2.2</span> CinC: DUXC expresison in canine muscle cells</h3>
<div class="figure"><span id="fig:CinC-MAplot"></span>
<img src="gitbook_files/figure-html/CinC-MAplot-1.png" alt="MA plot of DUXC expression in canine" width="672" />
<p class="caption">
Figure 3.1: MA plot of DUXC expression in canine
</p>
</div>
</div>
<div id="hinc-dux4-expression-in-canine-muscle-cells" class="section level3">
<h3><span class="header-section-number">3.2.3</span> HinC: DUX4 expression in canine muscle cells</h3>
<div class="figure"><span id="fig:HinC-MAplot"></span>
<img src="gitbook_files/figure-html/HinC-MAplot-1.png" alt="MA plot of DUX4 expression in canine" width="672" />
<p class="caption">
Figure 3.2: MA plot of DUX4 expression in canine
</p>
</div>
</div>
<div id="duxc-alt-expression---caltinc" class="section level3">
<h3><span class="header-section-number">3.2.4</span> DUXC-ALT expression - CALTinC</h3>
<p>DUXC-ALT positively affects a few genes (20) but the down-regulates are similar with the DUXC and DUX4 down-regulates. This suggests that even the in-frame extra exon of DUXC-ALT disrupts the transcriptional activity, the conserved C-terminal get to maintain its function and represses expression of some INF-induced genes (e.g. ISG15, CCL5, CXCL10, RSAD2, MX1/MX2).</p>
<div class="figure"><span id="fig:CALTinC-MAplot"></span>
<img src="gitbook_files/figure-html/CALTinC-MAplot-1.png" alt="MA plot of DUXC-ALT expression in canine" width="672" />
<p class="caption">
Figure 3.3: MA plot of DUXC-ALT expression in canine
</p>
</div>
<p>Showing below is a venn diagram showing the overlaps of the down-regulates of DUX, DUX4, and DUXC-ALT.</p>
<div class="figure"><span id="fig:down-venn"></span>
<img src="gitbook_files/figure-html/down-venn-1.png" alt="Venn diagram of down-regualated genes in DUXC, DUX4 and DUXC-ALTC." width="672" />
<p class="caption">
Figure 3.4: Venn diagram of down-regualated genes in DUXC, DUX4 and DUXC-ALTC.
</p>
</div>
<p>NOTE: DUXC-ALT up-regulated 20 genes and nine of them are overlapped with the up-regulated by DUXC – JAG1, NPTX1, ID1, ODAPH, DHRS9, SLC2A2, SYT4, SERPINB12, TRAPPC3L.</p>
<p><strong>GO analysis for the down-regulated gene set</strong></p>
<p>The down-regulated genes of DUXC-ALT are interesting; they might reveals the repressive functions that the conservative c-terminal regulates. Below is the top 10 repressed GO terms for the DUXC-ALT down-reguated.</p>
<p><strong>Top 10 GO terms</strong></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">knitr<span class="op">::</span><span class="kw">kable</span>(enriched.BP[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">c</span>(<span class="st">&quot;category&quot;</span>, </a>
<a class="sourceLine" id="cb3-2" data-line-number="2">                                 <span class="st">&quot;over_represented_pvalue&quot;</span>, </a>
<a class="sourceLine" id="cb3-3" data-line-number="3">                                 <span class="st">&quot;numInCat&quot;</span>, <span class="st">&quot;numDEInCat&quot;</span>, <span class="st">&quot;term&quot;</span>)])</a></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">category</th>
<th align="right">over_represented_pvalue</th>
<th align="right">numInCat</th>
<th align="right">numDEInCat</th>
<th align="left">term</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0009607" class="uri">GO:0009607</a></td>
<td align="right">0e+00</td>
<td align="right">270</td>
<td align="right">19</td>
<td align="left">response to biotic stimulus</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0043207" class="uri">GO:0043207</a></td>
<td align="right">0e+00</td>
<td align="right">253</td>
<td align="right">18</td>
<td align="left">response to external biotic stimulus</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0051707" class="uri">GO:0051707</a></td>
<td align="right">0e+00</td>
<td align="right">253</td>
<td align="right">18</td>
<td align="left">response to other organism</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0009615" class="uri">GO:0009615</a></td>
<td align="right">0e+00</td>
<td align="right">96</td>
<td align="right">11</td>
<td align="left">response to virus</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0051704" class="uri">GO:0051704</a></td>
<td align="right">0e+00</td>
<td align="right">557</td>
<td align="right">22</td>
<td align="left">multi-organism process</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0098542" class="uri">GO:0098542</a></td>
<td align="right">0e+00</td>
<td align="right">115</td>
<td align="right">11</td>
<td align="left">defense response to other organism</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0006952" class="uri">GO:0006952</a></td>
<td align="right">0e+00</td>
<td align="right">377</td>
<td align="right">18</td>
<td align="left">defense response</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0051607" class="uri">GO:0051607</a></td>
<td align="right">0e+00</td>
<td align="right">70</td>
<td align="right">9</td>
<td align="left">defense response to virus</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0009605" class="uri">GO:0009605</a></td>
<td align="right">0e+00</td>
<td align="right">656</td>
<td align="right">23</td>
<td align="left">response to external stimulus</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0045087" class="uri">GO:0045087</a></td>
<td align="right">2e-07</td>
<td align="right">187</td>
<td align="right">12</td>
<td align="left">innate immune response</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="cross-species-models-dux4-expression-in-canine-and-murine" class="section level2">
<h2><span class="header-section-number">3.3</span> Cross-species models: DUX4 expression in canine and murine</h2>
<p>This section shows the comparison of the cross-species models of DUX4 expression in canine and murine: the correlation of gene expression (assembled by log2 fold change) induced by DUX4 and DUXC is significant (Pearson=<span class="math inline">\(0.83\)</span>) whereas that of DUX4 and Dux in murine is weak (Pearson=<span class="math inline">\(0.275\)</span>).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">.tidy_results &lt;-<span class="st"> </span><span class="cf">function</span>(dds, <span class="dt">lfcThreshold=</span><span class="dv">1</span>, <span class="dt">alpha=</span><span class="fl">0.05</span>) {</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">  res &lt;-<span class="st"> </span><span class="kw">results</span>(dds, <span class="dt">lfcThreshold=</span>lfcThreshold, <span class="dt">alpha=</span>alpha)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">  <span class="kw">as</span>(res, <span class="st">&quot;data.frame&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="st">    </span><span class="kw">rownames_to_column</span>(<span class="dt">var=</span><span class="st">&quot;ENSEMBL&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(ENSEMBL, log2FoldChange, padj)</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">}</a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co"># get linear regression model</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8">lm_eqn &lt;-<span class="st"> </span><span class="cf">function</span>(df){</a>
<a class="sourceLine" id="cb4-9" data-line-number="9">    m &lt;-<span class="st"> </span><span class="kw">lm</span>(y <span class="op">~</span><span class="st"> </span>x, df);</a>
<a class="sourceLine" id="cb4-10" data-line-number="10">    eq &lt;-<span class="st"> </span><span class="kw">substitute</span>(<span class="kw">italic</span>(y) <span class="op">==</span><span class="st"> </span>a <span class="op">+</span><span class="st"> </span>b <span class="op">%.%</span><span class="st"> </span><span class="kw">italic</span>(x)<span class="op">*</span><span class="st">&quot;,&quot;</span><span class="op">~</span><span class="er">~</span><span class="kw">italic</span>(r)<span class="op">~</span><span class="st">&quot;=&quot;</span><span class="op">~</span>r2, </a>
<a class="sourceLine" id="cb4-11" data-line-number="11">         <span class="kw">list</span>(<span class="dt">a =</span> <span class="kw">format</span>(<span class="kw">unname</span>(<span class="kw">coef</span>(m)[<span class="dv">1</span>]), <span class="dt">digits =</span> <span class="dv">2</span>),</a>
<a class="sourceLine" id="cb4-12" data-line-number="12">              <span class="dt">b =</span> <span class="kw">format</span>(<span class="kw">unname</span>(<span class="kw">coef</span>(m)[<span class="dv">2</span>]), <span class="dt">digits =</span> <span class="dv">2</span>),</a>
<a class="sourceLine" id="cb4-13" data-line-number="13">              <span class="dt">r2 =</span> <span class="kw">format</span>(<span class="kw">sqrt</span>(<span class="kw">summary</span>(m)<span class="op">$</span>r.squared), <span class="dt">digits =</span> <span class="dv">3</span>)))</a>
<a class="sourceLine" id="cb4-14" data-line-number="14">    <span class="kw">as.character</span>(<span class="kw">as.expression</span>(eq));</a>
<a class="sourceLine" id="cb4-15" data-line-number="15">}</a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co"># tidy up the DESeq2 results</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17">CinC &lt;-<span class="st"> </span><span class="kw">.tidy_results</span>(CinC.ens.dds) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">CANINE_ENSEMBL=</span>ENSEMBL, <span class="dt">CinC_logFC=</span>log2FoldChange,</a>
<a class="sourceLine" id="cb4-19" data-line-number="19">                <span class="dt">CinC_padj=</span>padj)</a>
<a class="sourceLine" id="cb4-20" data-line-number="20">HinC &lt;-<span class="st"> </span><span class="kw">.tidy_results</span>(HinC.ens.dds) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="st">   </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">CANINE_ENSEMBL=</span>ENSEMBL, <span class="dt">HinC_logFC=</span>log2FoldChange,</a>
<a class="sourceLine" id="cb4-22" data-line-number="22">                 <span class="dt">HinC_padj=</span>padj)</a>
<a class="sourceLine" id="cb4-23" data-line-number="23">MinM &lt;-<span class="st"> </span><span class="kw">.tidy_results</span>(C2C12.ens.ddsl2<span class="op">$</span>MinMvsNODOX) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24"><span class="st">   </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">MOUSE_ENSEMBL=</span>ENSEMBL, <span class="dt">MinM_logFC=</span>log2FoldChange,</a>
<a class="sourceLine" id="cb4-25" data-line-number="25">                 <span class="dt">MinM_padj=</span>padj)</a>
<a class="sourceLine" id="cb4-26" data-line-number="26">HinM &lt;-<span class="st"> </span><span class="kw">.tidy_results</span>(C2C12.ens.ddsl2<span class="op">$</span>HinMvsNODOX) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb4-27" data-line-number="27"><span class="st">   </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">MOUSE_ENSEMBL=</span>ENSEMBL, <span class="dt">HinM_logFC=</span>log2FoldChange,</a>
<a class="sourceLine" id="cb4-28" data-line-number="28">                 <span class="dt">HinM_padj=</span>padj)</a></code></pre></div>
<div id="dux4-and-duxc-expression-in-canine" class="section level3">
<h3><span class="header-section-number">3.3.1</span> DUX4 and DUXC expression in canine</h3>
<p>Pearson correlation between DUX4 and DUXC expression (assembled by logFC) in canine is <span class="math inline">\(0.803\)</span>. Code below draws the 2D density plot below presenting their linear relationship: dashed line is the linear regression line <span class="math inline">\(y=0.11 + 0.8x\)</span> whereas the solid line is <span class="math inline">\(x=y\)</span>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">HinC_CinC &lt;-<span class="st"> </span>HinC <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="st">  </span><span class="kw">left_join</span>(CinC, <span class="dt">by=</span><span class="st">&quot;CANINE_ENSEMBL&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(HinC_logFC), <span class="op">!</span><span class="kw">is.na</span>(CinC_logFC)) </a>
<a class="sourceLine" id="cb5-4" data-line-number="4">  </a>
<a class="sourceLine" id="cb5-5" data-line-number="5">df &lt;-<span class="st"> </span>HinC_CinC <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(HinC_logFC, CinC_logFC) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">y=</span>HinC_logFC, <span class="dt">x=</span>CinC_logFC)  </a>
<a class="sourceLine" id="cb5-7" data-line-number="7">label &lt;-<span class="st"> </span><span class="kw">lm_eqn</span>(df)</a>
<a class="sourceLine" id="cb5-8" data-line-number="8">gg &lt;-<span class="st"> </span><span class="kw">ggplot</span>(HinC_CinC, <span class="kw">aes</span>(<span class="dt">x=</span>CinC_logFC, <span class="dt">y=</span>HinC_logFC)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="st">  </span><span class="kw">stat_density2d</span>(<span class="dt">geom=</span><span class="st">&quot;tile&quot;</span>, <span class="kw">aes</span>(<span class="dt">fill=</span>..density..<span class="op">^</span><span class="fl">0.15</span>, <span class="dt">alpha=</span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb5-10" data-line-number="10">                 <span class="dt">contour=</span><span class="ot">FALSE</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="fl">0.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="st">  </span><span class="kw">stat_density2d</span>(<span class="dt">geom=</span><span class="st">&quot;tile&quot;</span>, <span class="kw">aes</span>(<span class="dt">fill=</span>..density..<span class="op">^</span><span class="fl">0.15</span>,    </a>
<a class="sourceLine" id="cb5-13" data-line-number="13">                                  <span class="dt">alpha=</span><span class="kw">ifelse</span>(..density..<span class="op">^</span><span class="fl">0.15</span><span class="op">&lt;</span><span class="fl">0.3</span>,<span class="dv">0</span>,<span class="dv">1</span>)),</a>
<a class="sourceLine" id="cb5-14" data-line-number="14">                 <span class="dt">contour=</span><span class="ot">FALSE</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="st">  </span><span class="kw">scale_fill_gradientn</span>(<span class="dt">colours =</span> <span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, blues9))(<span class="dv">256</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="dt">method=</span><span class="st">&quot;lm&quot;</span>, <span class="dt">se=</span><span class="ot">FALSE</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>, <span class="dt">color=</span><span class="st">&quot;gray25&quot;</span>,</a>
<a class="sourceLine" id="cb5-17" data-line-number="17">              <span class="dt">alpha=</span><span class="fl">0.5</span>,</a>
<a class="sourceLine" id="cb5-18" data-line-number="18">              <span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="st">  </span><span class="kw">geom_abline</span>(<span class="dt">slope=</span><span class="dv">1</span>, <span class="dt">intercept=</span><span class="dv">0</span>, <span class="dt">color=</span><span class="st">&quot;gray25&quot;</span>, <span class="dt">alpha=</span><span class="fl">0.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="kw">TeX</span>(<span class="st">&quot;HinC vs. CinC $</span><span class="ch">\\</span><span class="st">log_2$ fold change&quot;</span>),</a>
<a class="sourceLine" id="cb5-21" data-line-number="21">       <span class="dt">y=</span><span class="kw">TeX</span>(<span class="st">&quot;HinC: $</span><span class="ch">\\</span><span class="st">log_2$(DUX4/luciferase)&quot;</span>),</a>
<a class="sourceLine" id="cb5-22" data-line-number="22">       <span class="dt">x=</span><span class="kw">TeX</span>(<span class="st">&quot;CinC: $</span><span class="ch">\\</span><span class="st">log_2$(DUXC/luciferase)&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-23" data-line-number="23"><span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb5-24" data-line-number="24"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>, <span class="dt">face=</span><span class="st">&quot;bold&quot;</span>, <span class="dt">size=</span><span class="dv">16</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-25" data-line-number="25"><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">x=</span><span class="fl">7.5</span>, <span class="dt">y=</span><span class="fl">7.5</span>, <span class="dt">label=</span><span class="st">&quot;x=y&quot;</span>, <span class="dt">angle=</span><span class="dv">38</span>, <span class="dt">vjust=</span><span class="op">-</span><span class="dv">1</span>, <span class="dt">color=</span><span class="st">&quot;gray25&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-26" data-line-number="26"><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">x=</span><span class="fl">7.5</span>, <span class="dt">y=</span><span class="fl">5.5</span>, <span class="dt">label=</span><span class="st">&quot;y=0.11+0.8x; pearson=0.803&quot;</span>, <span class="dt">angle=</span><span class="dv">28</span>,</a>
<a class="sourceLine" id="cb5-27" data-line-number="27">            <span class="dt">vjust=</span><span class="dv">1</span>, <span class="dt">color=</span><span class="st">&quot;gray25&quot;</span>)</a>
<a class="sourceLine" id="cb5-28" data-line-number="28">gg</a></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<div class="figure"><span id="fig:HinC-vs-CinC"></span>
<img src="gitbook_files/figure-html/HinC-vs-CinC-1.png" alt="2D density scatter plot between DUX4 and DUXC expression in canine. " width="672" />
<p class="caption">
Figure 3.5: 2D density scatter plot between DUX4 and DUXC expression in canine.
</p>
</div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;figures&quot;</span>, <span class="st">&quot;HinC_CinC_logFC_smooth_scatter.pdf&quot;</span>),</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">       <span class="dt">width=</span><span class="dv">5</span>, <span class="dt">height=</span><span class="fl">3.75</span>)</a></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
</div>
<div id="dux4-and-dux-expression-in-murine" class="section level3">
<h3><span class="header-section-number">3.3.2</span> DUX4 and Dux expression in murine</h3>
<p>Pearson correlation between DUX4 and Dux expression in murine is <span class="math inline">\(0.275\)</span>; linear regression model is <span class="math inline">\(y=-0.024 + 0.21x\)</span>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">HinM_MinM &lt;-<span class="st"> </span>HinM <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="st">  </span><span class="kw">left_join</span>(MinM, <span class="dt">by=</span><span class="st">&quot;MOUSE_ENSEMBL&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(HinM_logFC), <span class="op">!</span><span class="kw">is.na</span>(MinM_logFC)) </a>
<a class="sourceLine" id="cb9-4" data-line-number="4">  </a>
<a class="sourceLine" id="cb9-5" data-line-number="5">df &lt;-<span class="st"> </span>HinM_MinM <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(HinM_logFC, MinM_logFC) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">y=</span>HinM_logFC, <span class="dt">x=</span>MinM_logFC)  </a>
<a class="sourceLine" id="cb9-7" data-line-number="7">label &lt;-<span class="st"> </span><span class="kw">lm_eqn</span>(df)</a>
<a class="sourceLine" id="cb9-8" data-line-number="8">gg &lt;-<span class="st"> </span><span class="kw">ggplot</span>(HinM_MinM, <span class="kw">aes</span>(<span class="dt">x=</span>MinM_logFC, <span class="dt">y=</span>HinM_logFC)) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="st">  </span><span class="kw">stat_density2d</span>(<span class="dt">geom=</span><span class="st">&quot;tile&quot;</span>, <span class="kw">aes</span>(<span class="dt">fill=</span>..density..<span class="op">^</span><span class="fl">0.15</span>, <span class="dt">alpha=</span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb9-10" data-line-number="10">                 <span class="dt">contour=</span><span class="ot">FALSE</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="fl">0.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="st">  </span><span class="kw">stat_density2d</span>(<span class="dt">geom=</span><span class="st">&quot;tile&quot;</span>, <span class="kw">aes</span>(<span class="dt">fill=</span>..density..<span class="op">^</span><span class="fl">0.15</span>,    </a>
<a class="sourceLine" id="cb9-13" data-line-number="13">                                  <span class="dt">alpha=</span><span class="kw">ifelse</span>(..density..<span class="op">^</span><span class="fl">0.15</span><span class="op">&lt;</span><span class="fl">0.3</span>,<span class="dv">0</span>,<span class="dv">1</span>)),</a>
<a class="sourceLine" id="cb9-14" data-line-number="14">                 <span class="dt">contour=</span><span class="ot">FALSE</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="st">  </span><span class="kw">scale_fill_gradientn</span>(<span class="dt">colours =</span> <span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, blues9))(<span class="dv">256</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="st">  </span><span class="kw">geom_smooth</span>(<span class="dt">method=</span><span class="st">&quot;lm&quot;</span>, <span class="dt">se=</span><span class="ot">FALSE</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>, <span class="dt">color=</span><span class="st">&quot;gray25&quot;</span>,</a>
<a class="sourceLine" id="cb9-17" data-line-number="17">              <span class="dt">alpha=</span><span class="fl">0.5</span>,</a>
<a class="sourceLine" id="cb9-18" data-line-number="18">              <span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-19" data-line-number="19"><span class="st">  </span><span class="kw">geom_abline</span>(<span class="dt">slope=</span><span class="dv">1</span>, <span class="dt">intercept=</span><span class="dv">0</span>, <span class="dt">color=</span><span class="st">&quot;gray25&quot;</span>, <span class="dt">alpha=</span><span class="fl">0.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-20" data-line-number="20"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="kw">TeX</span>(<span class="st">&quot;HinM vs. MinM $</span><span class="ch">\\</span><span class="st">log_2$ fold change&quot;</span>),</a>
<a class="sourceLine" id="cb9-21" data-line-number="21">       <span class="dt">y=</span><span class="kw">TeX</span>(<span class="st">&quot;HinM: $</span><span class="ch">\\</span><span class="st">log_2$(DUX4/no DUX4)&quot;</span>),</a>
<a class="sourceLine" id="cb9-22" data-line-number="22">       <span class="dt">x=</span><span class="kw">TeX</span>(<span class="st">&quot;MinM: $</span><span class="ch">\\</span><span class="st">log_2$(Dux/no Dux)&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-23" data-line-number="23"><span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb9-24" data-line-number="24"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>, <span class="dt">face=</span><span class="st">&quot;bold&quot;</span>, <span class="dt">size=</span><span class="dv">16</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-25" data-line-number="25"><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">x=</span><span class="dv">10</span>, <span class="dt">y=</span><span class="dv">10</span>, <span class="dt">label=</span><span class="st">&quot;x=y&quot;</span>, <span class="dt">angle=</span><span class="dv">38</span>, <span class="dt">vjust=</span><span class="op">-</span><span class="dv">1</span>, <span class="dt">color=</span><span class="st">&quot;gray25&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-26" data-line-number="26"><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">x=</span><span class="dv">10</span>, <span class="dt">y=</span><span class="fl">1.4</span>, <span class="dt">label=</span><span class="st">&quot;y=-0.024+0.21x; pearson=0.275&quot;</span>, <span class="dt">angle=</span><span class="dv">8</span>,</a>
<a class="sourceLine" id="cb9-27" data-line-number="27">            <span class="dt">vjust=</span><span class="dv">1</span>, <span class="dt">color=</span><span class="st">&quot;gray25&quot;</span>)</a>
<a class="sourceLine" id="cb9-28" data-line-number="28">  </a>
<a class="sourceLine" id="cb9-29" data-line-number="29">gg</a></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<div class="figure"><span id="fig:HinM-vs-MinM"></span>
<img src="gitbook_files/figure-html/HinM-vs-MinM-1.png" alt="2D density scatter plot between DUX4 and Dux expression in murine." width="672" />
<p class="caption">
Figure 3.6: 2D density scatter plot between DUX4 and Dux expression in murine.
</p>
</div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;figures&quot;</span>, <span class="st">&quot;HinM_MinM_logFC_smooth_scatter.pdf&quot;</span>),</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">       <span class="dt">width=</span><span class="dv">5</span>, <span class="dt">height=</span><span class="fl">3.75</span>)</a></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="makeSE.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="DUXC-FSHD-transcriptome.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/FredHutch/canFam3.DuxFamily/tree/master/gitbook/03-DUXC-transcriptome.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/FredHutch/canFam3.DuxFamily/tree/master/gitbook/03-DUXC-transcriptome.Rmd",
"text": null
},
"download": ["gitbook.pdf", "gitbook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
