<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</title>
  <meta name="description" content="Bioinformatics and statistics analysis" />
  <meta name="generator" content="bookdown 0.32 and GitBook 2.6.7" />

  <meta property="og:title" content="Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Bioinformatics and statistics analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD" />
  
  <meta name="twitter:description" content="Bioinformatics and statistics analysis" />
  

<meta name="author" content="Chao-Jen Wong" />


<meta name="date" content="2023-04-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="chipseq.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DUXC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#nomenclature"><i class="fa fa-check"></i><b>0.1</b> Nomenclature</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#datasets"><i class="fa fa-check"></i><b>0.2</b> Datasets</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#annotation"><i class="fa fa-check"></i><b>0.3</b> Annotation</a>
<ul>
<li class="chapter" data-level="0.3.1" data-path="index.html"><a href="index.html#canfam3-ensembl-gtf"><i class="fa fa-check"></i><b>0.3.1</b> canFam3 ensembl GTF</a></li>
</ul></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>0.4</b> Software</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="chipseq.html"><a href="chipseq.html"><i class="fa fa-check"></i><b>1</b> ChIP-seq for the canine model</a>
<ul>
<li class="chapter" data-level="1.1" data-path="chipseq.html"><a href="chipseq.html#chip-seq-samples"><i class="fa fa-check"></i><b>1.1</b> ChIP-seq samples</a></li>
<li class="chapter" data-level="1.2" data-path="chipseq.html"><a href="chipseq.html#analysis-workflow"><i class="fa fa-check"></i><b>1.2</b> Analysis workflow</a></li>
<li class="chapter" data-level="1.3" data-path="chipseq.html"><a href="chipseq.html#peaks-and-functional-distribution"><i class="fa fa-check"></i><b>1.3</b> Peaks and functional distribution</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="chipseq.html"><a href="chipseq.html#number-of-peaks"><i class="fa fa-check"></i><b>1.3.1</b> Number of peaks</a></li>
<li class="chapter" data-level="1.3.2" data-path="chipseq.html"><a href="chipseq.html#functional-discribution"><i class="fa fa-check"></i><b>1.3.2</b> Functional discribution</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chipseq.html"><a href="chipseq.html#overlap-of-peaks"><i class="fa fa-check"></i><b>1.4</b> Overlap of peaks</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="chipseq.html"><a href="chipseq.html#duxc-and-dux4"><i class="fa fa-check"></i><b>1.4.1</b> DUXC and DUX4</a></li>
<li class="chapter" data-level="1.4.2" data-path="chipseq.html"><a href="chipseq.html#duxc-and-duxc-alt"><i class="fa fa-check"></i><b>1.4.2</b> DUXC and DUXC-ALT</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="chipseq.html"><a href="chipseq.html#motifs-analysis"><i class="fa fa-check"></i><b>1.5</b> Motifs analysis</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="makeSE.html"><a href="makeSE.html"><i class="fa fa-check"></i><b>2</b> RNA-seq to DESeq2</a>
<ul>
<li class="chapter" data-level="2.1" data-path="makeSE.html"><a href="makeSE.html#rna-seq-preprocessing"><i class="fa fa-check"></i><b>2.1</b> RNA-seq preprocessing</a></li>
<li class="chapter" data-level="2.2" data-path="makeSE.html"><a href="makeSE.html#reads-counts"><i class="fa fa-check"></i><b>2.2</b> Reads counts</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="makeSE.html"><a href="makeSE.html#define-gene-features"><i class="fa fa-check"></i><b>2.2.1</b> Define gene features</a></li>
<li class="chapter" data-level="2.2.2" data-path="makeSE.html"><a href="makeSE.html#read-counts"><i class="fa fa-check"></i><b>2.2.2</b> Read counts</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="makeSE.html"><a href="makeSE.html#deseq2"><i class="fa fa-check"></i><b>2.3</b> DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html"><i class="fa fa-check"></i><b>3</b> DUXC transcriptome</a>
<ul>
<li class="chapter" data-level="3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#load-datasets-and-settings"><i class="fa fa-check"></i><b>3.1</b> Load datasets and settings</a></li>
<li class="chapter" data-level="3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#deseq2-1"><i class="fa fa-check"></i><b>3.2</b> DESeq2</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hypothesis-settings"><i class="fa fa-check"></i><b>3.2.1</b> hypothesis settings</a></li>
<li class="chapter" data-level="3.2.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cinc-duxc-expresison-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2.2</b> CinC: DUXC expresison in canine muscle cells</a></li>
<li class="chapter" data-level="3.2.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hinc-dux4-expression-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2.3</b> HinC: DUX4 expression in canine muscle cells</a></li>
<li class="chapter" data-level="3.2.4" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#duxc-alt-expression---caltinc"><i class="fa fa-check"></i><b>3.2.4</b> DUXC-ALT expression - CALTinC</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cross-species-models-dux4-expression-in-canine-and-murine"><i class="fa fa-check"></i><b>3.3</b> Cross-species models: DUX4 expression in canine and murine</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-duxc-expression-in-canine"><i class="fa fa-check"></i><b>3.3.1</b> DUX4 and DUXC expression in canine</a></li>
<li class="chapter" data-level="3.3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-dux-expression-in-murine"><i class="fa fa-check"></i><b>3.3.2</b> DUX4 and Dux expression in murine</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html"><i class="fa fa-check"></i><b>4</b> DUXC and Dux transcriptome compared to that of FSHD</a>
<ul>
<li class="chapter" data-level="4.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#duxc-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.1</b> DUXC transcriptome compared to that of FSHD</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs"><i class="fa fa-check"></i><b>4.1.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.1.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome"><i class="fa fa-check"></i><b>4.1.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#dux-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.2</b> Dux transcriptome compared to that of FSHD</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs-1"><i class="fa fa-check"></i><b>4.2.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.2.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome-1"><i class="fa fa-check"></i><b>4.2.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html"><i class="fa fa-check"></i><b>5</b> DUXC transcriptome compared to early embryo transcriptome profile</a>
<ul>
<li class="chapter" data-level="5.1" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#mouse-2c-like-signature"><i class="fa fa-check"></i><b>5.1</b> Mouse 2C-like signature</a></li>
<li class="chapter" data-level="5.2" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#cleavage-stage-signature"><i class="fa fa-check"></i><b>5.2</b> cleavage-stage signature</a></li>
<li class="chapter" data-level="5.3" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#ipscs"><i class="fa fa-check"></i><b>5.3</b> iPSCs</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="repeats.html"><a href="repeats.html"><i class="fa fa-check"></i><b>6</b> Repeat elements</a>
<ul>
<li class="chapter" data-level="6.1" data-path="repeats.html"><a href="repeats.html#takeaway"><i class="fa fa-check"></i><b>6.1</b> Takeaway</a></li>
<li class="chapter" data-level="6.2" data-path="repeats.html"><a href="repeats.html#repeat-element-expression"><i class="fa fa-check"></i><b>6.2</b> Repeat element expression</a></li>
<li class="chapter" data-level="6.3" data-path="repeats.html"><a href="repeats.html#differential-analysis"><i class="fa fa-check"></i><b>6.3</b> Differential analysis</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="repeats.html"><a href="repeats.html#normalization"><i class="fa fa-check"></i><b>6.3.1</b> Normalization</a></li>
<li class="chapter" data-level="6.3.2" data-path="repeats.html"><a href="repeats.html#deseq2-results"><i class="fa fa-check"></i><b>6.3.2</b> DESeq2 results</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="repeats.html"><a href="repeats.html#repeat-family-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> Repeat family enrichment analysis</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="repeats.html"><a href="repeats.html#correlation-between-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>6.4.1</b> Correlation between HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="6.4.2" data-path="repeats.html"><a href="repeats.html#repeat-family-activation"><i class="fa fa-check"></i><b>6.4.2</b> Repeat family activation</a></li>
<li class="chapter" data-level="6.4.3" data-path="repeats.html"><a href="repeats.html#repeat-classs-activation"><i class="fa fa-check"></i><b>6.4.3</b> Repeat classs activation</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="repeats.html"><a href="repeats.html#ltr-retroelements-distribution-and-activation"><i class="fa fa-check"></i><b>6.5</b> LTR retroelements distribution and activation</a></li>
<li class="chapter" data-level="6.6" data-path="repeats.html"><a href="repeats.html#peaks-distribution-in-class-of-repeat-elements"><i class="fa fa-check"></i><b>6.6</b> Peaks distribution in class of repeat elements</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="LTR-activation.html"><a href="LTR-activation.html"><i class="fa fa-check"></i><b>7</b> LTR activation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="LTR-activation.html"><a href="LTR-activation.html#takaway"><i class="fa fa-check"></i><b>7.1</b> Takaway</a></li>
<li class="chapter" data-level="7.2" data-path="LTR-activation.html"><a href="LTR-activation.html#background-in-murine-and-human"><i class="fa fa-check"></i><b>7.2</b> background in murine and human</a></li>
<li class="chapter" data-level="7.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-subfamily-activation-in-canine"><i class="fa fa-check"></i><b>7.3</b> ERVL subfamily activation in canine</a></li>
<li class="chapter" data-level="7.4" data-path="LTR-activation.html"><a href="LTR-activation.html#malr-activation-in-canine"><i class="fa fa-check"></i><b>7.4</b> MaLR activation in canine</a></li>
<li class="chapter" data-level="7.5" data-path="LTR-activation.html"><a href="LTR-activation.html#compare-to-cinc-and-hinc-to-hinh"><i class="fa fa-check"></i><b>7.5</b> Compare to CinC and HinC to HinH</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="LTR-activation.html"><a href="LTR-activation.html#copy-number"><i class="fa fa-check"></i><b>7.5.1</b> Copy number</a></li>
<li class="chapter" data-level="7.5.2" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-malr"><i class="fa fa-check"></i><b>7.5.2</b> ERVL-MaLR</a></li>
<li class="chapter" data-level="7.5.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl"><i class="fa fa-check"></i><b>7.5.3</b> ERVL</a></li>
<li class="chapter" data-level="7.5.4" data-path="LTR-activation.html"><a href="LTR-activation.html#erv1"><i class="fa fa-check"></i><b>7.5.4</b> ERV1</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="LTR-activation.html"><a href="LTR-activation.html#duxc-alt-and-ervl-transcriptions"><i class="fa fa-check"></i><b>7.6</b> DUXC-ALT and ERVL transcriptions</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="LTR-activation.html"><a href="LTR-activation.html#erv-subfamily-expression"><i class="fa fa-check"></i><b>7.6.1</b> ERV subfamily expression</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="non-LTR.html"><a href="non-LTR.html"><i class="fa fa-check"></i><b>8</b> Non-LTR retrotrasposons and satellites</a>
<ul>
<li class="chapter" data-level="8.1" data-path="non-LTR.html"><a href="non-LTR.html#lines"><i class="fa fa-check"></i><b>8.1</b> LINEs</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="non-LTR.html"><a href="non-LTR.html#l1s-activation"><i class="fa fa-check"></i><b>8.1.1</b> L1s activation</a></li>
<li class="chapter" data-level="8.1.2" data-path="non-LTR.html"><a href="non-LTR.html#line1-and-duxcdux4-binding-sites"><i class="fa fa-check"></i><b>8.1.2</b> LINE1 and DUXC/DUX4 binding sites</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="non-LTR.html"><a href="non-LTR.html#sine"><i class="fa fa-check"></i><b>8.2</b> SINE</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="non-LTR.html"><a href="non-LTR.html#distribution-of-sine-family"><i class="fa fa-check"></i><b>8.2.1</b> Distribution of SINE family</a></li>
<li class="chapter" data-level="8.2.2" data-path="non-LTR.html"><a href="non-LTR.html#expression-of-sines-in-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>8.2.2</b> Expression of SINEs in HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="8.2.3" data-path="non-LTR.html"><a href="non-LTR.html#trna-lys"><i class="fa fa-check"></i><b>8.2.3</b> tRNA-Lys</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="non-LTR.html"><a href="non-LTR.html#satellites"><i class="fa fa-check"></i><b>8.3</b> Satellites</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="peaks-line1.html"><a href="peaks-line1.html"><i class="fa fa-check"></i><b>9</b> DUXC peaks and LINE1s</a>
<ul>
<li class="chapter" data-level="9.1" data-path="peaks-line1.html"><a href="peaks-line1.html#narrow-peaks-and-expressed-l1s"><i class="fa fa-check"></i><b>9.1</b> Narrow Peaks and expressed L1s</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="peaks-line1.html"><a href="peaks-line1.html#motif-discovery-using-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.1</b> Motif discovery using L1-overlapped peaks</a></li>
<li class="chapter" data-level="9.1.2" data-path="peaks-line1.html"><a href="peaks-line1.html#match-dux4-and-duxc-motif-to-l1-overapped-peaks"><i class="fa fa-check"></i><b>9.1.2</b> Match DUX4 and DUXC motif to L1-overapped peaks</a></li>
<li class="chapter" data-level="9.1.3" data-path="peaks-line1.html"><a href="peaks-line1.html#location-of-aatca-on-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.3</b> Location of AATCA on L1-overlapped peaks</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="peaks-line1.html"><a href="peaks-line1.html#line1s-and-aatca"><i class="fa fa-check"></i><b>9.2</b> LINE1s and AATCA</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</h1>
<p class="author"><em>Chao-Jen Wong</em></p>
<p class="date"><em>2023-04-10</em></p>
</div>
<div id="preface" class="section level1 unnumbered hasAnchor">
<h1>Preface<a href="index.html#preface" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="figure"><span style="display:block;" id="fig:duxc-summary"></span>
<img src="images/duxc_summary.png" alt="The gene models of the ancestral double homeodomain gene DUXC and its alternative isoform DUXC-ALT in relation to the retrotransposed human DUX4 (hDUX4) and murine Dux (mDux)." width="1141" />
<p class="caption">
Figure 0.1: The gene models of the ancestral double homeodomain gene DUXC and its alternative isoform DUXC-ALT in relation to the retrotransposed human DUX4 (hDUX4) and murine Dux (mDux).
</p>
</div>
<p>The aim of this book is to promote the reproducibility and transparency of the computational work for our manuscript titled <a href="https://doi.org/10.1093/hmg/ddab352"><em>Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</em></a> <span class="citation">(<a href="#ref-duxc" role="doc-biblioref">Chao-Jen Wong 2022</a>)</span>. Built on R Markdown, knitr <span class="citation">(<a href="#ref-R-knitr" role="doc-biblioref">Xie 2019</a>)</span>, and bookdown <span class="citation">(<a href="#ref-bookdown2016" role="doc-biblioref">Xie 2016</a>)</span>, this book provides the workflow and reproducible R code for our bioinformatics analysis and generates tables and figures on the fly. Note that certain sections of the code may be hidden from view due to their length and tediousness, but can be accessed through the <a href="https://github.com/FredHutch/canFam3.DuxFamily/tree/master/gitbook">Rmd source files</a>.</p>
<div id="nomenclature" class="section level2 hasAnchor" number="0.1">
<h2><span class="header-section-number">0.1</span> Nomenclature<a href="index.html#nomenclature" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:nomenclature-table">Table 0.1: </span>The nomenclatures used in this book.
</caption>
<thead>
<tr>
<th style="text-align:left;">
nomenclature
</th>
<th style="text-align:left;">
model
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
CinC
</td>
<td style="text-align:left;">
DUXC in canine
</td>
</tr>
<tr>
<td style="text-align:left;">
HinC
</td>
<td style="text-align:left;">
hDUX4 in canine
</td>
</tr>
<tr>
<td style="text-align:left;">
CALTinC
</td>
<td style="text-align:left;">
DUXC-ALT in canine
</td>
</tr>
<tr>
<td style="text-align:left;">
HinH
</td>
<td style="text-align:left;">
hDUX4 in human
</td>
</tr>
<tr>
<td style="text-align:left;">
HinM
</td>
<td style="text-align:left;">
hDUX4 in mouse
</td>
</tr>
<tr>
<td style="text-align:left;">
MinM
</td>
<td style="text-align:left;">
mDux in mouse
</td>
</tr>
</tbody>
</table>
</div>
<div id="datasets" class="section level2 hasAnchor" number="0.2">
<h2><span class="header-section-number">0.2</span> Datasets<a href="index.html#datasets" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Within the <a href="https://github.com/FredHutch/canFam3.DuxFamily/tree/master/data">data</a> folder of the repository, we have included <code>DESeqDataSet</code> instances that contain read counts for DUXC, DUXC-ALT, human DUX4 (hDUX4), and murine Dux (mDux), as well as their corresponding repeat elements expression and metadata. Additionally, the folder contains results from the ChIP-seq peaks calling in the form of <code>GRanges</code>.</p>
<p>The raw canine RNA-seq data has been deposited to GEO, and can be accessed through the series number <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE188928">GSE188928</a>.</p>
<p>There are three categories of datsets:</p>
<ul>
<li>DUX4, DUXC, and DUX-ALT transcriptome (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE188928">GSE188928</a>) in canine skeletal muscle and repeat elements expression stored in DESeqDataSet instances</li>
</ul>
<pre><code>├── CALTinC.ens.dds.rda: DUXC-ALT transcriptome in canine skeletal muscle
├── CinC.ens.dds.rda: DUXC transcriptome
├── HinC.ens.dds.rda: DUX4 transcriptome in canine skeletal muscle
├── rmsk.dds.rda: DUX4, DUXC, DUXC-ALT repeat elements expression
├── CinC_rmsk.rda: DUXC repeat elements DESeq2 results
├── HinC_rmsk.rda: DUX4 repeat elements DESeq2 resutls
├── CALTinC_rmsk.rda: DUXC-ALT repeat elements DESeq2 results</code></pre>
<ul>
<li>human DUX4 transcriptome in human myoblast (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85461">GSE85461</a>) and mouse Dux transcriptome in mouse myoblast cell lines (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE87282">GSE87282</a>)<br />
</li>
</ul>
<pre><code>├── C2C12.ens.ddsl2.rda: Dux and DUX4 transcriptome in mouse myoblast cell lines
├── HinH.ens.dds.rda: DUX4 transcriptome in human myoblast
├── HinH_rmsk.rda: repeat element expression in HinH transcriptome (DESeq2 resutls)</code></pre>
<ul>
<li>miscellaneous. The orthologous are collected from Ensembl horology database V88.<br />
</li>
</ul>
<pre><code>├── human_cleavage.rda: Hendrikson 2017 zygotic activation gene set
├── canine_human_mouse_ensOrtholog.rda: human-canine and mouse-canine paired orthologous
├── human_mouse_ensOrtholog.rda: human-mouse paired Ortholog
├── peaks_list.rda: ChIP-seq peaks 
├── chipseq_si.rda: ChIP-seq sample information
└── z4_ensembl.rda: Akiyama 2015 2C-like gene set from </code></pre>
</div>
<div id="annotation" class="section level2 hasAnchor" number="0.3">
<h2><span class="header-section-number">0.3</span> Annotation<a href="index.html#annotation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="canfam3-ensembl-gtf" class="section level3 hasAnchor" number="0.3.1">
<h3><span class="header-section-number">0.3.1</span> canFam3 ensembl GTF<a href="index.html#canfam3-ensembl-gtf" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We used Bioconductor’s <em>AnnotationHub</em> package to extract canFam3 ensembl’s GTF file and built a <code>TxDb</code> (transcript database) annotation package. The code chunk below is an example of how to build a <code>TxDb</code> package using the <code>AnnotationHub</code> and <code>GenomicFeatures</code> packages.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="index.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># example:</span></span>
<span id="cb4-2"><a href="index.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span>
<span id="cb4-3"><a href="index.html#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicFeatures)</span>
<span id="cb4-4"><a href="index.html#cb4-4" aria-hidden="true" tabindex="-1"></a>ah <span class="ot">=</span> <span class="fu">AnnotationHub</span>()</span>
<span id="cb4-5"><a href="index.html#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fu">query</span>(ah, <span class="fu">c</span>(<span class="st">&quot;canFam3&quot;</span>, <span class="st">&quot;ensembl&quot;</span>))</span>
<span id="cb4-6"><a href="index.html#cb4-6" aria-hidden="true" tabindex="-1"></a>gr <span class="ot">&lt;-</span> ah[<span class="st">&quot;AH79003&quot;</span>] <span class="do">## download Canis_familiaris ensembl version 99</span></span>
<span id="cb4-7"><a href="index.html#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="index.html#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="do">## construce a meata data.frame</span></span>
<span id="cb4-9"><a href="index.html#cb4-9" aria-hidden="true" tabindex="-1"></a>md <span class="ot">&lt;-</span> </span>
<span id="cb4-10"><a href="index.html#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">name=</span><span class="fu">c</span>(<span class="st">&quot;ensembl_version&quot;</span>, <span class="st">&quot;Data source&quot;</span>, </span>
<span id="cb4-11"><a href="index.html#cb4-11" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;Organism&quot;</span>, <span class="st">&quot;Resource URL&quot;</span>, <span class="st">&quot;Type of Gene ID&quot;</span>),</span>
<span id="cb4-12"><a href="index.html#cb4-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">value=</span><span class="fu">c</span>(<span class="st">&quot;canFam3.1.99&quot;</span>, <span class="st">&quot;AnnotationHub query AH79003&quot;</span>, </span>
<span id="cb4-13"><a href="index.html#cb4-13" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;Canis familiaris &quot;</span>, <span class="st">&quot;http://ensembl.org&quot;</span>, </span>
<span id="cb4-14"><a href="index.html#cb4-14" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;Ensembl Gene ID&quot;</span>))</span>
<span id="cb4-15"><a href="index.html#cb4-15" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">&lt;-</span> GenomicFeatures<span class="sc">::</span><span class="fu">makeTxDbFromGRanges</span>(gr, <span class="at">metadata=</span>md)</span>
<span id="cb4-16"><a href="index.html#cb4-16" aria-hidden="true" tabindex="-1"></a>makeTxDbPackage<span class="sc">::</span><span class="fu">makeTxDbPackage</span>(</span>
<span id="cb4-17"><a href="index.html#cb4-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">txdb=</span>txdb, <span class="at">version=</span><span class="st">&quot;1.0.0&quot;</span>,</span>
<span id="cb4-18"><a href="index.html#cb4-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">pkgname=</span><span class="st">&quot;TxDb.Cfamiliaris.CanFam3.ensGene&quot;</span>,</span>
<span id="cb4-19"><a href="index.html#cb4-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">maintainer=</span><span class="st">&quot;Chao-Jen Wong &lt;cwon2@fredhutch.org&gt;&quot;</span>,</span>
<span id="cb4-20"><a href="index.html#cb4-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">author=</span><span class="st">&quot;Chao-Jen Wong&quot;</span>,</span>
<span id="cb4-21"><a href="index.html#cb4-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">destDir=</span><span class="st">&quot;~/tapscott/canfam3&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="software" class="section level2 hasAnchor" number="0.4">
<h2><span class="header-section-number">0.4</span> Software<a href="index.html#software" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>R/3.6.3, tidyverse and Bioconductor (v3.10) packages</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-duxc" class="csl-entry">
Chao-Jen Wong, Ashlee T Langford, Jennifer L Whiddon. 2022. <span>“Canine DUXC: Implications for DUX4 Retrotransposition and Preclinical Models of FSHD.”</span> <em>Human Molecular Genetics</em>.
</div>
<div id="ref-bookdown2016" class="csl-entry">
Xie, Yihui. 2016. <em>Bookdown: Authoring Books and Technical Documents with <span>R</span> Markdown</em>. Boca Raton, Florida: Chapman; Hall/CRC. <a href="https://github.com/rstudio/bookdown">https://github.com/rstudio/bookdown</a>.
</div>
<div id="ref-R-knitr" class="csl-entry">
———. 2019. <em>Knitr: A General-Purpose Package for Dynamic Report Generation in r</em>. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="chipseq.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/FredHutch/canFam3.DuxFamily/tree/master/gitbook/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/FredHutch/canFam3.DuxFamily/tree/master/gitbook/index.Rmd",
"text": null
},
"download": ["gitbook.pdf", "gitbook.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
