<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 LTR activation | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</title>
  <meta name="description" content="Bioinformatics and statistics analysis" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 LTR activation | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Bioinformatics and statistics analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 LTR activation | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD" />
  
  <meta name="twitter:description" content="Bioinformatics and statistics analysis" />
  

<meta name="author" content="Chao-Jen Wong" />


<meta name="date" content="2021-11-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="repeats.html"/>
<link rel="next" href="non-LTR.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DUXC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#nomenclature"><i class="fa fa-check"></i><b>0.1</b> Nomenclature</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#datasets"><i class="fa fa-check"></i><b>0.2</b> Datasets</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#annotation"><i class="fa fa-check"></i><b>0.3</b> Annotation</a><ul>
<li class="chapter" data-level="0.3.1" data-path="index.html"><a href="index.html#canfam3-ensembl-gtf"><i class="fa fa-check"></i><b>0.3.1</b> canFam3 ensembl GTF</a></li>
</ul></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>0.4</b> Software</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="chipseq.html"><a href="chipseq.html"><i class="fa fa-check"></i><b>1</b> ChIP-seq for the canine model</a><ul>
<li class="chapter" data-level="1.1" data-path="chipseq.html"><a href="chipseq.html#chip-seq-samples"><i class="fa fa-check"></i><b>1.1</b> ChIP-seq samples</a></li>
<li class="chapter" data-level="1.2" data-path="chipseq.html"><a href="chipseq.html#analysis-workflow"><i class="fa fa-check"></i><b>1.2</b> Analysis workflow</a></li>
<li class="chapter" data-level="1.3" data-path="chipseq.html"><a href="chipseq.html#peaks-and-functional-distribution"><i class="fa fa-check"></i><b>1.3</b> Peaks and functional distribution</a><ul>
<li class="chapter" data-level="1.3.1" data-path="chipseq.html"><a href="chipseq.html#number-of-peaks"><i class="fa fa-check"></i><b>1.3.1</b> Number of peaks</a></li>
<li class="chapter" data-level="1.3.2" data-path="chipseq.html"><a href="chipseq.html#functional-discribution"><i class="fa fa-check"></i><b>1.3.2</b> Functional discribution</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chipseq.html"><a href="chipseq.html#overlap-of-peaks"><i class="fa fa-check"></i><b>1.4</b> Overlap of peaks</a><ul>
<li class="chapter" data-level="1.4.1" data-path="chipseq.html"><a href="chipseq.html#duxc-and-dux4"><i class="fa fa-check"></i><b>1.4.1</b> DUXC and DUX4</a></li>
<li class="chapter" data-level="1.4.2" data-path="chipseq.html"><a href="chipseq.html#duxc-and-duxc-alt"><i class="fa fa-check"></i><b>1.4.2</b> DUXC and DUXC-ALT</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="chipseq.html"><a href="chipseq.html#motifs-analysis"><i class="fa fa-check"></i><b>1.5</b> Motifs analysis</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="makeSE.html"><a href="makeSE.html"><i class="fa fa-check"></i><b>2</b> RNA-seq to DESeq2</a><ul>
<li class="chapter" data-level="2.1" data-path="makeSE.html"><a href="makeSE.html#rna-seq-preprocessing"><i class="fa fa-check"></i><b>2.1</b> RNA-seq preprocessing</a></li>
<li class="chapter" data-level="2.2" data-path="makeSE.html"><a href="makeSE.html#reads-counts"><i class="fa fa-check"></i><b>2.2</b> Reads counts</a><ul>
<li class="chapter" data-level="2.2.1" data-path="makeSE.html"><a href="makeSE.html#define-gene-features"><i class="fa fa-check"></i><b>2.2.1</b> Define gene features</a></li>
<li class="chapter" data-level="2.2.2" data-path="makeSE.html"><a href="makeSE.html#read-counts"><i class="fa fa-check"></i><b>2.2.2</b> Read counts</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="makeSE.html"><a href="makeSE.html#deseq2"><i class="fa fa-check"></i><b>2.3</b> DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html"><i class="fa fa-check"></i><b>3</b> DUXC transcriptome</a><ul>
<li class="chapter" data-level="3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#load-datasets-and-settings"><i class="fa fa-check"></i><b>3.1</b> Load datasets and settings</a></li>
<li class="chapter" data-level="3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#deseq2-1"><i class="fa fa-check"></i><b>3.2</b> DESeq2</a><ul>
<li class="chapter" data-level="3.2.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hypothesis-settings"><i class="fa fa-check"></i><b>3.2.1</b> hypothesis settings</a></li>
<li class="chapter" data-level="3.2.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cinc-duxc-expresison-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2.2</b> CinC: DUXC expresison in canine muscle cells</a></li>
<li class="chapter" data-level="3.2.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hinc-dux4-expression-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2.3</b> HinC: DUX4 expression in canine muscle cells</a></li>
<li class="chapter" data-level="3.2.4" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#duxc-alt-expression---caltinc"><i class="fa fa-check"></i><b>3.2.4</b> DUXC-ALT expression - CALTinC</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cross-species-models-dux4-expression-in-canine-and-murine"><i class="fa fa-check"></i><b>3.3</b> Cross-species models: DUX4 expression in canine and murine</a><ul>
<li class="chapter" data-level="3.3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-duxc-expression-in-canine"><i class="fa fa-check"></i><b>3.3.1</b> DUX4 and DUXC expression in canine</a></li>
<li class="chapter" data-level="3.3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-dux-expression-in-murine"><i class="fa fa-check"></i><b>3.3.2</b> DUX4 and Dux expression in murine</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html"><i class="fa fa-check"></i><b>4</b> DUXC and Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#duxc-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.1</b> DUXC transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs"><i class="fa fa-check"></i><b>4.1.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.1.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome"><i class="fa fa-check"></i><b>4.1.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#dux-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.2</b> Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.2.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs-1"><i class="fa fa-check"></i><b>4.2.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.2.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome-1"><i class="fa fa-check"></i><b>4.2.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html"><i class="fa fa-check"></i><b>5</b> DUXC transcriptome compared to early embryo transcriptome profile</a><ul>
<li class="chapter" data-level="5.1" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#mouse-2c-like-signature"><i class="fa fa-check"></i><b>5.1</b> Mouse 2C-like signature</a></li>
<li class="chapter" data-level="5.2" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#cleavage-stage-signature"><i class="fa fa-check"></i><b>5.2</b> cleavage-stage signature</a></li>
<li class="chapter" data-level="5.3" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#ipscs"><i class="fa fa-check"></i><b>5.3</b> iPSCs</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="repeats.html"><a href="repeats.html"><i class="fa fa-check"></i><b>6</b> Repeat elements</a><ul>
<li class="chapter" data-level="6.1" data-path="repeats.html"><a href="repeats.html#takeaway"><i class="fa fa-check"></i><b>6.1</b> Takeaway</a></li>
<li class="chapter" data-level="6.2" data-path="repeats.html"><a href="repeats.html#repeat-element-expression"><i class="fa fa-check"></i><b>6.2</b> Repeat element expression</a></li>
<li class="chapter" data-level="6.3" data-path="repeats.html"><a href="repeats.html#differential-analysis"><i class="fa fa-check"></i><b>6.3</b> Differential analysis</a><ul>
<li class="chapter" data-level="6.3.1" data-path="repeats.html"><a href="repeats.html#normalization"><i class="fa fa-check"></i><b>6.3.1</b> Normalization</a></li>
<li class="chapter" data-level="6.3.2" data-path="repeats.html"><a href="repeats.html#deseq2-results"><i class="fa fa-check"></i><b>6.3.2</b> DESeq2 results</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="repeats.html"><a href="repeats.html#repeat-family-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> Repeat family enrichment analysis</a><ul>
<li class="chapter" data-level="6.4.1" data-path="repeats.html"><a href="repeats.html#correlation-between-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>6.4.1</b> Correlation between HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="6.4.2" data-path="repeats.html"><a href="repeats.html#repeat-family-activation"><i class="fa fa-check"></i><b>6.4.2</b> Repeat family activation</a></li>
<li class="chapter" data-level="6.4.3" data-path="repeats.html"><a href="repeats.html#repeat-classs-activation"><i class="fa fa-check"></i><b>6.4.3</b> Repeat classs activation</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="repeats.html"><a href="repeats.html#ltr-retroelements-distribution-and-activation"><i class="fa fa-check"></i><b>6.5</b> LTR retroelements distribution and activation</a></li>
<li class="chapter" data-level="6.6" data-path="repeats.html"><a href="repeats.html#peaks-distribution-in-class-of-repeat-elements"><i class="fa fa-check"></i><b>6.6</b> Peaks distribution in class of repeat elements</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="LTR-activation.html"><a href="LTR-activation.html"><i class="fa fa-check"></i><b>7</b> LTR activation</a><ul>
<li class="chapter" data-level="7.1" data-path="LTR-activation.html"><a href="LTR-activation.html#takaway"><i class="fa fa-check"></i><b>7.1</b> Takaway</a></li>
<li class="chapter" data-level="7.2" data-path="LTR-activation.html"><a href="LTR-activation.html#background-in-murine-and-human"><i class="fa fa-check"></i><b>7.2</b> background in murine and human</a></li>
<li class="chapter" data-level="7.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-subfamily-activation-in-canine"><i class="fa fa-check"></i><b>7.3</b> ERVL subfamily activation in canine</a></li>
<li class="chapter" data-level="7.4" data-path="LTR-activation.html"><a href="LTR-activation.html#malr-activation-in-canine"><i class="fa fa-check"></i><b>7.4</b> MaLR activation in canine</a></li>
<li class="chapter" data-level="7.5" data-path="LTR-activation.html"><a href="LTR-activation.html#compare-to-cinc-and-hinc-to-hinh"><i class="fa fa-check"></i><b>7.5</b> Compare to CinC and HinC to HinH</a><ul>
<li class="chapter" data-level="7.5.1" data-path="LTR-activation.html"><a href="LTR-activation.html#copy-number"><i class="fa fa-check"></i><b>7.5.1</b> Copy number</a></li>
<li class="chapter" data-level="7.5.2" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-malr"><i class="fa fa-check"></i><b>7.5.2</b> ERVL-MaLR</a></li>
<li class="chapter" data-level="7.5.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl"><i class="fa fa-check"></i><b>7.5.3</b> ERVL</a></li>
<li class="chapter" data-level="7.5.4" data-path="LTR-activation.html"><a href="LTR-activation.html#erv1"><i class="fa fa-check"></i><b>7.5.4</b> ERV1</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="LTR-activation.html"><a href="LTR-activation.html#duxc-alt-and-ervl-transcriptions"><i class="fa fa-check"></i><b>7.6</b> DUXC-ALT and ERVL transcriptions</a><ul>
<li class="chapter" data-level="7.6.1" data-path="LTR-activation.html"><a href="LTR-activation.html#erv-subfamily-expression"><i class="fa fa-check"></i><b>7.6.1</b> ERV subfamily expression</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="non-LTR.html"><a href="non-LTR.html"><i class="fa fa-check"></i><b>8</b> Non-LTR retrotrasposons and satellites</a><ul>
<li class="chapter" data-level="8.1" data-path="non-LTR.html"><a href="non-LTR.html#lines"><i class="fa fa-check"></i><b>8.1</b> LINEs</a><ul>
<li class="chapter" data-level="8.1.1" data-path="non-LTR.html"><a href="non-LTR.html#l1s-activation"><i class="fa fa-check"></i><b>8.1.1</b> L1s activation</a></li>
<li class="chapter" data-level="8.1.2" data-path="non-LTR.html"><a href="non-LTR.html#line1-and-duxcdux4-binding-sites"><i class="fa fa-check"></i><b>8.1.2</b> LINE1 and DUXC/DUX4 binding sites</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="non-LTR.html"><a href="non-LTR.html#sine"><i class="fa fa-check"></i><b>8.2</b> SINE</a><ul>
<li class="chapter" data-level="8.2.1" data-path="non-LTR.html"><a href="non-LTR.html#distribution-of-sine-family"><i class="fa fa-check"></i><b>8.2.1</b> Distribution of SINE family</a></li>
<li class="chapter" data-level="8.2.2" data-path="non-LTR.html"><a href="non-LTR.html#expression-of-sines-in-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>8.2.2</b> Expression of SINEs in HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="8.2.3" data-path="non-LTR.html"><a href="non-LTR.html#trna-lys"><i class="fa fa-check"></i><b>8.2.3</b> tRNA-Lys</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="non-LTR.html"><a href="non-LTR.html#satellites"><i class="fa fa-check"></i><b>8.3</b> Satellites</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="peaks-line1.html"><a href="peaks-line1.html"><i class="fa fa-check"></i><b>9</b> DUXC peaks and LINE1s</a><ul>
<li class="chapter" data-level="9.1" data-path="peaks-line1.html"><a href="peaks-line1.html#narrow-peaks-and-expressed-l1s"><i class="fa fa-check"></i><b>9.1</b> Narrow Peaks and expressed L1s</a><ul>
<li class="chapter" data-level="9.1.1" data-path="peaks-line1.html"><a href="peaks-line1.html#motif-discovery-using-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.1</b> Motif discovery using L1-overlapped peaks</a></li>
<li class="chapter" data-level="9.1.2" data-path="peaks-line1.html"><a href="peaks-line1.html#match-dux4-and-duxc-motif-to-l1-overapped-peaks"><i class="fa fa-check"></i><b>9.1.2</b> Match DUX4 and DUXC motif to L1-overapped peaks</a></li>
<li class="chapter" data-level="9.1.3" data-path="peaks-line1.html"><a href="peaks-line1.html#location-of-aatca-on-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.3</b> Location of AATCA on L1-overlapped peaks</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="peaks-line1.html"><a href="peaks-line1.html#line1s-and-aatca"><i class="fa fa-check"></i><b>9.2</b> LINE1s and AATCA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i># References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="LTR-activation" class="section level1">
<h1><span class="header-section-number">Chapter 7</span> LTR activation</h1>
<p>Previously @ref{repeats}, we showd the repeat elemnets profiling which revealed LTR activation by DUXC and DUX4 in the canine model. Here we investigated the subfamilies of LTR - ERVL and ERVL-MaLR. In the canine model, DUX4 and DUXC both activate elements form MaLR and ERVL. In addition, we found that DUXC prones to activate more ERVL elements (MLT2D, MER76, HERVL74, LTR1A2, LTR101, LTR33C) wheareas DUX4 to more MaLR elements (MLT1A/B/E2/J).</p>
<p>In this chapter, we (1) explain the background of ERVL and ERVL-MaLR subfamilies in human and murine, (2) exam ERVL and ERVL-MaLR activation by DUXC and DUX4 in canine myoblast, and (3) compare to the LTR subfamily activation in human myoblast.</p>
<p><strong>Load and datsets and libraries</strong></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># define directory</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">pkg_dir &lt;-<span class="st"> &quot;~/CompBio/canFam3.DuxFamily&quot;</span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3">fig_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;figures&quot;</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">data_dir &lt;-<span class="st"> </span><span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="co"># load library</span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(DESeq2))</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(tidyverse))</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(tidyr))</a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(ggrepel))</a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(purrr))</a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(canFam3.rmsk))</a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="kw">load</span>(<span class="kw">file.path</span>(<span class="st">&quot;~/CompBio/hg38/hg38.rmsk.rda&quot;</span>))</a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(wesanderson)) <span class="co"># modern palettes</span></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"></a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="co"># load datasets and local functions</span></a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="kw">source</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;scripts&quot;</span>, <span class="st">&quot;tools.R&quot;</span>))</a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;HinC_rmsk.rda&quot;</span>)) <span class="co"># made in chapter 6</span></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;CinC_rmsk.rda&quot;</span>)) <span class="co"># made in chapter 6</span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;CALTinC_rmsk.rda&quot;</span>)) <span class="co"># made in chapter 6</span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="kw">load</span>(<span class="kw">file.path</span>(pkg_dir, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;HinH_rmsk.rda&quot;</span>))</a></code></pre></div>
<div id="takaway" class="section level2">
<h2><span class="header-section-number">7.1</span> Takaway</h2>
<p>The ERVL-MaLR elements in canine are categorized as MLT1, presenting ancient mammalian LTRs. Most of the MLT1s annotated in the canine genome were mobilized before the split of common ancient mammalian of canine and primate lineages, except MLT1A0 (-int) and MLT1A1 (-int) that might have become more mobilized in the primate lineage. Compared with DUXC, DUX4 induces higher expression levels of MLT1s in canine, though its affinity toward MLT1s has no apparent correlation with the copy number or mobilization in the primate lineage.</p>
<p><strong>ERVL1</strong>: ERV1 (denoted LTR* and MER*) are enriched in both DUX4 and DUXC expressed canine skeletal muscle; 50% and 18% of ERVL1 elements are positively affected by DUXC and DUXC, respectively. DUX4 induced more carnivore-specific ERV1 (CarLTR, CfERV1, CdERVF) than DUXC.</p>
<p><strong>ERVL-MaLR</strong>: Studies indicate that DUX4 uses MLT1 as promoter to re-program nearby genes’s structure during cleavage-stage and FSHD affected skeletal muscle. MLT1 presents ancient mammalian LTRs whereas THE1 and MST radiats in primate lineage. In canine genome, most of the ERVL-MaLR elements are MLT1 and mobilized before the primate lineages; the copy numbers of MLT1s existed in canine genome are compariable to that of in human genome, excpet MLT1A1 (2.2) and MLT1A0-int (&gt;100), both of which are DUX4 preferred repeats for retrontransposon-mediated transcriptome in humand and canine myoblosts. DUX4 and DUXC positively affect 51% and 25% of ERVLs, respectively.</p>
<p><strong>ERVL</strong>: (denoted MLT2, MER* and LTR*) DUX4 and DUXC positively affect 49% and 23% of ERVLs, respectively.</p>
</div>
<div id="background-in-murine-and-human" class="section level2">
<h2><span class="header-section-number">7.2</span> background in murine and human</h2>
<p>Though sharing common ancestry, MuERV-L (denoted MT2_Mm and MT2) evolved differntly from MaLR in murine. The ERVL-MaLR include three major subfamilies: MLT, ORR1 and MT. MLT; MLT1 and MLT2, present ancenstral mammalian LTR; ORR1 and MT are the youngest repeats in rodent, and they radiate during rodent evolution. (Note: MT, ORR1 nd MT2 radiate during rodent evolution.)</p>
<p>Whidden et al. <span class="citation">(Jennifer L. Whiddon <a href="#ref-whidden2017">2017</a>)</span>, Hackdrison et al. <span class="citation">(<span class="citeproc-not-found" data-reference-id="hackdrison2017"><strong>???</strong></span>)</span> and Franke et al <span class="citation">(Vedran Franke <a href="#ref-franke2017">2017</a>)</span> have investigated the activities of the mouse endogenous retrovirus type-L (MuERV-L) and nonautonomous mammalian (ERVL-MaLR) subfamilies under the influence of Dux and early embroynic activation. The studies confirm the species-specific preference on activation of retrotransposon subfamilies during the 2C-like (cleavage) stage or Dux/DUX4 over-expression: human bound to ancestral (MLT1) and primate-specific (THE1, MST) ERVL-MaLR subfamilies whereas murine to MuERV-L and young submailies of MaLR that evolved during rodent evolution.</p>
<p>Frank 2015 <span class="citation">(Vedran Franke <a href="#ref-franke2017">2017</a>)</span> showed that one of the ERVL functions is gene-remodeling via four co-options:<br />
1. 5’ exon via retrotransoson-derived spliced donor (SD)<br />
2. 3’ exon via retrotransposon-derived SD or poly-(A)<br />
3. internal exon via retrotransposon-derived SD or DA<br />
4. intraexonic, not contribute to mRNA process</p>
<p>In general, the younger ERVL subfamilies <span class="citation">(Vedran Franke <a href="#ref-franke2017">2017</a>)</span> - ORR1, MT, MT2 - showed higher propotion of 5’exon co-option than older family (MLT) in murine. Human bound to use more <em>MLT1</em> (subfamily of MaLR) 5’exon co-option, contributing <em>MLT</em> to promotor and first exons. Unlike rodent, human does not use MT2 for 5’exon co-option; Rodent uses more <em>MT</em> (subfamily of MaLR) and <em>MT2</em> (subfamily of MuERL-V) 5’exon co-option. Whidden et al. <span class="citation">(Jennifer L. Whiddon <a href="#ref-whidden2017">2017</a>)</span> confirmed that Dux and DUX4 express different subsets of MaLR transcripts in C2C12: Dux uses MT2 (MuERL-V) and MT as promoters whereas DUX4 uses <em>MLT</em> as promoters to activate the targets.</p>
<p>Previous chapter @ref{repeats} stated that DUXC showed higher propotion of ERVL differential expression whereas DUX4 showed higher propotion in that of MaLR. Althought we observe differntial expression in MaLR in DUXC and DUX4, the question is are they in distinct subfamilies of MaLR?</p>
</div>
<div id="ervl-subfamily-activation-in-canine" class="section level2">
<h2><span class="header-section-number">7.3</span> ERVL subfamily activation in canine</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># extract ERVL retroelement</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">duxc_ERVL &lt;-<span class="st"> </span>CinC_rmsk<span class="op">$</span>res_df <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(repFamily <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ERVL&quot;</span>, <span class="st">&quot;ERVL?&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">up_reg =</span> <span class="kw">ifelse</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&gt;</span><span class="dv">0</span>, <span class="ot">TRUE</span>, <span class="ot">FALSE</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">up_reg =</span> <span class="kw">replace</span>(up_reg, <span class="kw">is.na</span>(up_reg), <span class="ot">FALSE</span>))</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">                                         </a>
<a class="sourceLine" id="cb2-7" data-line-number="7">dux4_ERVL &lt;-<span class="st"> </span>HinC_rmsk<span class="op">$</span>res_df <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(repFamily <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ERVL&quot;</span>,<span class="st">&quot;ERVL?&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">up_reg =</span> <span class="kw">ifelse</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="ot">TRUE</span>, <span class="ot">FALSE</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">up_reg =</span> <span class="kw">replace</span>(up_reg, <span class="kw">is.na</span>(up_reg), <span class="ot">FALSE</span>))</a>
<a class="sourceLine" id="cb2-11" data-line-number="11"></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co"># combine dux4 and duxc and make scatter plot; label DUX4 and DUX4 only</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13">ERVL &lt;-<span class="st"> </span><span class="kw">full_join</span>(duxc_ERVL, dux4_ERVL, <span class="dt">by=</span><span class="st">&quot;repName&quot;</span>, <span class="dt">suffix=</span><span class="kw">c</span>(<span class="st">&quot;_DUXC&quot;</span>, <span class="st">&quot;_DUX4&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">status =</span> <span class="st">&quot;none&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">status =</span> <span class="kw">replace</span>(status, padj_DUX4 <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, <span class="st">&quot;DUX4&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">status =</span> <span class="kw">replace</span>(status, padj_DUXC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, <span class="st">&quot;DUXC&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">status =</span> <span class="kw">replace</span>(status, padj_DUXC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>padj_DUX4 <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, <span class="st">&quot;both&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">status =</span> <span class="kw">factor</span>(status, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;none&quot;</span>,<span class="st">&quot;DUX4&quot;</span>, <span class="st">&quot;DUXC&quot;</span>, <span class="st">&quot;both&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">show_name =</span> <span class="kw">as.character</span>(repName)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb2-20" data-line-number="20"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">show_name =</span> <span class="kw">ifelse</span>(status <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;DUX4&quot;</span>, <span class="st">&quot;DUXC&quot;</span>, <span class="st">&quot;both&quot;</span>), show_name, <span class="st">&quot;&quot;</span>))</a></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">my_color &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;#999999&quot;</span>, </a>
<a class="sourceLine" id="cb3-2" data-line-number="2">              <span class="kw">as.character</span>(<span class="kw">wes_palette</span>(<span class="dt">n=</span><span class="dv">5</span>, <span class="dt">name=</span><span class="st">&quot;FantasticFox1&quot;</span>, <span class="dt">type=</span><span class="st">&quot;discrete&quot;</span>))[<span class="dv">3</span><span class="op">:</span><span class="dv">5</span>])</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">gg1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(ERVL, <span class="kw">aes</span>(<span class="dt">x=</span>log2FoldChange_DUXC, <span class="dt">y=</span>log2FoldChange_DUX4, <span class="dt">color=</span>status)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>my_color) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="st">  </span><span class="co">#geom_smooth(method=&quot;lm&quot;, se=FALSE, show.legend=FALSE, color=&quot;gray66&quot;, alpha=0.5,</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="st">  </span><span class="co">#            linetype=&quot;dashed&quot;) +</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="st">  </span><span class="kw">geom_abline</span>(<span class="dt">intercept=</span><span class="dv">0</span>, <span class="dt">slope=</span><span class="dv">1</span>, <span class="dt">color=</span><span class="st">&quot;gray75&quot;</span>, <span class="dt">alpha=</span><span class="fl">0.2</span>, <span class="dt">size=</span><span class="fl">0.5</span>) <span class="op">+</span><span class="st">              </span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;ERVL: HinC vs CinC log2FC&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="st">  </span><span class="kw">theme_bw</span>()<span class="op">+</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.83</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="st">  </span><span class="kw">geom_text_repel</span>(<span class="kw">aes</span>(<span class="dt">label=</span>show_name), <span class="dt">size=</span><span class="fl">2.5</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb3-13" data-line-number="13">gg1</a></code></pre></div>
<div class="figure"><span id="fig:ERVL-scatter-plot"></span>
<img src="gitbook_files/figure-html/ERVL-scatter-plot-1.png" alt="Scatter plot of log2FC of DUXC and DUX4" width="672" />
<p class="caption">
Figure 7.1: Scatter plot of log2FC of DUXC and DUX4
</p>
</div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="kw">file.path</span>(fig_dir, <span class="st">&quot;rmsk_ERVL_DUX4_vs_DUXC.pdf&quot;</span>), </a>
<a class="sourceLine" id="cb4-2" data-line-number="2">       <span class="dt">width=</span><span class="dv">5</span>, <span class="dt">height=</span><span class="dv">5</span>, <span class="dt">device=</span><span class="st">&quot;pdf&quot;</span>)</a></code></pre></div>
</div>
<div id="malr-activation-in-canine" class="section level2">
<h2><span class="header-section-number">7.4</span> MaLR activation in canine</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># extract ERVL subfamilies</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2">duxc_MaLR &lt;-<span class="st"> </span>CinC_rmsk<span class="op">$</span>res_df <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(repFamily <span class="op">==</span><span class="st"> &quot;ERVL-MaLR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">up_reg =</span> <span class="kw">ifelse</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&gt;</span><span class="dv">0</span>, <span class="ot">TRUE</span>, <span class="ot">FALSE</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">up_reg =</span> <span class="kw">replace</span>(up_reg, <span class="kw">is.na</span>(up_reg), <span class="ot">FALSE</span>))</a>
<a class="sourceLine" id="cb5-6" data-line-number="6">                                         </a>
<a class="sourceLine" id="cb5-7" data-line-number="7">dux4_MaLR &lt;-<span class="st"> </span>HinC_rmsk<span class="op">$</span>res_df <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(repFamily <span class="op">==</span><span class="st"> &quot;ERVL-MaLR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">up_reg =</span> <span class="kw">ifelse</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="ot">TRUE</span>, <span class="ot">FALSE</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">up_reg =</span> <span class="kw">replace</span>(up_reg, <span class="kw">is.na</span>(up_reg), <span class="ot">FALSE</span>))</a>
<a class="sourceLine" id="cb5-11" data-line-number="11"></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co"># combine dux4 and duxc and make scatter plot; label DUX4 and DUX4 only</span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13">MaLR &lt;-<span class="st"> </span><span class="kw">full_join</span>(duxc_MaLR, dux4_MaLR, <span class="dt">by=</span><span class="st">&quot;repName&quot;</span>, <span class="dt">suffix=</span><span class="kw">c</span>(<span class="st">&quot;_DUXC&quot;</span>, <span class="st">&quot;_DUX4&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">status =</span> <span class="st">&quot;none&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">status =</span> <span class="kw">replace</span>(status, padj_DUX4 <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, <span class="st">&quot;DUX4&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">status =</span> <span class="kw">replace</span>(status, padj_DUXC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, <span class="st">&quot;DUXC&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">status =</span> <span class="kw">replace</span>(status, padj_DUXC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>padj_DUX4 <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, <span class="st">&quot;both&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">status =</span> <span class="kw">factor</span>(status, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;none&quot;</span>,<span class="st">&quot;DUX4&quot;</span>, <span class="st">&quot;DUXC&quot;</span>, <span class="st">&quot;both&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">show_name =</span> <span class="kw">as.character</span>(repName)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">show_name =</span> <span class="kw">ifelse</span>(status <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;DUX4&quot;</span>, <span class="st">&quot;DUXC&quot;</span>, <span class="st">&quot;both&quot;</span>), show_name, <span class="st">&quot;&quot;</span>))</a></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">my_color &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;#999999&quot;</span>, </a>
<a class="sourceLine" id="cb6-2" data-line-number="2">              <span class="kw">as.character</span>(<span class="kw">wes_palette</span>(<span class="dt">n=</span><span class="dv">5</span>, <span class="dt">name=</span><span class="st">&quot;FantasticFox1&quot;</span>, <span class="dt">type=</span><span class="st">&quot;discrete&quot;</span>))[<span class="dv">3</span><span class="op">:</span><span class="dv">5</span>])</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">gg1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(MaLR, <span class="kw">aes</span>(<span class="dt">x=</span>log2FoldChange_DUXC, <span class="dt">y=</span>log2FoldChange_DUX4, <span class="dt">color=</span>status)) <span class="op">+</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">1</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>my_color) <span class="op">+</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="st">  </span><span class="co">#geom_smooth(method=&quot;lm&quot;, se=FALSE, show.legend=FALSE, color=&quot;gray66&quot;, alpha=0.5,</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="st">  </span><span class="co">#            linetype=&quot;dashed&quot;) +</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="st">  </span><span class="kw">geom_abline</span>(<span class="dt">intercept=</span><span class="dv">0</span>, <span class="dt">slope=</span><span class="dv">1</span>, <span class="dt">color=</span><span class="st">&quot;gray75&quot;</span>, <span class="dt">alpha=</span><span class="fl">0.2</span>, <span class="dt">size=</span><span class="fl">0.5</span>) <span class="op">+</span><span class="st">              </span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;ERVL-MaLR: HinC vs CinC log2FC&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="st">  </span><span class="kw">theme_bw</span>()<span class="op">+</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.83</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="st">  </span><span class="kw">geom_text_repel</span>(<span class="kw">aes</span>(<span class="dt">label=</span>show_name), <span class="dt">size=</span><span class="fl">2.5</span>, <span class="dt">show.legend=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb6-13" data-line-number="13">gg1</a></code></pre></div>
<div class="figure"><span id="fig:MaLR-scatter-plot"></span>
<img src="gitbook_files/figure-html/MaLR-scatter-plot-1.png" alt="Scatter plot of log2FC of DUXC and DUX4" width="672" />
<p class="caption">
Figure 7.2: Scatter plot of log2FC of DUXC and DUX4
</p>
</div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="kw">file.path</span>(fig_dir, <span class="st">&quot;rmsk_MaLR_DUX4_vs_DUXC.pdf&quot;</span>), </a>
<a class="sourceLine" id="cb7-2" data-line-number="2">       <span class="dt">width=</span><span class="dv">5</span>, <span class="dt">height=</span><span class="dv">5</span>, <span class="dt">device=</span><span class="st">&quot;pdf&quot;</span>)</a></code></pre></div>
<p>Visualizaiton of the number of ERVL subfamilies activated by DUX4 and DUXC in canine.</p>
</div>
<div id="compare-to-cinc-and-hinc-to-hinh" class="section level2">
<h2><span class="header-section-number">7.5</span> Compare to CinC and HinC to HinH</h2>
<p>Here we compared LTR retroelment activation of CinC and HinC to that of HinH. We aimed to answer whether copy number is associated with higher induction by DUX4 than by DUXC? Update: The answer is no. Copy number is not associated with higher induction of DUX4 nor DUXC.</p>
<div id="copy-number" class="section level3">
<h3><span class="header-section-number">7.5.1</span> Copy number</h3>
<p>The code below estimates the copy number of LTR subfamilies in hg38 and canFam3.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># get the number of instance of a repeat element within a family</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2">get_family_copy_number &lt;-<span class="st"> </span><span class="cf">function</span>(hg38.rmsk, canFam3.rmsk, family_name) {</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">  <span class="cf">if</span> (<span class="op">!</span><span class="kw">any</span>(<span class="kw">c</span>(<span class="kw">levels</span>(hg38.rmsk<span class="op">$</span>repFamily), <span class="kw">levels</span>(hg38.rmsk<span class="op">$</span>repFamilf)) <span class="op">%in%</span><span class="st"> </span>family_name))</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">    <span class="kw">stop</span>(<span class="kw">paste</span>(family_name, <span class="dt">sep=</span><span class="st">&quot;, &quot;</span>), <span class="st">&quot; not found in both hg38 and canFam3 rmsk!&quot;</span>)</a>
<a class="sourceLine" id="cb8-5" data-line-number="5">  </a>
<a class="sourceLine" id="cb8-6" data-line-number="6">  hg38_sub&lt;-<span class="st"> </span>plyranges<span class="op">::</span><span class="kw">filter</span>(hg38.rmsk, repFamily <span class="op">%in%</span><span class="st"> </span>family_name)  <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="st">    </span>plyranges<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">repClass =</span> <span class="kw">factor</span>(repClass), <span class="dt">repFamily =</span> <span class="kw">factor</span>(repFamily),</a>
<a class="sourceLine" id="cb8-8" data-line-number="8">                      <span class="dt">repName =</span> <span class="kw">factor</span>(repName)) </a>
<a class="sourceLine" id="cb8-9" data-line-number="9">  hg38_copy_n &lt;-<span class="st"> </span><span class="kw">as</span>(<span class="kw">mcols</span>(hg38_sub), <span class="st">&quot;data.frame&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">count</span>(repName)</a>
<a class="sourceLine" id="cb8-10" data-line-number="10">                    </a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="co"># canine</span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12">  canFam3_sub &lt;-<span class="st"> </span>plyranges<span class="op">::</span><span class="kw">filter</span>(canFam3.rmsk, repFamily <span class="op">%in%</span><span class="st"> </span>family_name) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="st">    </span>plyranges<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">repClass =</span> <span class="kw">factor</span>(repClass), <span class="dt">repFamily =</span> <span class="kw">factor</span>(repFamily),</a>
<a class="sourceLine" id="cb8-14" data-line-number="14">                      <span class="dt">repName =</span> <span class="kw">factor</span>(repName))</a>
<a class="sourceLine" id="cb8-15" data-line-number="15">  canFam3_copy_n &lt;-<span class="st"> </span><span class="kw">as</span>(<span class="kw">mcols</span>(canFam3_sub), <span class="st">&quot;data.frame&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">count</span>(repName)</a>
<a class="sourceLine" id="cb8-16" data-line-number="16"></a>
<a class="sourceLine" id="cb8-17" data-line-number="17">  <span class="kw">full_join</span>(canFam3_copy_n, hg38_copy_n, </a>
<a class="sourceLine" id="cb8-18" data-line-number="18">            <span class="dt">by=</span><span class="st">&quot;repName&quot;</span>, <span class="dt">suffix=</span><span class="kw">c</span>(<span class="st">&quot;.canFam3&quot;</span>, <span class="st">&quot;.hg38&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-19" data-line-number="19"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">ratio =</span> n.hg38 <span class="op">/</span><span class="st"> </span>n.canFam3)</a>
<a class="sourceLine" id="cb8-20" data-line-number="20">}</a>
<a class="sourceLine" id="cb8-21" data-line-number="21"></a>
<a class="sourceLine" id="cb8-22" data-line-number="22"><span class="co"># estimate copy number of MaLR, ERV1 and ERVL retroelemtns in canine and human using RMSK</span></a>
<a class="sourceLine" id="cb8-23" data-line-number="23">MaLR_copy_number &lt;-<span class="st"> </span><span class="kw">get_family_copy_number</span>(hg38.rmsk, canFam3.rmsk, <span class="st">&quot;ERVL-MaLR&quot;</span>)</a>
<a class="sourceLine" id="cb8-24" data-line-number="24">ERVL_copy_number &lt;-<span class="st"> </span><span class="kw">get_family_copy_number</span>(hg38.rmsk, canFam3.rmsk, <span class="st">&quot;ERVL&quot;</span>)</a>
<a class="sourceLine" id="cb8-25" data-line-number="25">ERV1_copy_number &lt;-<span class="st"> </span><span class="kw">get_family_copy_number</span>(hg38.rmsk, canFam3.rmsk, <span class="st">&quot;ERV1&quot;</span>)</a>
<a class="sourceLine" id="cb8-26" data-line-number="26"><span class="co"># HERV: expended in human/primate lineage</span></a>
<a class="sourceLine" id="cb8-27" data-line-number="27"><span class="kw">head</span>(MaLR_copy_number)</a></code></pre></div>
<pre><code>##      repName n.canFam3 n.hg38    ratio
## 1    MLT-int       171    394 2.304094
## 2   MLT1-int       226    251 1.110619
## 3      MLT1A      8364   9312 1.113343
## 4  MLT1A-int       371    456 1.229111
## 5     MLT1A0     11780  21235 1.802632
## 6 MLT1A0-int       414    971 2.345411</code></pre>
<p>Tidy up activated LTR retroelements MaLR, ERVL, and ERV1) in HinC, CinC and HinH together.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="co"># 1. union the activated LTR retroelements of HinC and  CinC; then append HinH&#39;s statistics</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co"># 2. incorporate with copy number</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3">activated_LTRs &lt;-<span class="st"> </span><span class="kw">inner_join</span>(HinC_rmsk<span class="op">$</span>res_df, CinC_rmsk<span class="op">$</span>res_df, </a>
<a class="sourceLine" id="cb10-4" data-line-number="4">                   <span class="dt">by=</span><span class="st">&quot;repName&quot;</span>, <span class="dt">suffix=</span><span class="kw">c</span>(<span class="st">&quot;_HinC&quot;</span>, <span class="st">&quot;_CinC&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="st">  </span><span class="kw">left_join</span>(dplyr<span class="op">::</span><span class="kw">select</span>(HinH_rmsk<span class="op">$</span>res_df, repName, log2FoldChange, padj), <span class="dt">by=</span><span class="st">&quot;repName&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="st">  </span><span class="kw">rename</span>(<span class="dt">repFamily =</span> repFamily_CinC, <span class="dt">log2FoldChange_HinH =</span> log2FoldChange, <span class="dt">padj_HinH =</span> padj) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(repFamily <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ERVL-MaLR&quot;</span>, <span class="st">&quot;ERVL&quot;</span>, <span class="st">&quot;ERV1&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(log2FoldChange_HinC <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span> <span class="op">|</span><span class="st"> </span>log2FoldChange_CinC <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span> <span class="op">|</span><span class="st"> </span>log2FoldChange_HinH <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(padj_HinC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">|</span><span class="st"> </span>padj_CinC <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">|</span><span class="st"> </span>padj_HinH <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(repName, repFamily, </a>
<a class="sourceLine" id="cb10-11" data-line-number="11">                log2FoldChange_HinC, padj_HinC, log2FoldChange_CinC, padj_CinC,</a>
<a class="sourceLine" id="cb10-12" data-line-number="12">                log2FoldChange_HinH, padj_HinH)</a>
<a class="sourceLine" id="cb10-13" data-line-number="13">      </a>
<a class="sourceLine" id="cb10-14" data-line-number="14">logfc_gather &lt;-<span class="st"> </span><span class="kw">gather</span>(activated_LTRs, <span class="dt">key=</span>tr_factor, <span class="dt">value=</span>logFC, </a>
<a class="sourceLine" id="cb10-15" data-line-number="15">                       log2FoldChange_HinC, log2FoldChange_CinC, log2FoldChange_HinH, </a>
<a class="sourceLine" id="cb10-16" data-line-number="16">                       <span class="op">-</span>repName, <span class="op">-</span>repFamily) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-17" data-line-number="17"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">padj =</span> padj_HinC) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-18" data-line-number="18"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">padj =</span> <span class="kw">ifelse</span>(tr_factor <span class="op">==</span><span class="st"> &quot;log2FoldChange_CinC&quot;</span>, padj_CinC, padj)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-19" data-line-number="19"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">padj =</span> <span class="kw">ifelse</span>(tr_factor <span class="op">==</span><span class="st"> &quot;log2FoldChange_HinH&quot;</span>, padj_HinH, padj)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-20" data-line-number="20"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(repName, repFamily, tr_factor, logFC, padj) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-21" data-line-number="21"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">sig =</span> <span class="kw">ifelse</span>(padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, <span class="ot">TRUE</span>, <span class="ot">FALSE</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-22" data-line-number="22"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">tr_factor =</span> <span class="kw">sapply</span>(<span class="kw">strsplit</span>(tr_factor, <span class="st">&quot;_&quot;</span>, <span class="dt">fixed=</span><span class="ot">TRUE</span>), <span class="st">&quot;[[&quot;</span>, <span class="dv">2</span>))</a></code></pre></div>
</div>
<div id="ervl-malr" class="section level3">
<h3><span class="header-section-number">7.5.2</span> ERVL-MaLR</h3>
<p>The bar plot below displays signicantly expressed ERVL-MaLR elements in either CinC, HinC or HinH (<span class="math inline">\(adj-p &lt; 0.05\)</span>). Only <em>MLT1A1</em> and <em>MLT1A0-int</em> have higher copy number in human (ratio &gt; 2) compared to canine (ratio &gt; 0), but they don’t have higher induction in human.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">data &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(logfc_gather, repFamily <span class="op">==</span><span class="st"> &quot;ERVL-MaLR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="st">  </span><span class="kw">left_join</span>(MaLR_copy_number, <span class="dt">by=</span><span class="st">&quot;repName&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">repName =</span> <span class="kw">ifelse</span>(ratio <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>, <span class="kw">paste0</span>(repName, <span class="st">&quot;*&quot;</span>), repName))</a>
<a class="sourceLine" id="cb11-4" data-line-number="4"></a>
<a class="sourceLine" id="cb11-5" data-line-number="5">gg1 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(data, <span class="kw">aes</span>(<span class="dt">x =</span> repName, <span class="dt">y=</span>logFC, <span class="dt">fill=</span>sig)) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="st">  </span><span class="kw">facet_wrap</span> ( <span class="op">~</span><span class="st"> </span>tr_factor) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-8" data-line-number="8"><span class="st">  </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb11-9" data-line-number="9"><span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;ERVL-MaLR&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#999999&quot;</span>, <span class="st">&quot;#E69F00&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb11-12" data-line-number="12"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>)</a>
<a class="sourceLine" id="cb11-13" data-line-number="13">gg1</a></code></pre></div>
<div class="figure"><span id="fig:plot-MaLR-logFC"></span>
<img src="gitbook_files/figure-html/plot-MaLR-logFC-1.png" alt="repName expression (logFC) in ERVL-MaLR family in CinC, HinC and HinH; Orange indicates up/down-regulation; * indicates higher copy number in human." width="672" />
<p class="caption">
Figure 7.3: repName expression (logFC) in ERVL-MaLR family in CinC, HinC and HinH; Orange indicates up/down-regulation; * indicates higher copy number in human.
</p>
</div>
</div>
<div id="ervl" class="section level3">
<h3><span class="header-section-number">7.5.3</span> ERVL</h3>
<p>Shown below is the significently expressed ERVL elememts expression in either CinC, HinC, or HinH. <em>MER21B</em> and <em>MER21-int</em> have highr copy number in human.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">data &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(logfc_gather, repFamily <span class="op">==</span><span class="st"> &quot;ERVL&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="st">  </span><span class="kw">left_join</span>(ERVL_copy_number, <span class="dt">by=</span><span class="st">&quot;repName&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">repName =</span> <span class="kw">ifelse</span>(ratio <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>, <span class="kw">paste0</span>(repName, <span class="st">&quot;*&quot;</span>), repName))</a>
<a class="sourceLine" id="cb12-4" data-line-number="4"></a>
<a class="sourceLine" id="cb12-5" data-line-number="5">gg2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(data, <span class="kw">aes</span>(<span class="dt">x =</span> repName, <span class="dt">y=</span>logFC, <span class="dt">fill=</span>sig)) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7"><span class="st">  </span><span class="kw">facet_wrap</span> ( <span class="op">~</span><span class="st"> </span>tr_factor) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-8" data-line-number="8"><span class="st">  </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-10" data-line-number="10"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;ERVL&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-11" data-line-number="11"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#999999&quot;</span>, <span class="st">&quot;#E69F00&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-12" data-line-number="12"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>)</a>
<a class="sourceLine" id="cb12-13" data-line-number="13">gg2</a></code></pre></div>
<div class="figure"><span id="fig:plot-ERVL-logFC"></span>
<img src="gitbook_files/figure-html/plot-ERVL-logFC-1.png" alt="repName expression (logFC) in ERVL family in CinC, HinC and HinH; Orange indicates up/down-regulation; * indicates higher copy number in human." width="672" />
<p class="caption">
Figure 7.4: repName expression (logFC) in ERVL family in CinC, HinC and HinH; Orange indicates up/down-regulation; * indicates higher copy number in human.
</p>
</div>
</div>
<div id="erv1" class="section level3">
<h3><span class="header-section-number">7.5.4</span> ERV1</h3>
<p>Shown below is the significently expressed ERV1 elements in either CinC, HinC, or HinH. Suprisingly, DUX4 up-regulated more carnior-specific retroelement than DUXC in canine.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">data &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(logfc_gather, repFamily <span class="op">==</span><span class="st"> &quot;ERV1&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="st">  </span><span class="kw">left_join</span>(ERV1_copy_number, <span class="dt">by=</span><span class="st">&quot;repName&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">repName =</span> <span class="kw">ifelse</span>(<span class="op">!</span><span class="kw">is.na</span>(ratio) <span class="op">&amp;</span><span class="st"> </span>ratio <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>, <span class="kw">paste0</span>(repName, <span class="st">&quot;*&quot;</span>), repName)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">repName =</span> <span class="kw">ifelse</span>(<span class="kw">is.na</span>(ratio), <span class="kw">paste0</span>(repName, <span class="st">&quot;#&quot;</span>), repName))</a>
<a class="sourceLine" id="cb13-5" data-line-number="5"> </a>
<a class="sourceLine" id="cb13-6" data-line-number="6">gg3 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(data, <span class="kw">aes</span>(<span class="dt">x =</span> repName, <span class="dt">y=</span>logFC, <span class="dt">fill=</span>sig)) <span class="op">+</span></a>
<a class="sourceLine" id="cb13-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb13-8" data-line-number="8"><span class="st">  </span><span class="kw">facet_wrap</span> ( <span class="op">~</span><span class="st"> </span>tr_factor) <span class="op">+</span></a>
<a class="sourceLine" id="cb13-9" data-line-number="9"><span class="st">  </span><span class="kw">coord_flip</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb13-10" data-line-number="10"><span class="st">  </span><span class="kw">theme_minimal</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb13-11" data-line-number="11"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;ERV1&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb13-12" data-line-number="12"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#999999&quot;</span>, <span class="st">&quot;#E69F00&quot;</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb13-13" data-line-number="13"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>)</a>
<a class="sourceLine" id="cb13-14" data-line-number="14">gg3</a></code></pre></div>
<div class="figure"><span id="fig:plot-ERV1-logFC"></span>
<img src="gitbook_files/figure-html/plot-ERV1-logFC-1.png" alt="repName expression (logFC) in ERV1 family in CinC, HinC and HinH; Orange indicates up/down-regulation; * indicates higher copy number in human. # indicates the copy number ratio (human/canine) is not available." width="672" />
<p class="caption">
Figure 7.5: repName expression (logFC) in ERV1 family in CinC, HinC and HinH; Orange indicates up/down-regulation; * indicates higher copy number in human. # indicates the copy number ratio (human/canine) is not available.
</p>
</div>
</div>
</div>
<div id="duxc-alt-and-ervl-transcriptions" class="section level2">
<h2><span class="header-section-number">7.6</span> DUXC-ALT and ERVL transcriptions</h2>
<p>Does DUXC-ALT induce ERVL expression? There are 683 repeat elements exhibiting some expression in DUXC-ALT expressed cell line. The binding sites of DUXC-ALT suggested that DUXC-ALT mostly binds to LTR retroelements. We aim to exam further into DUXC-ALT’s function by looking at the expession fold changes.</p>
<div id="erv-subfamily-expression" class="section level3">
<h3><span class="header-section-number">7.6.1</span> ERV subfamily expression</h3>
<p>Shown below is the histogram of DUXC-ALT’s repeat elements expression. Most the expression of the expression of repeat elements do not have apparent changes by DUXC-ALT, however, some ERV transcripts are present with mild fold changes. This histogram below shows that a few repeat element transcripts are present with mild fold changes.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">calt_df &lt;-<span class="st"> </span>CALTinC_rmsk<span class="op">$</span>res_df</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="kw">ggplot</span>(calt_df, <span class="kw">aes</span>(<span class="dt">x=</span>log2FoldChange)) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="st">  </span><span class="kw">geom_histogram</span>(<span class="dt">bins=</span><span class="dv">30</span>, <span class="dt">color=</span><span class="st">&quot;white&quot;</span>, <span class="dt">fill=</span><span class="st">&quot;#DD8D29&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;Repeat element expression in DUXC-ALT&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="st">  </span><span class="kw">coord_cartesian</span>(<span class="dt">xlim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">2</span>,<span class="dv">2</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">10</span>))</a></code></pre></div>
<div class="figure"><span id="fig:calt-rmsk-histogram"></span>
<img src="gitbook_files/figure-html/calt-rmsk-histogram-1.png" alt="Histogram of log2 Fold Change of CALTinC repeat elements" width="672" />
<p class="caption">
Figure 7.6: Histogram of log2 Fold Change of CALTinC repeat elements
</p>
</div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw">ggsave</span>(<span class="dt">filename=</span><span class="kw">file.path</span>(fig_dir, <span class="st">&quot;rmsk_calt_histogram.pdf&quot;</span>),</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">       <span class="dt">width=</span><span class="dv">3</span>, <span class="dt">height=</span><span class="dv">2</span>)</a></code></pre></div>
<p>Disregard the <em>p</em>-adj and <em>p</em>-value, we listed the repeat elements with log2 fold change greater and less then 0.5, and most of them are ERVL1. This might support the peak calling of DUXC-Alt where more than 60% of binding sites are overlapping with LTRs.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="co"># generate table of repeat element with log2FC &gt; 0.5</span></a>
<a class="sourceLine" id="cb16-2" data-line-number="2">calt_up &lt;-<span class="st"> </span>calt_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(log2FoldChange <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>)</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">calt_up <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(repFamily <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ERV1&quot;</span>, <span class="st">&quot;ERVL?&quot;</span>, <span class="st">&quot;ERVL&quot;</span>, <span class="st">&quot;ERVL-MaLR&quot;</span>, <span class="st">&quot;Gypsy&quot;</span>, <span class="st">&quot;LTR&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>baseMean, <span class="op">-</span>lfcSE, <span class="op">-</span>stat) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="st">  </span>knitr<span class="op">::</span><span class="kw">kable</span>(<span class="dt">caption=</span><span class="st">&quot;Repeat element with log2FC &gt; 0.5 in DUXC-Alt.&quot;</span>)</a></code></pre></div>
<table>
<caption><span id="tab:calt-rmsk-upregulated">Table 7.1: </span>Repeat element with log2FC &gt; 0.5 in DUXC-Alt.</caption>
<thead>
<tr class="header">
<th align="left">repName</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
<th align="left">repClass</th>
<th align="left">repFamily</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CarERV2b1_LTR</td>
<td align="right">0.7129453</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERV1</td>
</tr>
<tr class="even">
<td align="left">CarLTR1A1C</td>
<td align="right">0.6510292</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERV1</td>
</tr>
<tr class="odd">
<td align="left">LTR109A2</td>
<td align="right">0.8342682</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERV1</td>
</tr>
<tr class="even">
<td align="left">LTR55</td>
<td align="right">0.6147695</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERVL?</td>
</tr>
<tr class="odd">
<td align="left">LTR72B_CF</td>
<td align="right">0.6561270</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERV1</td>
</tr>
<tr class="even">
<td align="left">LTR81B</td>
<td align="right">0.5263868</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">Gypsy</td>
</tr>
<tr class="odd">
<td align="left">LTR90B</td>
<td align="right">0.9274856</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">LTR</td>
</tr>
<tr class="even">
<td align="left">MamGypLTR3</td>
<td align="right">0.7477494</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">Gypsy</td>
</tr>
<tr class="odd">
<td align="left">MamGypLTR3a</td>
<td align="right">0.6591307</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">Gypsy</td>
</tr>
<tr class="even">
<td align="left">MER101_CF</td>
<td align="right">0.5369590</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERV1</td>
</tr>
<tr class="odd">
<td align="left">MER88</td>
<td align="right">0.8910420</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERVL</td>
</tr>
<tr class="even">
<td align="left">MLT2E</td>
<td align="right">0.6513342</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERVL</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="co"># generate table of repeat elements with log2FC &lt; -1</span></a>
<a class="sourceLine" id="cb17-2" data-line-number="2">calt_down &lt;-<span class="st"> </span>calt_df <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(log2FoldChange <span class="op">&lt;</span><span class="st"> </span><span class="dv">-1</span>)</a>
<a class="sourceLine" id="cb17-3" data-line-number="3">calt_down <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb17-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>baseMean, <span class="op">-</span>lfcSE, <span class="op">-</span>stat) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb17-5" data-line-number="5"><span class="st">  </span>knitr<span class="op">::</span><span class="kw">kable</span>(<span class="dt">caption=</span><span class="st">&quot;Repeat element with log2FC &lt;  -1 in DUXC-Alt.&quot;</span>)</a></code></pre></div>
<table>
<caption><span id="tab:calt-rmsk-downregulated">Table 7.2: </span>Repeat element with log2FC &lt; -1 in DUXC-Alt.</caption>
<thead>
<tr class="header">
<th align="left">repName</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
<th align="left">repClass</th>
<th align="left">repFamily</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">(CATCG)n</td>
<td align="right">-1.378227</td>
<td align="right">0.8176728</td>
<td align="right">1</td>
<td align="left">Simple_repeat</td>
<td align="left">Simple_repeat</td>
</tr>
<tr class="even">
<td align="left">(CCCCAA)n</td>
<td align="right">-1.329845</td>
<td align="right">0.8430315</td>
<td align="right">1</td>
<td align="left">Simple_repeat</td>
<td align="left">Simple_repeat</td>
</tr>
<tr class="odd">
<td align="left">(CCGGA)n</td>
<td align="right">-1.430167</td>
<td align="right">0.7516265</td>
<td align="right">1</td>
<td align="left">Simple_repeat</td>
<td align="left">Simple_repeat</td>
</tr>
<tr class="even">
<td align="left">(GAGAA)n</td>
<td align="right">-3.657527</td>
<td align="right">0.4001370</td>
<td align="right">1</td>
<td align="left">Simple_repeat</td>
<td align="left">Simple_repeat</td>
</tr>
<tr class="odd">
<td align="left">(GTGTG)n</td>
<td align="right">-1.087761</td>
<td align="right">0.9440174</td>
<td align="right">1</td>
<td align="left">Simple_repeat</td>
<td align="left">Simple_repeat</td>
</tr>
<tr class="even">
<td align="left">(TAGG)n</td>
<td align="right">-2.614182</td>
<td align="right">0.3792413</td>
<td align="right">1</td>
<td align="left">Simple_repeat</td>
<td align="left">Simple_repeat</td>
</tr>
<tr class="odd">
<td align="left">(TTCGGG)n</td>
<td align="right">-1.300816</td>
<td align="right">0.4825708</td>
<td align="right">1</td>
<td align="left">Simple_repeat</td>
<td align="left">Simple_repeat</td>
</tr>
<tr class="even">
<td align="left">5S</td>
<td align="right">-1.405084</td>
<td align="right">0.6721832</td>
<td align="right">1</td>
<td align="left">rRNA</td>
<td align="left">rRNA</td>
</tr>
<tr class="odd">
<td align="left">7SLRNA</td>
<td align="right">-1.598665</td>
<td align="right">0.0323175</td>
<td align="right">1</td>
<td align="left">srpRNA</td>
<td align="left">srpRNA</td>
</tr>
<tr class="even">
<td align="left">CarLTR1A2B</td>
<td align="right">-1.363858</td>
<td align="right">0.7539420</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERV1</td>
</tr>
<tr class="odd">
<td align="left">CfERV1b_LTR</td>
<td align="right">-1.194750</td>
<td align="right">0.7512672</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERV1</td>
</tr>
<tr class="even">
<td align="left">CfERVF1_LTR</td>
<td align="right">-1.222406</td>
<td align="right">0.7152418</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERV1</td>
</tr>
<tr class="odd">
<td align="left">DNA1_Mam</td>
<td align="right">-1.104721</td>
<td align="right">0.9054235</td>
<td align="right">1</td>
<td align="left">DNA</td>
<td align="left">TcMar</td>
</tr>
<tr class="even">
<td align="left">Eulor5A</td>
<td align="right">-1.665583</td>
<td align="right">0.6233808</td>
<td align="right">1</td>
<td align="left">DNA?</td>
<td align="left">DNA?</td>
</tr>
<tr class="odd">
<td align="left">L1M3b</td>
<td align="right">-1.215819</td>
<td align="right">0.6528945</td>
<td align="right">1</td>
<td align="left">LINE</td>
<td align="left">L1</td>
</tr>
<tr class="even">
<td align="left">L1M3e</td>
<td align="right">-1.278423</td>
<td align="right">0.6166096</td>
<td align="right">1</td>
<td align="left">LINE</td>
<td align="left">L1</td>
</tr>
<tr class="odd">
<td align="left">LTR109A1</td>
<td align="right">-1.536070</td>
<td align="right">0.6485770</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERV1</td>
</tr>
<tr class="even">
<td align="left">LTR81C</td>
<td align="right">-1.375129</td>
<td align="right">0.5548298</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">Gypsy</td>
</tr>
<tr class="odd">
<td align="left">LTR91</td>
<td align="right">-1.879468</td>
<td align="right">0.5683125</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERVL</td>
</tr>
<tr class="even">
<td align="left">MamGypLTR2c</td>
<td align="right">-1.111055</td>
<td align="right">0.7839170</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">Gypsy</td>
</tr>
<tr class="odd">
<td align="left">MER45R</td>
<td align="right">-1.272985</td>
<td align="right">0.6944189</td>
<td align="right">1</td>
<td align="left">DNA</td>
<td align="left">hAT-Tip100</td>
</tr>
<tr class="even">
<td align="left">MER74C</td>
<td align="right">-1.256894</td>
<td align="right">0.7651719</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERVL</td>
</tr>
<tr class="odd">
<td align="left">MER87</td>
<td align="right">-1.367775</td>
<td align="right">0.8284964</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERV1</td>
</tr>
<tr class="even">
<td align="left">MLT1A0-int</td>
<td align="right">-1.083928</td>
<td align="right">0.8646201</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERVL-MaLR</td>
</tr>
<tr class="odd">
<td align="left">MLT1A1</td>
<td align="right">-1.892689</td>
<td align="right">0.5746772</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERVL-MaLR</td>
</tr>
<tr class="even">
<td align="left">MLT1G1-int</td>
<td align="right">-1.273883</td>
<td align="right">0.5981955</td>
<td align="right">1</td>
<td align="left">LTR</td>
<td align="left">ERVL-MaLR</td>
</tr>
</tbody>
</table>
<p>DUXC-Alt suppressed many of ERVL and ERVL-MaLR (MLT1) elements, though weak <em>p</em>-values are presented.</p>

</div>
</div>
</div>
<h3># References</h3>
<div id="refs" class="references">
<div id="ref-whidden2017">
<p>Jennifer L. Whiddon, Chao-Jen Wong, Ashlee T. Langford1. 2017. “Conservation and Innovation in the Dux4-Family Gene Network.” <em>Nature Genetic</em>. <a href="https://doi.org/10.1038/ng.3846">https://doi.org/10.1038/ng.3846</a>.</p>
</div>
<div id="ref-franke2017">
<p>Vedran Franke, Rosa Karlic, Sravya Ganesh. 2017. “Long Terminal Repeats Power Evolution of Genes and Gene Expression Programs in Mammalian Oocytes and Zygotes.” <em>Genomic Research</em>, no. 27. <a href="https://doi.org/10.1101/gr.216150.116">https://doi.org/10.1101/gr.216150.116</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="repeats.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="non-LTR.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/FredHutch/canFam3.DuxFamily/tree/master/gitbook/07-retrotransposon-transcriptome.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/FredHutch/canFam3.DuxFamily/tree/master/gitbook/07-retrotransposon-transcriptome.Rmd",
"text": null
},
"download": ["gitbook.pdf", "gitbook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
