<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</title>
  <meta name="description" content="Bioinformatics and statistics analysis" />
  <meta name="generator" content="bookdown 0.18 and GitBook 2.6.7" />

  <meta property="og:title" content="References | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Bioinformatics and statistics analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD" />
  
  <meta name="twitter:description" content="Bioinformatics and statistics analysis" />
  

<meta name="author" content="Chao-Jen Wong" />


<meta name="date" content="2021-11-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="peaks-line1.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />












<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DUXC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#nomenclature"><i class="fa fa-check"></i><b>0.1</b> Nomenclature</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#datasets"><i class="fa fa-check"></i><b>0.2</b> Datasets</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#annotation"><i class="fa fa-check"></i><b>0.3</b> Annotation</a><ul>
<li class="chapter" data-level="0.3.1" data-path="index.html"><a href="index.html#canfam3-ensembl-gtf"><i class="fa fa-check"></i><b>0.3.1</b> canFam3 ensembl GTF</a></li>
</ul></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>0.4</b> Software</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="chipseq.html"><a href="chipseq.html"><i class="fa fa-check"></i><b>1</b> ChIP-seq for the canine model</a><ul>
<li class="chapter" data-level="1.1" data-path="chipseq.html"><a href="chipseq.html#chip-seq-samples"><i class="fa fa-check"></i><b>1.1</b> ChIP-seq samples</a></li>
<li class="chapter" data-level="1.2" data-path="chipseq.html"><a href="chipseq.html#analysis-workflow"><i class="fa fa-check"></i><b>1.2</b> Analysis workflow</a></li>
<li class="chapter" data-level="1.3" data-path="chipseq.html"><a href="chipseq.html#peaks-and-functional-distribution"><i class="fa fa-check"></i><b>1.3</b> Peaks and functional distribution</a><ul>
<li class="chapter" data-level="1.3.1" data-path="chipseq.html"><a href="chipseq.html#number-of-peaks"><i class="fa fa-check"></i><b>1.3.1</b> Number of peaks</a></li>
<li class="chapter" data-level="1.3.2" data-path="chipseq.html"><a href="chipseq.html#functional-discribution"><i class="fa fa-check"></i><b>1.3.2</b> Functional discribution</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chipseq.html"><a href="chipseq.html#overlap-of-peaks"><i class="fa fa-check"></i><b>1.4</b> Overlap of peaks</a><ul>
<li class="chapter" data-level="1.4.1" data-path="chipseq.html"><a href="chipseq.html#duxc-and-dux4"><i class="fa fa-check"></i><b>1.4.1</b> DUXC and DUX4</a></li>
<li class="chapter" data-level="1.4.2" data-path="chipseq.html"><a href="chipseq.html#duxc-and-duxc-alt"><i class="fa fa-check"></i><b>1.4.2</b> DUXC and DUXC-ALT</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="chipseq.html"><a href="chipseq.html#motifs-analysis"><i class="fa fa-check"></i><b>1.5</b> Motifs analysis</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="makeSE.html"><a href="makeSE.html"><i class="fa fa-check"></i><b>2</b> RNA-seq to DESeq2</a><ul>
<li class="chapter" data-level="2.1" data-path="makeSE.html"><a href="makeSE.html#rna-seq-preprocessing"><i class="fa fa-check"></i><b>2.1</b> RNA-seq preprocessing</a></li>
<li class="chapter" data-level="2.2" data-path="makeSE.html"><a href="makeSE.html#reads-counts"><i class="fa fa-check"></i><b>2.2</b> Reads counts</a><ul>
<li class="chapter" data-level="2.2.1" data-path="makeSE.html"><a href="makeSE.html#define-gene-features"><i class="fa fa-check"></i><b>2.2.1</b> Define gene features</a></li>
<li class="chapter" data-level="2.2.2" data-path="makeSE.html"><a href="makeSE.html#read-counts"><i class="fa fa-check"></i><b>2.2.2</b> Read counts</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="makeSE.html"><a href="makeSE.html#deseq2"><i class="fa fa-check"></i><b>2.3</b> DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html"><i class="fa fa-check"></i><b>3</b> DUXC transcriptome</a><ul>
<li class="chapter" data-level="3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#load-datasets-and-settings"><i class="fa fa-check"></i><b>3.1</b> Load datasets and settings</a></li>
<li class="chapter" data-level="3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#deseq2-1"><i class="fa fa-check"></i><b>3.2</b> DESeq2</a><ul>
<li class="chapter" data-level="3.2.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hypothesis-settings"><i class="fa fa-check"></i><b>3.2.1</b> hypothesis settings</a></li>
<li class="chapter" data-level="3.2.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cinc-duxc-expresison-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2.2</b> CinC: DUXC expresison in canine muscle cells</a></li>
<li class="chapter" data-level="3.2.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#hinc-dux4-expression-in-canine-muscle-cells"><i class="fa fa-check"></i><b>3.2.3</b> HinC: DUX4 expression in canine muscle cells</a></li>
<li class="chapter" data-level="3.2.4" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#duxc-alt-expression---caltinc"><i class="fa fa-check"></i><b>3.2.4</b> DUXC-ALT expression - CALTinC</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#cross-species-models-dux4-expression-in-canine-and-murine"><i class="fa fa-check"></i><b>3.3</b> Cross-species models: DUX4 expression in canine and murine</a><ul>
<li class="chapter" data-level="3.3.1" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-duxc-expression-in-canine"><i class="fa fa-check"></i><b>3.3.1</b> DUX4 and DUXC expression in canine</a></li>
<li class="chapter" data-level="3.3.2" data-path="DUXC-transcriptome.html"><a href="DUXC-transcriptome.html#dux4-and-dux-expression-in-murine"><i class="fa fa-check"></i><b>3.3.2</b> DUX4 and Dux expression in murine</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html"><i class="fa fa-check"></i><b>4</b> DUXC and Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#duxc-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.1</b> DUXC transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.1.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs"><i class="fa fa-check"></i><b>4.1.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.1.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome"><i class="fa fa-check"></i><b>4.1.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#dux-transcriptome-compared-to-that-of-fshd"><i class="fa fa-check"></i><b>4.2</b> Dux transcriptome compared to that of FSHD</a><ul>
<li class="chapter" data-level="4.2.1" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#get-canine-human-homologs-1"><i class="fa fa-check"></i><b>4.2.1</b> Get canine-human homologs</a></li>
<li class="chapter" data-level="4.2.2" data-path="DUXC-FSHD-transcriptome.html"><a href="DUXC-FSHD-transcriptome.html#gsea-enrichment-of-fshd-transcriptome-1"><i class="fa fa-check"></i><b>4.2.2</b> GSEA: enrichment of FSHD transcriptome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html"><i class="fa fa-check"></i><b>5</b> DUXC transcriptome compared to early embryo transcriptome profile</a><ul>
<li class="chapter" data-level="5.1" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#mouse-2c-like-signature"><i class="fa fa-check"></i><b>5.1</b> Mouse 2C-like signature</a></li>
<li class="chapter" data-level="5.2" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#cleavage-stage-signature"><i class="fa fa-check"></i><b>5.2</b> cleavage-stage signature</a></li>
<li class="chapter" data-level="5.3" data-path="DUXC-early-embryo.html"><a href="DUXC-early-embryo.html#ipscs"><i class="fa fa-check"></i><b>5.3</b> iPSCs</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="repeats.html"><a href="repeats.html"><i class="fa fa-check"></i><b>6</b> Repeat elements</a><ul>
<li class="chapter" data-level="6.1" data-path="repeats.html"><a href="repeats.html#takeaway"><i class="fa fa-check"></i><b>6.1</b> Takeaway</a></li>
<li class="chapter" data-level="6.2" data-path="repeats.html"><a href="repeats.html#repeat-element-expression"><i class="fa fa-check"></i><b>6.2</b> Repeat element expression</a></li>
<li class="chapter" data-level="6.3" data-path="repeats.html"><a href="repeats.html#differential-analysis"><i class="fa fa-check"></i><b>6.3</b> Differential analysis</a><ul>
<li class="chapter" data-level="6.3.1" data-path="repeats.html"><a href="repeats.html#normalization"><i class="fa fa-check"></i><b>6.3.1</b> Normalization</a></li>
<li class="chapter" data-level="6.3.2" data-path="repeats.html"><a href="repeats.html#deseq2-results"><i class="fa fa-check"></i><b>6.3.2</b> DESeq2 results</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="repeats.html"><a href="repeats.html#repeat-family-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> Repeat family enrichment analysis</a><ul>
<li class="chapter" data-level="6.4.1" data-path="repeats.html"><a href="repeats.html#correlation-between-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>6.4.1</b> Correlation between HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="6.4.2" data-path="repeats.html"><a href="repeats.html#repeat-family-activation"><i class="fa fa-check"></i><b>6.4.2</b> Repeat family activation</a></li>
<li class="chapter" data-level="6.4.3" data-path="repeats.html"><a href="repeats.html#repeat-classs-activation"><i class="fa fa-check"></i><b>6.4.3</b> Repeat classs activation</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="repeats.html"><a href="repeats.html#ltr-retroelements-distribution-and-activation"><i class="fa fa-check"></i><b>6.5</b> LTR retroelements distribution and activation</a></li>
<li class="chapter" data-level="6.6" data-path="repeats.html"><a href="repeats.html#peaks-distribution-in-class-of-repeat-elements"><i class="fa fa-check"></i><b>6.6</b> Peaks distribution in class of repeat elements</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="LTR-activation.html"><a href="LTR-activation.html"><i class="fa fa-check"></i><b>7</b> LTR activation</a><ul>
<li class="chapter" data-level="7.1" data-path="LTR-activation.html"><a href="LTR-activation.html#takaway"><i class="fa fa-check"></i><b>7.1</b> Takaway</a></li>
<li class="chapter" data-level="7.2" data-path="LTR-activation.html"><a href="LTR-activation.html#background-in-murine-and-human"><i class="fa fa-check"></i><b>7.2</b> background in murine and human</a></li>
<li class="chapter" data-level="7.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-subfamily-activation-in-canine"><i class="fa fa-check"></i><b>7.3</b> ERVL subfamily activation in canine</a></li>
<li class="chapter" data-level="7.4" data-path="LTR-activation.html"><a href="LTR-activation.html#malr-activation-in-canine"><i class="fa fa-check"></i><b>7.4</b> MaLR activation in canine</a></li>
<li class="chapter" data-level="7.5" data-path="LTR-activation.html"><a href="LTR-activation.html#compare-to-cinc-and-hinc-to-hinh"><i class="fa fa-check"></i><b>7.5</b> Compare to CinC and HinC to HinH</a><ul>
<li class="chapter" data-level="7.5.1" data-path="LTR-activation.html"><a href="LTR-activation.html#copy-number"><i class="fa fa-check"></i><b>7.5.1</b> Copy number</a></li>
<li class="chapter" data-level="7.5.2" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl-malr"><i class="fa fa-check"></i><b>7.5.2</b> ERVL-MaLR</a></li>
<li class="chapter" data-level="7.5.3" data-path="LTR-activation.html"><a href="LTR-activation.html#ervl"><i class="fa fa-check"></i><b>7.5.3</b> ERVL</a></li>
<li class="chapter" data-level="7.5.4" data-path="LTR-activation.html"><a href="LTR-activation.html#erv1"><i class="fa fa-check"></i><b>7.5.4</b> ERV1</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="LTR-activation.html"><a href="LTR-activation.html#duxc-alt-and-ervl-transcriptions"><i class="fa fa-check"></i><b>7.6</b> DUXC-ALT and ERVL transcriptions</a><ul>
<li class="chapter" data-level="7.6.1" data-path="LTR-activation.html"><a href="LTR-activation.html#erv-subfamily-expression"><i class="fa fa-check"></i><b>7.6.1</b> ERV subfamily expression</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="non-LTR.html"><a href="non-LTR.html"><i class="fa fa-check"></i><b>8</b> Non-LTR retrotrasposons and satellites</a><ul>
<li class="chapter" data-level="8.1" data-path="non-LTR.html"><a href="non-LTR.html#lines"><i class="fa fa-check"></i><b>8.1</b> LINEs</a><ul>
<li class="chapter" data-level="8.1.1" data-path="non-LTR.html"><a href="non-LTR.html#l1s-activation"><i class="fa fa-check"></i><b>8.1.1</b> L1s activation</a></li>
<li class="chapter" data-level="8.1.2" data-path="non-LTR.html"><a href="non-LTR.html#line1-and-duxcdux4-binding-sites"><i class="fa fa-check"></i><b>8.1.2</b> LINE1 and DUXC/DUX4 binding sites</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="non-LTR.html"><a href="non-LTR.html#sine"><i class="fa fa-check"></i><b>8.2</b> SINE</a><ul>
<li class="chapter" data-level="8.2.1" data-path="non-LTR.html"><a href="non-LTR.html#distribution-of-sine-family"><i class="fa fa-check"></i><b>8.2.1</b> Distribution of SINE family</a></li>
<li class="chapter" data-level="8.2.2" data-path="non-LTR.html"><a href="non-LTR.html#expression-of-sines-in-hinc-and-cinc-transcriptome"><i class="fa fa-check"></i><b>8.2.2</b> Expression of SINEs in HinC and CinC transcriptome</a></li>
<li class="chapter" data-level="8.2.3" data-path="non-LTR.html"><a href="non-LTR.html#trna-lys"><i class="fa fa-check"></i><b>8.2.3</b> tRNA-Lys</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="non-LTR.html"><a href="non-LTR.html#satellites"><i class="fa fa-check"></i><b>8.3</b> Satellites</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="peaks-line1.html"><a href="peaks-line1.html"><i class="fa fa-check"></i><b>9</b> DUXC peaks and LINE1s</a><ul>
<li class="chapter" data-level="9.1" data-path="peaks-line1.html"><a href="peaks-line1.html#narrow-peaks-and-expressed-l1s"><i class="fa fa-check"></i><b>9.1</b> Narrow Peaks and expressed L1s</a><ul>
<li class="chapter" data-level="9.1.1" data-path="peaks-line1.html"><a href="peaks-line1.html#motif-discovery-using-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.1</b> Motif discovery using L1-overlapped peaks</a></li>
<li class="chapter" data-level="9.1.2" data-path="peaks-line1.html"><a href="peaks-line1.html#match-dux4-and-duxc-motif-to-l1-overapped-peaks"><i class="fa fa-check"></i><b>9.1.2</b> Match DUX4 and DUXC motif to L1-overapped peaks</a></li>
<li class="chapter" data-level="9.1.3" data-path="peaks-line1.html"><a href="peaks-line1.html#location-of-aatca-on-l1-overlapped-peaks"><i class="fa fa-check"></i><b>9.1.3</b> Location of AATCA on L1-overlapped peaks</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="peaks-line1.html"><a href="peaks-line1.html#line1s-and-aatca"><i class="fa fa-check"></i><b>9.2</b> LINE1s and AATCA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Canine DUXC: Implications for DUX4 retrotransposition and preclinical models of FSHD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered">
<h1>References</h1>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="peaks-line1.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/FredHutch/canFam3.DuxFamily/tree/master/gitbook/11-References.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/FredHutch/canFam3.DuxFamily/tree/master/gitbook/11-References.Rmd",
"text": null
},
"download": ["gitbook.pdf", "gitbook.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
